{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "luVoVmH3IZSh",
        "outputId": "1a565109-e0f7-4de3-d956-78548faa6e2e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting sacremoses\n",
            "  Downloading sacremoses-0.1.1-py3-none-any.whl.metadata (8.3 kB)\n",
            "Requirement already satisfied: regex in /usr/local/lib/python3.11/dist-packages (from sacremoses) (2024.11.6)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.11/dist-packages (from sacremoses) (8.1.8)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from sacremoses) (1.4.2)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from sacremoses) (4.67.1)\n",
            "Downloading sacremoses-0.1.1-py3-none-any.whl (897 kB)\n",
            "\u001b[?25l   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/897.5 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m897.5/897.5 kB\u001b[0m \u001b[31m42.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: sacremoses\n",
            "Successfully installed sacremoses-0.1.1\n"
          ]
        }
      ],
      "source": [
        "!pip install sacremoses"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "4ro0lliZCR6g"
      },
      "outputs": [],
      "source": [
        "\n",
        "import pandas as pd\n",
        "import numpy as np\n",
        "import torch\n",
        "from torch.utils.data import Dataset\n",
        "from transformers import BertTokenizer, BertForSequenceClassification, Trainer, TrainingArguments, BioGptTokenizer, BioGptForCausalLM\n",
        "from sklearn.model_selection import train_test_split\n",
        "from sklearn.utils.class_weight import compute_class_weight\n",
        "from sklearn.metrics import accuracy_score, f1_score, precision_score, recall_score\n",
        "\n",
        "# Load data\n",
        "train_variants = pd.read_csv(\"training_variants.csv\")\n",
        "train_text = pd.read_csv(\"training_text.csv\", sep='\\|\\|', engine='python', header=None, skiprows=1, names=['ID', 'Text'])\n",
        "train = pd.merge(train_variants, train_text, on='ID', how='inner')\n",
        "\n",
        "\n",
        "# Remove empty texts\n",
        "train = train[train['Text'] != \"\"]\n",
        "\n",
        "# Ensure labels are between 1 and 9\n",
        "assert train['Class'].min() == 1 and train['Class'].max() == 9, \"Class labels are not in the range 1-9\"\n",
        "\n",
        "# Ensure all text values are strings\n",
        "train['Text'] = train['Text'].astype(str).fillna(\"\")\n",
        "\n",
        "# Define device\n",
        "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
        "\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "def basic_stats(df):\n",
        "    print(\"Dataset Shape\", df.shape)\n",
        "    print(f\"\\nMissing Vlaues: \\n{df.isnull().sum()} \")\n",
        "    print(f\"\\nClass Distribution:{df['Class'].value_counts().sort_index()}\")\n",
        "\n",
        "    #Gene statistics\n",
        "    print(\"\\nNumber of unique genes: \", df['Gene'].nunique())\n",
        "    print(f\"\\nTop 10 most common genes:{df['Gene'].value_counts().head(10)} \")\n",
        "\n",
        "    #Variation statistics\n",
        "    print(f\"\\nNumber of unique variations:{df['Variation'].nunique()}\")\n",
        "    print(f\"\\nTop 10 most common variations: {df['Variation'].value_counts().head(10)}\")\n",
        "\n",
        "print(\"=== Basic Statistics ===\")\n",
        "basic_stats(train)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 398
        },
        "id": "80sXZihTBA-9",
        "outputId": "0cbba390-dbcc-4ef4-a530-3bc085165099"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>count</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Class</th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>953</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>686</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>568</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>452</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>275</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>242</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>89</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>37</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>19</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div><br><label><b>dtype:</b> int64</label>"
            ],
            "text/plain": [
              "Class\n",
              "7    953\n",
              "4    686\n",
              "1    568\n",
              "2    452\n",
              "6    275\n",
              "5    242\n",
              "3     89\n",
              "9     37\n",
              "8     19\n",
              "Name: count, dtype: int64"
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "\n",
        "\n",
        "# Count the occurrences of each class label\n",
        "class_counts = train['Class'].value_counts()\n",
        "\n",
        "class_counts"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 597
        },
        "id": "yNOS_sdwGwFm",
        "outputId": "57bf6b79-db84-4b5a-fec9-4cb15f8eec9f"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "summary": "{\n  \"name\": \"train\",\n  \"rows\": 3321,\n  \"fields\": [\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 958,\n        \"min\": 0,\n        \"max\": 3320,\n        \"num_unique_values\": 3321,\n        \"samples\": [\n          1057,\n          812,\n          2658\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 264,\n        \"samples\": [\n          \"PDGFRA\",\n          \"KEAP1\",\n          \"ARID1A\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Variation\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2996,\n        \"samples\": [\n          \"E1384K\",\n          \"H597Y\",\n          \"V851A\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Class\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 1,\n        \"max\": 9,\n        \"num_unique_values\": 9,\n        \"samples\": [\n          8,\n          2,\n          6\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1921,\n        \"samples\": [\n          \"An internal tandem duplication (ITD) of the FLT3 gene is found in nearly 20% of acute myeloid leukemia (AML) and 5% of myelodysplastic syndrome cases. Our serial studies on 51 samples with the FLT3 gene mutation indicated that the ITD was frequently (47/51) clustered in the tyrosine-rich stretch from codon 589 to 599 and rarely (3/51) in its downstream region, both of which are located within the juxtamembrane (JM) domain. One remaining sample had an insertion into the JM domain of nucleotides of unknown origin. To eludicate the biological relevance of the ITD or the insertion, we expressed various types of mutant FLT3 in Cos 7 cells. All mutant FLT3 studied were ligand-independently dimerized and their tyrosine residues were phosphorylated. The Y589 of FLT3 was essential for the phosphorylation in the wild FLT3, but a Y589F conversion did not affect the phosphorylation status of the mutant FLT3. These findings suggest that the elongation of the JM domain rather than increase of tyrosine residues causes gainof-function of FLT3. Thus, ITD is a novel modality of somatic mutation which activates its product. Since the DNA corresponding to codon 593 to 602 potentially forms a palindromic intermediate, we propose that a DNA-replication error might be associated with generating the ITD of the FLT3 gene. Keywords: tandem duplication; length mutation; FLT3; acute myeloid leukemia; phosphorylation Introduction FLT3, a classs III receptor tyrosine kinase (RTK), along with KIT, FMS and PDGF-R, has an extracellular region composed of five immunoglobulin-like domains, a single transmembrane region, a juxtamembrane (JM) domain, two tyrosine kinase domains (TK1 and TK2) spaced by a kinase insert domain, and a C-terminal domain.1,2 FLT3 is preferentially expressed in early hematopoietic progenitors in addition to the central nervous system, placenta and liver.2,3 FLT3 is activated by its ligand (FL) which is expressed as a membrane-bound or soluble form by bone marrow stroma cells.4 FLT3 molecules are dimerized by the FL-binding, and are phosphorylated on its intracellular tyrosine residues. The phosphorylated FLT3 transduces activation signals through association with various cytoplasmic proteins, including RAS GTPase-activating protein, phospholipase C-g, Vav, Shc, Cbl, Grb2 and Src family tyrosine kinases.5\\u20137 Most clinical samples from acute myeloid leukemia (AML) and pre-B acute lymphoblastic leukemia (ALL) also express FLT3. Exogenous FL stimulates the proliferation of AML cells rather than ALL cells, although the mechanism of different responsiveness has been little elucidated.7,8 Recently, an internal tandem duplication (ITD) of the FLT3 Correspondence: T Naoe, Department of Infectious Diseases, Nagoya University School of Medicine, Nagoya 466-8560, Japan; Fax: 81 52 744 2801 Received 19 May 1998; accepted 25 May 1998 gene (FLT3/ITD) was found in AML.9 In FLT3/ITD, a fragment of the JM domain-coding sequence, which primarily consisted of exon 11 but sometimes included intron 11 and exon 12, was arranged as a direct head-to-tail succession. Its location and length varied from sample to sample, but the duplicated sequence was always in frame.9\\u201311 Thus, mutant FLT3 contained a long JM domain although other domains were unaffected. Clinically, FLT3/ITD was observed in nearly 20% of de novo AML cases regardless of the French\\u2013American\\u2013British (FAB) classification and in 5% of myelodysplastic syndrome (MDS) cases.11 Its presence was related to leukocytosis in the M3 subtype of AML.10 FLT3/ITD emerged during progression of MDS12 or at relapse of AML (unpublished data), suggesting an association with leukemia progression rather than initiation. Although FLT3/ITD is one of the most frequent gene mutations found in acute myeloid leukemia, mutant FLT3 products and their biological significance have not been characterized. In this study, we summarized the FLT3 gene mutations from a total of 51 AML and MDS samples. To elucidate the biological relevance of the mutations, we analyzed the expression and phosphorylation of mutant FLT3 transfected in Cos 7 cells. We showed that elongation of the JM sequence was important to gain-of-function of FLT3. Thus, this is the first report to show that ITD is a novel modality of somatic mutation which elongates the critical sequences of proto-oncogene and activates its product. Materials and methods Leukemia samples and detection of FLT3/ITD High molecular weight DNA was extracted from leukemia cells, and a DNA fragment covering the JM domain was amplified by polymerase chain reaction (PCR) as previously described.10 Abnormal sized-bands were cut out from the agarose gel, purified with a Qiaex gel extraction kit (Qiagen, Chatsworth, CA, USA), and cloned into the pMOSBlue T-vector (Amersham, Bucks, UK) as recommended by the manufacturer. Ten recombinant colonies were chosen and cultured in LB medium. Plasmid DNA was prepared using a QIAprep spin plasmid miniprep kit (Qiagen), and both strands were sequenced using fluorescein-conjugated 21M13 and T7 primers on a DNA sequencer (377; Applied Biosystems, Foster City, CA, USA). Expression of the FLT3 transcripts was examined using the reverse transcriptase-mediated PCR (RT-PCR) method as previously described.10 Abnormal-sized products were cloned and sequenced as above to confirm that the transcript was in-frame. Constitutive phosphorylation of mutant FLT3 H Kiyoi et al 1334 Analysis of transfectants A MunI\\u2013EcoRV fragment covering the tandem repeat of the mutated FLT3 cDNA was amplified from leukemia cells by RT-PCR using the following primer pairs: MunI-F (5\\u2032-CAACAATTGGTGTTTGTCTCCTCTT-3\\u2032) and EcoRV-R (5\\u2032-CATGATATCTCGAGCCAATCCAAAG-3\\u2032). Amplified products were digested by a combination of MunI and EcoRV endonucleases (Boehringer Mannheim Yamanouchi, Tokyo, Japan), separated on an agarose gel, and purified as described above. The expression vector pCDHF3, which contains a full-length cDNA of the FLT3 gene, was kindly provided by Dr Oliver Rosnet (INSERM, France). A fragment from the 1640 nt to the 2503 nt was removed by MunI and EcoRV digestion. The purified mutated fragment was cloned into the digested pCDHF3 vector at the MunI and EcoRV sites. Point mutations which changed tyrosine residues Y589 and Y591 to phenylalanine residues were introduced into the wild and mutated FLT3 sequence cloned into the pCDHF3 vector with the oligonucleotides 5\\u2032-CCTCAGATAATGAGTTCTTCTACGTTGATTTCAG-3\\u2032 and 5\\u2032-CTGAAATCAACGTAGAAGAACTCATTATCTGAGG-3\\u2032 for Y589F and 5\\u2032-CCTCAGATAATGAGTACTTCTTCGTTGATTTCAG-3\\u2032 and 5\\u2032-CTGAAATCAACGAAGAAGTACTCATTATCTGAGG-3\\u2032 for Y591F using the QuikChange Site-Directed Mutagenesis Kit (Stratagene, La Jolla, CA, USA). All constructs were confirmed by sequencing. The wild and mutated constructs were transfected to Cos 7 cells by Lipofection (Gibco BRL, Gaithersburg, MD, USA) according to the manufacturer\\u2019s instructions. After 72 h culture, cells were lysed in lysed buffer: 20 mM Tris-Hcl, pH 7.5, 150 mM NaCl, 2 mM EDTA, 1% Nonidet P-40, 0.1% sodium dodecyl sulfate (SDS), 50 mM NaF, 10 mg/ml aprotinin, 10 mg/ml leupeptin, 1 mM Na3VO4, 50 mM Na2MoO4 and 1 mM phenylmethylsulfonyl fluoride (PMSF). After incubation for 1 h at 4\\u00b0C, lysates were spun at 15 000 g at 4\\u00b0C for 30 min, and pellets were discarded. To study the effect of FL, cells were stimulated by FL at a concentration of 50 ng/ml for 10 min, then lysed as above. A portion of the lysed samples separated by SDS-polyacrylamide gel electrophoresis (SDSPAGE), and electroblotted on to Immobilon PVDF membranes (Millipore, Bedford, MA, USA). The membranes were blocked overnight at 4\\u00b0C in 1% blocking reagent (Boehringer Mannheim Yamanouchi) in 0.1 M maleic acid and 0.15 M NaCl, pH 7.5, and incubated in mouse anti-phosphotyrosine antibody (PY-20; Takara, Kyoto, Japan or 4G10; Upstate Biotechnology, Lake Placid, NY, USA) for 1 h, then in HRPlabeled goat anti-mouse IgG antibody (BioRad, Hercules, CA, USA) for 1 h. Signals were developed using ECL chemiluminescent reagents (Amersham). After developing the signal, we incubated the membrane in stripping buffer (62.5 mM TrisHCl, pH 6.8, 2% SDS and 100 mM 2-mercaptoethanol) at 70\\u00b0C for 30 min, and reprobed it with rabbit anti-human FLT3 antibody (Santa Cruz Biotechnology, Santa Cruz, CA, USA) for 1 h, then HRP-labeled goat anti-rabbit IgG antibody (BioRad) for 1 h. The remaining lysates were incubated with anti-FLT3 antibody and precipitated with protein A/G Plusagarose (Santa Cruz Biotechnology). The precipitated samples were separated and electroblotted. The membrane was probed with anti-phosphotyrosine antibody, then reprobed with anti-FLT3 antibody as described above. Chemical cross-linking Dimerization of FLT3 was analyzed by chemical cross-linking. Cos 7 cells were transfected with wild and mutated FLT3 as described above, and cultured for 72 h. The transfected cells were incubated with or without FL (50 ng/ml) for 90 min at 4\\u00b0C, and were washed five times in ice-cold PBS. Cross-linking was performed in PBS with BS3 (1 mM) for 30 min at 22\\u00b0C. The reaction was stopped by adding glycine-HCl (pH 7.5) to the final concentration of 150 mM and incubating for 5 min. Sodium orthovanadate (100 mM) was included in all buffers. The cells were lysed, subjected to the SDS-PAGE and blotted on to the membrane as described above. The membrane was incubated in mouse anti-phosphotyrosine antibody for 1 h, then in HRPlabeled goat anti-mouse IgG antibody (BioRad) for 1 h. Signals were developed as described above. After developing the signal, the membrane was incubated in stripping buffer, and reprobed with rabbit anti-human FLT3 antibody. Results Individual variation of FLT3/ITD Through our serial studies on FLT3/ITD, we have sequenced FLT3/ITD from a total of 50 AML and MDS samples. We determined which tyrosine residue(s) (denoted by Y) were involved in ITD, since the JM domain contains a Y-rich stretch, 589YFYVDFREYEY599, associated with signal transduction. In 47 of the 50 samples, all or some of these Y were duplicated in mutant FLT3; most frequently Y589/- Y591/Y597/Y599 followed by Y597/Y599. None of the samples had duplication of the fragment upsteam to Y589, but the downstream fragments were involved in five of the 50 samples. Notably, in one sample which was not reported in our previous studies, a 36-bp insertion of unknown origin was observed in exon 11 (the sequence is presented as Mt6 in Figure 1a). These data are summarized in Table 1. Constitutive activation of mutant FLT3 expressed in Cos 7 cells To analyse the phosphorylation status of mutant FLT3 it was immunoprecipitated from the clinical samples. However, significant signals were not obtained by immunoblot analysis (data not shown). Then the wild or the mutant FLT3 cDNA was transfected into Cos 7 cells. Since ITD varied from sample to sample as to position and size, six types of FLT3/ITD were chosen. Mutants 1 to 5 (Mt1\\u2013Mt5) had ITD including Y597/Y599, Y589/Y591/Y597/Y599 (long), Y589/Y591/Y597/- Y599 (short), Y591/Y597/Y599, Y589/Y591/Y597, respectively. Mutant 6 (Mt6) contained the 36 bp insertion of unknown origin (Figure 1a). The wild type of FLT3 was detected as a major band at 130 kDa and a minor band at 155 kDa by immunoblot analysis using total cell lysates (Figure 1b), which corresponds to the published data. Both bands of FLT3 were immunoprecipitated with anti-FLT3 antibody and detected by immunoblot analysis. The wild type of FLT3 was significantly phosphorylated after the FL-stimulation. Each mutant type of FLT3 was observed as the same size and the same amount of bands as the wild FLT3 by immunoblot analysis using total cell lysates (Figure 1b). However, the anti-FLT3 antibody (anti-C-terminal 20 amino acids) failed to immunoprecipitate FLT3 from Mt1. Other mutant types of FLT3 were also precipitated to a lesser degree than the wild FLT3. These findings suggest that the mutant FLT has a conformational change of its C-terminus region. Constitutive phosphorylation of mutant FLT3 H Kiyoi et al 1335 Figure 1 (a) Amino acid sequences duplicated by FLT3/ITD (Mt1 to Mt5) or inserted (Mt6). Germ line amino acid sequence of the JM is presented at the top. The TK domain starts at 608E. Non-coded amino acids sequences were added between the duplication due to various mechanisms as described previously. (b) Tyrosine-phosphorylation of FLT3 expressed in Cos 7 cells. The wild or the mutant type of FLT3 cDNA was transfected into Cos 7 cells. The lysates were immunoprecipitated by the anti-FLT3 antibody and PA-agarose. The precipitated samples were detected by immunoblot analysis using anti-FLT3 antibody or anti-phosphotyrosine antibody. IP, immunoprecipitation; IB, immunoblot. (c) Constitutive tyrosine-phosphorylation of the wild FLT3 by the co-transfection of the mutant FLT3. The wild and the mutant type of FLT3 were simultaneously transfected into Cos 7 cells. All mutant FLT3 including Mt6 were constitutively phosphorylated in a ligand-independent manner (Figure 1b). These results indicate that FLT3 was constitutively activated by these mutations. To study the effect of the mutant FLT3 on the wild FLT3, both FLT3 cDNA were simultaneously transfected into Cos 7 cells (Figure 1c). Since Mt1 was not immunoprecipitated, ligand-independent phosphorylation in the transfectant indicated the phosphorylation of the wild FLT3. This finding indicates that the wild FLT3 was phosphorylated by the mutant FLT3 expressed in the same cell. Since the dimerization of RTK is known to be a key event for the following activation, we determined whether the ITD or the insertion led to receptor dimerization without FL stimulation. Transfected Cos 7 cells were treated with BS3 that was a hydrophilic cross-linker,13 and FLT3 was detected by immunoblot analysis. An approximate molecular size of 260 kDa that corresponded to a probable homodimer of FLT3 was detected after the FL stimulation (Figure 2). On the other hand, the same sized band was detected in the mutant FLT3 without FL-stimulation. These findings suggest that the ITD or the insertion at the JM domain induced the extracellular dimerization of FLT3 without FL stimulation. Increase of tyrosine residues is unessential for constitutive phosphorylation of FLT3 On comparing the primary structures between FLT3 and PDGFR, the Y589 of FLT3 corresponds to Y579 of PDGF-R, which is known to be phosphorylated by ligand binding.14 Since the phosphorylated Y579 of PDGF-R binds with the SH2 domain of the Src family, Y589 of FLT3 might similarly be associated with signal transduction. Accordingly, we next characterized the role of Y589 and Y591 in the wild or mutant type of FLT3. Using mutated oligonucleotides, Y589 and Y591 were converted to phenylalanine (denoted by Y589F and Y591F, respectively). The ligand-dependent phosphorylation was decreased in a Y589F substitution of wild FLT3, whereas it was rather enhanced in a Y591F substitution (Figure 3a). A series of Y589F substitutions of mutant FLT3 showed constitutive phosphorylation. Similarly, a series of Y591 substitutions did not affect the phosphorylation (Figure 3b). Thus, the increase of Y residues is not necessary and elongation of the JM domain may be more important for constitutive phosphorylation. Discussion In this study, we first summarized the location of FLT3/ITD from a total of 51 AML and MDS samples to assume the biological significance of FLT3/ITD. FLT3/ITD frequently involved either YFY at codons 589 to 591 or YEY at codons 597 to 599, but did not significantly affect the tyrosine-kinase domain. When we compared the primary structures between FLT3 and PDGF-R, we found the Y589 of FLT3 corresponds to Y579 of PDGFR, which is known to be phosphorylated by ligand binding. Since the phosphorylated Y579 of PDGF binds with the SH2 domain of the Src family,14 Y589 of FLT3 might similarly be associated with signal transduction. Thus, one possibility to explain the significance of FLT3/ITD was that FLT3/ITD is associated with the amplification of FLT3 signaling. Another possibility was that FLT3/ITD causes ligand-independent activation of FLT3. Our serial studies indicated that a few samples with FLT3/ITD contain no tyrosine residue, and we also found one sample carrying an insertion of unknown origin sequence into the JM domain. Accordingly, we investigated the latter possibility in this study. The biological significance of the ITD or the nucleotide insertion was elucidated by the Cos 7 expression system. All mutant FLT3 studied were ligand-independently phosphorylated regardless of the location and length of ITD. The mutant FLT3 with the insertion was also constitutively phosphorylated. So far, several kinds of activating mutations have been reported in type III RTKs. Mutations of c-Kit in mast cell neoplasm are notable for their similarity to those of FLT3. Notably, the JM domain was affected; a V559G substitution and a deletion of 7 amino acids.13,15 Both mutant c-Kits were Constitutive phosphorylation of mutant FLT3 H Kiyoi et al 1336 Table 1 Summary of previous studies on the position of FLT3/ITD Position of Y residues involved in FLT3/ITD Nakao Kiyoi Yokota Horiike This study Total et al9 et al10 et al11 et al12 Y589, Y591, Y597, Y599 4 9 2 15 Y589, Y591, Y597, 2 1 3 Y589, Y591, 2131 7 Y589, 0 Y591, Y597, Y599 2 2 Y591, Y597, 0 Y591, 1 1 Y597, Y599 1 5 7 13 Y597, 0 Y599 1 3 2 6 Downstream to Y599 1 1 1 3 Origin unknown 1 1 Total number of samples 5 15 23 7 1 51 Figure 2 Cross-linking experiment. The wild or mutant FLT3 was cross-linked by BS3 and detected by immunoblot analysis using antiphosphotyrosine antibody. not only autophosphorylated but also dimerized between extracellular domains without ligand stimulation. On the other hand, a D814Y mutation in TK2 domain caused autonomous phosphorylation without dimerization.16 Another member of RTK type III, PDGF, is involved in the t(5;12) translocation found in chronic myelomonocytic leukemia. The resultant TEL-PDGF fusion product was an activated mutant of PDGF, because TEL-derived sequence contributed to oligomerization of PDGF-derived sequence.17 Thus, the constitutive dimerization is a key mechanism for activating mutation of RTK. FLT3/ITD also seems to cause ligand-independent dimerization of FLT3 for the following reasons. First, since the wild FLT3 was ligand-independently phosphorylated by the simultaneous transfection of the mutant FLT3, the mutant FLT3 must physically associate with the wild FLT3. Second, the crosslinking experiment showed that mutant FLT3 formed a highmolecular complex, presumably a dimer, on cell surface without FL stimulation. The JM domain mutations might induce an extracellular conformational change which favors dimer formation of FLT3. To further confirm that elongation is more important for the constitutive activation than the increase of tyrosine residues, Figure 3 The role of Y589 and Y591 in wild and mutant types of FLT3. The Y589 and Y591 were converted to phenylalanine. The precipitated samples by the anti-FLT3 antibody were detected by immunoblot analysis using anti-phosphotyrosine antibody. The ligand-dependent phosphorylation was decreased in a Y589F substitution of wild FLT3 (a). A series of Y589F substitutions of mutant FLT3 showed constitutive phosphorylation. Similarly a series of Y591 substitutions did not affect phosphorylation (b). we substituted Y589 or Y591 for phenylalanine. It was shown that a Y589F conversion did not affect the phosphorylation status of the mutant FLT3 in contrast to the wild FLT3. Why then is the ITD clustered around Y597/Y599? What is the molecular mechanism of generating ITD? Generally, when a single-stranded DNA has the potential to form complex secondary structures including hairpins, it may be responsible for replication-based mutations: deletion, addition, frameshift and duplication. That is, hairpin formation of a lagging template or strand causes replication slippage generating DNA deletion or duplication, respectively.18,19 This model has been proposed to account for microsatellite instability in mutator mutants and triplet disease such as inherited neuromuscular disease.20 We suppose it is further applicable to FLT3/ITD for the following reason. According to the sequence data from 50 samples with FLT3/ITD, duplication frequency at each tyrosine codon was as follows: 25/50 (50%) at Y589, 28/50 (56%) Constitutive phosphorylation of mutant FLT3 H Kiyoi et al 1337 Figure 4 (a) Location of the clustered region of FLT3/ITD. The JM domain is partly duplicated and elongated from 6 to 38 (median 15) amino acids by FLT3/ITD, but the other domains including TK domains are not affected. (b) A proposed mechanism for the generation of FLT3/ITD. Single-strand DNA corresponding to D593 to K602 potentially forms a palindromic intermediate. If this occurs in the lagging strand during DNA replication, FLT3/ITD would be generated by replication slippage. Numbers denote the position of amino acids. at Y591, 33/50 (66%) at Y597, 36/50 (72%) at Y599 (Table 1). Additionally, F605 was involved in 22/50 (44%) (data not shown). Accordingly, Y597 and Y599 were most frequently involved in FLT3/ITD. As shown in Figure 4, the DNA corresponding to D593 to K602 potentially forms a palindromic intermediate. If a lagging strand makes a hairpin during DNA replication and the following mismatch repair system is impaired, the tandemly duplicated fragment will be fixed in DNA. If FLT3/ITD occurs in an out-of-frame manner, the leukemia cell carrying it does not acquire a growth advantage and will not be selected. Although simple models should be inspected using various replication systems, we think the replication error is thus an underlying mechanism for FLT3/ITD. Finally, investigation of FLT3/ITD would provide a new perspective on leukemia as to genetic instability and aberrant signal transduction, as well as leukemia progression.  Somatic mutations of the receptor tyrosine kinase Flt3 consisting of internal tandem duplications (ITD) occur in 20% of patients with acute myeloid leukemia. They are associated with a poor prognosis of the disease. In this study, we characterized the oncogenic potential and signaling properties of Flt3 mutations. We constructed chimeric molecules that consisted of the murine Flt3 backbone and a 510-base pair human Flt3 fragment, which contained either 4 different ITD mutants or the wild-type coding sequence. Flt3 isoforms containing ITD mutations (Flt3- ITD) induced factor-independent growth and resistance to radiation-induced apoptosis in 32D cells. Cells containing Flt3- ITD, but not those containing wild-type Flt3 (Flt3-WT), formed colonies in methylcellulose. Injection of 32D/Flt3-ITD induced rapid development of a leukemiatype disease in syngeneic mice. Flt3-ITD mutations exhibited constitutive autophosphorylation of the immature form of the Flt3 receptor. Analysis of the involved signal transduction pathways revealed that Flt3-ITD only slightly activated the MAP kinases Erk1 and 2 and the protein kinase B (Akt) in the absence of ligand and retained ligand-induced activation of these enzymes. However, Flt3-ITD led to strong factor-independent activation of STAT5. The relative importance of the STAT5 and Ras pathways for ITD-induced colony formation was assessed by transfection of dominant negative (dn) forms of these proteins: transfection of dnSTAT5 inhibited colony formation by 50%. Despite its weak constitutive activation by Flt3-ITD, dnRas also strongly inhibited Flt3-ITD\\u2013 mediated colony formation. Taken together, Flt3-ITD mutations induce factor-independent growth and leukemogenesis of 32D cells that are mediated by the Ras and STAT5pathways.(Blood.2000;96:3907-3914) \\u00a9 2000 by The American Society of Hematology Introduction The receptor tyrosine kinase Flt31-3 and its ligand FL4-6 play an important role in survival and self-renewal of early hematopoietic progenitors and monocytic precursors and in early lymphoid development.7-9 The expression of Flt3 on leukemic cell lines and samples from patients with acute leukemia has been extensively studied. Flt3 is expressed on most acute leukemias of myeloid and B-lymphoid origin.10,11 Incubation of leukemic blasts with FL results in enhanced DNA synthesis in some, but not all cases of acute myeloid leukemia (AML)10-15 and in a reduced rate of spontaneous apoptosis of AML blasts.16 Chronic overexpression of Flt3 ligand in hematopoietic progenitors has been shown to induce hematologic malignancies after a long latency period.17 Somatic mutations involving a set of in-frame internal tandem duplications (ITDs) occur at the end of exon 11 of the Flt3 gene. These result in the duplication of a stretch of several amino acids in the juxtamembrane region of the receptor.18 ITD mutations are detectable in about 20% of AML samples18 but not in normal hematopoiesis,19 and they are associated with high leukemic cell numbers in patients with acute promyelocytic leukemia.20 The presence of ITD mutations of Flt3 on AML blasts indicates a very poor prognosis.21-23 So far, little is known about the functional consequences of ITD mutations. Expression of ITD-containing Flt3 in COS-7 cells resulted in its constitutive autophosphorylation.20 Our own results indicated that constitutive autophosphorylation of Flt3 is detectable in about 10% of AML cases.24 However, autophosphorylation did not correlate with ITD mutations. The growth of AML blasts with ITD mutations in stroma cell cultures was inhibited compared with blasts not containing ITD mutations.25 In this study, we show that ITD mutations of Flt3 (Flt3-ITD) have oncogenic potential. ITD mutations confer factor-independent growth and radiation resistance onto 32D cells. Also, they induce colony growth in semisolid media, which is not supported by Flt3 ligand in 32D cells transfected with wild-type Flt3 (Flt3-WT). When injected into syngeneic hosts, Flt3-ITD, but not Flt3-WT, leads to the rapid development of a disease resembling leukemia. Analysis of the signaling properties of Flt3-ITD reveals weak constitutive activation of Ras and Akt with ligand-dependent augmentation and strong constitutive activation of STAT5, a novel feature of Flt3 signaling. STAT5 is not activated by a ligand-bound wild-type receptor. Analyses by dominant negative (dn) constructs show an important role of Ras and STAT5 for ITD-induced colony formation. Taken together, ITD mutations of Flt3 have powerful transforming potential in myeloid cells mediated by Ras and aberrant STAT5 activation. From the Department of Medicine/Hematology and Oncology, and the Department of Neurology, University of Mu\\u00a8nster, Germany and the Department of Hematology and Oncology, Osaka University Medical School, Japan. Submitted December 28, 1999; accepted August 1, 2000. Supported by grant Se 600/2-2 from the Deutsche Forschungsgemeinschaft and Se 119804 and KR 229838 from the IMF Mu\\u00a8nster. This paper represents partial fulfillment of the requirements for the PhD program of R.F. and for the MD program of R.S. and C.S. at the University of Mu\\u00a8nster. M.M. and R.F. contributed equally to this work. Reprints: Hubert Serve or Wolfgang E. Berdel, Department of Medicine/ Hematology and Oncology, Medizinische Universita\\u00a8tsklinik Mu\\u00a8nster, AlbertSchweitzer-Strasse 33, 48129 Mu\\u00a8nster, Germany; e-mail: serve@uni-muenster.de. The publication costs of this article were defrayed in part by page charge payment. Therefore, and solely to indicate this fact, this article is hereby marked \\u2018\\u2018advertisement\\u2019\\u2019 in accordance with 18 U.S.C. section 1734. \\u00a9 2000 by The American Society of Hematology BLOOD, 1 DECEMBER 2000 z VOLUME 96, NUMBER 12 3907 From www.bloodjournal.org by guest on May 28, 2017. For personal use only. Materials and methods Reagents Recombinant human FL and recombinant murine interleukin 3 (IL-3) were purchased from Pepro Tech (Rocky Hill, NJ). Polyclonal rabbit antimouse Flt3 antibodies and the antiphosphoSTAT5 antibody were purchased from Upstate Biotechnology (Lake Placid, NY) and from Santa Cruz Biotechnology (Santa Cruz, CA). Monoclonal rat antimouse Flt3 antibody was obtained from Pharmingen (San Diego, CA). Monoclonal mouse antiphosphotyrosine antibody (PY20) was obtained from Transduction Laboratories (Lexington, KY). AntiphosphoErk2, anti-Erk2, and total anti-STAT5a/b antibodies were obtained from Santa Cruz Biotechnology. AntiphosphoAkt and anti-Akt were purchased from New England Biolabs (Beverly, MA). Anti-ras antibody was obtained from Upstate Biotechnology. The Annexin-V FITC kit was obtained from Immunotech (Marseille, France). Cell lines The IL-3\\u2013dependent murine myeloid cell line 32Dcl3 (kindly provided by Dr Felicitas Rosenthal, Freiburg, Germany) was cultured in RPMI 1640, supplemented with 10% WEHI-conditioned medium as a source of IL-3, 10% fetal calf serum (FCS), and antibiotics at 37\\u00b0C with 5% CO2. Construction of expression plasmids and transfection into 32Dcl3 cells ITD mutations were detected in patients with AML as described previously.24 The membrane proximal region of 4 mutated Flt3 receptor sequences, as well as wild-type human Flt3 derived from the Oci-AML5 cell line, were amplified by reverse transcriptase\\u2013polymerase chain reaction (RT-PCR) using Pfu DNA polymerase. The sense primer was 59-GCA CAT CTT GTG AGA CGA TCC-39 and the antisense primer was 59-CAC CAT AGC AAC AAT ATT CAA AAA TC-39, yielding a product corresponding to nucleotides 1562 to 2092 of the published sequence of human Flt3.3 One of 2 SspI sites of the PCR products, which is unique to the human Flt3 sequence, was removed by site-directed mutagenesis without alteration of the resulting amino acid code. A BsmBI/SspI fragment of these PCR products spanning nt1567 to 2077 was substitutionally introduced into the sequence of murine Flt3 (provided by Dr I. Lemischka, Princeton, NJ). The sequences of all mutants were confirmed by DNA sequencing. These constructs were cloned into an expression vector (pAL) under the control of the 59LTR of the Moloney murine sarcoma virus (MoMSV), derived from pLXSN (Clontech, Palo Alto, CA). Ten micrograms of plasmid DNA of either plasmid and 1 mg pMAM/BSD (Kaken Pharmaceutical, Japan) were cotransfected into 32Dcl3 cells by electroporation. Cells were selected with 15 mg/mL blasticidin (Invitrogen, Groningen, The Netherlands) in IL-3 supplemented culture. Polyclonal cell lines were used for further experiments. The ovine STAT5/Y694F mutant (dnSTAT5)26 was subcloned into an expression vector under the control of the 59LTR of the MoMSV (pOPALI), derived from pOPI3CAT (Stratagene, La Jolla, CA), which contained a neomycin resistance marker. Lac inducible system The 32Dcl3 were first transfected with pLAM-LacR, which was derived from pMAM and expresses the Lac repressor. The 32D/pLAM-LacR were further transfected with pOPRSVI (Stratagene) containing RasS17N (dnRas), the expression of which is suppressed by the Lac repressor in the uninduced state, and induced by the addition of isopropyl thiogalactose (IPTG). Highly inducible dnRas-expressing clones were selected by limiting dilution. One of the clones was cotransfected with the pAL/Flt3 constructs and pPur (Clontech) as a selection marker, and puromycin-selected polyclonal lines were used for further experiments. The expression of dnRas was induced by incubating the cell lines with 0.5 mmol/L IPTG for 24 hours, and the expression was verified by Western blotting with anti-ras Ab (Upstate Biotechnology). Flow cytometry Cells were preincubated for 15 minutes at 4\\u00b0C with mouse IgG and subsequently stained for 15 minutes with the indicated antibody. The cells were washed twice in phosphate-buffered saline (PBS)/0.1% bovine serum albumin (BSA) and analyzed with a FACSCalibur (Becton Dickinson, Heidelberg, Germany). Immunoprecipitation and Western blot analysis The 32Dcl3 cells transfected with Flt3 constructs were starved from IL-3 and serum for 12 hours in 0.5% FCS. Subsequently, cells were resuspended in 1 mL medium for 10 minutes at 37\\u00b0C with or without 100 ng/mL FL, washed once with ice-cold PBS, and lysed with buffer containing 50 mmol/L HEPES (pH 7.4), 10% glycerol, 150 mmol/L NaCl, 1% Triton X-100, 1 mmol/L EDTA, 1 mmol/L EGTA, 50 mmol/L ZnCl, 25 mmol/L NaF, proteinase inhibitors (Complete; Boehringer, Mannheim, Germany), 1 mmol/L pepstatin, and 1 mmol/L sodium orthovanadate. Cell lysates were clarified at 20 000g for 20 minutes. For immunoprecipitation, cell lysates were incubated with rabbit polyclonal antibody to murine Flt3 (Upstate Biotechnology) and with Protein A/G-Plus-Sepharose (Santa Cruz Biotechnology). The immunoprecipitates were washed 3 times with lysis buffer. Immunoprecipitates as well as total lysates were resuspended in SDS sample buffer, heated, and separated by SDS PAGE. Gels were blotted on Immobilon P membrane (Millipore, Bedford, MA) and stained with the indicated antibody. Antibody binding was detected by incubation with an horseradish peroxidase (HRP)-labeled secondary antibody, followed by chemiluminescence detection (ECL-Plus, Amersham Pharmacia Biotech, Uppsala, Sweden). 3H thymidine incorporation A total of 1 3 104 32D cells was IL-3 and serum starved (0.5% FCS) for 12 hours. Subsequently, cells were placed in 200 mL medium supplemented with the indicated concentrations of FL. After a 24-hour incubation at 37\\u00b0C, 0.037 MBq (1 mCi) 3H thymidine was added to each well, and cells were incubated for an additional 8 hours. Cells were harvested on glass fiber filters, and b-emission of the bound DNA was analyzed with a scintillation counter. Experiments were repeated at least 3 times. Each data point represents the mean 6 standard deviation of 4 wells. Radiation-induced apoptosis A total of 2.5 3 105 cells was starved from IL-3 and serum for 3 hours, placed in 24-well plates, and exposed to 5 Gy g-irradiation. Immediately after irradiation, cells were supplemented with FL (100 ng/mL), IL-3 (1 ng/mL), or no factor. Cell viability was analyzed using the Annexin-V assay. Cells staining negative for both Annexin-V and propidium iodide were counted as viable cells. Electrophoretic mobility shift assay The 32D cells exposed to the indicated cytokines were washed with ice-cold PBS containing 0.5 mmol/L PMSF and 0.2 mmol/L NaVO3 and lysed in 500 mL cold buffer I (20 mmol/L HEPES, pH 7.9, 0.2% [vol/vol] Nonidet P-40, 10% [vol/vol] glycerol, 1 mmol/L EDTA, 1 mmol/L EGTA, 1 mmol/L DTT, 0.2 mmol/L NaVO3, 0.5 mmol/L PMSF, 10 mg/mL leupeptin, and 10 mg/mL aprotinin). After incubation on ice for 10 minutes, cell lysates were centrifuged at 14 000 rpm for 10 minutes at 2\\u00b0C. The pellets were resuspended in 100 mL buffer II, containing 20 mmol/L HEPES (pH 7.9), 20% (vol/vol) glycerol, 420 mmol/L NaCl, 2 mmol/L EDTA, 1.5 mmol/L MgCl2, 0.5 mmol/L DTT, 2 mmol/L NaVO3, 0.5 mmol/L PMSF, 10 mg/mL leupeptin, and 10 mg/mL aprotinin, and incubated for 30 minutes at 4\\u00b0C. After centrifugation at 14 000 rpm for 20 minutes at 2\\u00b0C, the supernatant was dialyzed against buffer III (20 mmol/L HEPES [pH 7.9], 20% [vol/vol] glycerol, 100 mmol/L NaCl, 2 mmol/L EDTA, 0.5 mmol/L DTT, 2 mmol/L NaVO3, 0.5 mmol/L PMSF) for 4 hours at 4\\u00b0C. The DNA mobility shift assay was performed by incubating 10 mg nuclear extract with 1 mg poly(dI-dC) and 20 000 cpm of the end-labeled, double-stranded probe in binding buffer (10 mmol/L Tris [pH 7.5], 20% [vol/vol] glycerol, 3908 MIZUKI et al BLOOD, 1 DECEMBER 2000 z VOLUME 96, NUMBER 12 From www.bloodjournal.org by guest on May 28, 2017. For personal use only. 50 mmol/L NaCl, 1 mmol/L DTT, and 1 mmol/L EDTA) in a final volume of 20 mL. The binding reaction was incubated for 20 minutes at 25\\u00b0C. The double-stranded oligonucleotide was 59-AGA TTT CTA GGA ATT CAA TCC-39, harboring the consensus binding site for STAT5.27 DNA-protein complexes were run on a 4% nondenaturing polyacrylamide gel carried out in 0.5x Tris-Borate-EDTA buffer. The gel was dried and exposed to film. DNA supershift assays were performed by incubation of the nuclear proteins with a specific antibody directed against STAT5a/b (Santa Cruz Biotechnology, 2 mg each per assay) for 15 minutes at 25\\u00b0C before the addition of the probe. Clonal growth in methylcellulose To analyze clonal growth, 1 mL of a culture mixture containing Iscove modified Dulbecco medium (IMDM; Life Technologies, Grand Island, NY), 1% methylcellulose, and 10% FCS was plated on a 35-mm culture dish in the presence of FL (20 to 200 ng/mL), IL-3 (1 ng/mL), or no cytokine. Stably transfected 32Dcl3 cells were seeded at a concentration of 3 3 102 or 3 3 103 cells per dish. The assays were plated as quadruplicates, and colonies were counted on day 6. The numbers given show representative results of one of at least 3 independent experiments per construct. To analyze the effect of the expression of dnSTAT5 on the colony growth, 32Dcl3/Flt3-ITD cells were electroporated with either 20 mg pOPALI/ STAT5Y694F or the same amount of pOPALI/CAT. One day after the electroporation, cells were seeded at a concentration of 1 3 105 cells per dish in 1 mL of a culture mix containing IMDM, 1% methylcellulose, 20% FCS, and 0.6 mg/dL G418. The colonies were counted on day 8. In vivo tumorigenesis assay Eight- to 10-week-old female C3H/HeJ mice, which were syngeneic with 32Dcl3 cells, were used for the in vivo tumorigenesis experiments. The 1 3 106 cells were inoculated by intravenous injection. Moribund animals were killed. Bone marrow cells were extracted from the femur and stained with Pappenheim stain. The experimental protocols had been reviewed and approved by the local Animal Experimentation Committee. Results Construction and expression of the chimeric mutant Flt3 receptors We have previously identified Flt3-ITD mutations from patients with AML.24 To examine the role of ITD mutations in leukemogenesis, we constructed mouse Flt3 complementary DNA (cDNA) mutants containing the mutated human Flt3 sequences (Figure 1A). Four different ITD mutations were cloned into the mouse backbone. These mutations were chosen because of their variable involvement of tyrosine residues and differences in the lengths of the duplication. The wild-type human Flt3 cDNA of this region was amplified from OCI-AML5. All constructs were introduced into the IL-3\\u2013dependent myeloid progenitor cell line 32D under the control of the MoMSV 59LTR. After selection of the bulk culture for 14 days in the presence of IL-3, stable expression of murine Flt3 was established as shown in Figure 1B. All Flt3-ITD mutant receptors were expressed at comparable levels with the wild-type human/ murine chimera Flt3 (Flt3-WT) or with wild-type murine Flt3 (data not shown). The parental or mock transfected 32D lines did not express Flt3 (Figure 1B). Factor-independent DNA synthesis of 32D/Flt3-ITD We analyzed the DNA-synthesis of the different mutants by 3H thymidine incorporation. The parental 32Dcl3 line did not respond to FL, whereas the Flt3-WT transfectant (32D/Flt3-WT) grew dependent on either IL-3 or FL. However, all cell lines containing Flt3-ITD (32D/Flt3-ITD) proliferated independently of exogenously added IL-3 or FL (Figure 2). In addition, FL did not further increase the rate of DNA synthesis in 32D/Flt3-ITD. Thymidine incorporation of 32D/Flt3-ITD was much higher compared with Flt3-WT cells stimulated with maximal levels of FL. As shown in Figure 2A, DNA synthesis of ITD-containing 32D cells without exogenously added growth factors approximately equaled their synthesis rate under the influence of maximal IL-3 stimulation (10 ng/mL). Resistance to radiation-induced apoptosis of 32D/Flt3-ITD It had been reported that FL mediates survival of early hematopoietic precursors as well as AML cells.16,28 Therefore, we examined the role of ITD mutants in DNA damage-induced apoptosis. As shown in Figure 3, 32D/Flt3-WT were highly sensitive to g-irradiation, with less than 10% viable cells 24 hours after irradiation. On FL exposure, up to 60% of 32D/Flt3-WT were rescued from apoptosis. Interestingly, cells transfected with Flt3-ITD were entirely resistant to irradiation-induced apoptosis. The resistance was independent of FL presence or absence. Clonal growth of 32D/Flt3-ITD in methylcellulose To further analyze the effect of Flt3-ITD mutants on proliferation and survival, we examined the ability of 32D/Flt3-ITD cells to form colonies. The 32D cells transfected with the empty vector produced colonies in the presence of IL-3, but not in the presence of FL (Figure 4). Surprisingly, Flt3-WT could not support clonal growth of the 32D cells even in the presence of up to 200 ng/mL FL (Figure 4A). To confirm these findings, we repeated the experiment with 10 times the number of cells per dish. In addition, appropriate expression of Flt3 on the surface of the cells was checked Figure 1. Construction of Flt3 internal tandem duplications and their expression in 32D cells. (A) A human Flt3 fragment spanning nt1624 to 2134, containing either wild-type or 4 different ITD mutations, was substituted for the corresponding murine Flt3 sequences. The predicted amino acid sequence from either the wild-type or the 4 different ITDs is shown. Duplicated sequences are underlined. EC indicates extracellular domain; TM, transmembrane domain; JM, juxtamembrane domain; TK, tyrosine kinase domain; KI, kinase insert; hmWT, human/mouse wild-type; ITD, internal tandem duplication. (B) Expression of Flt3 receptors on the surface of 32D cells were analyzed by staining with PE-conjugated monoclonal rat antimouse Flt3 antibody. The shaded areas show staining with anti-Flt3; the blank areas show the isotype control. BLOOD, 1 DECEMBER 2000 z VOLUME 96, NUMBER 12 FLT3 MUTATIONS INDUCE TRANSFORMATION OF 32D CELLS 3909 From www.bloodjournal.org by guest on May 28, 2017. For personal use only. immediately before plating. We still did not detect any colonies, evidencing that Flt3-WT did not allow clonal growth of 32D cells in methylcellulose (Figure 4B). Results obtained with wild-type mouse Flt3 were identical (data not shown). In stark contrast, Flt3-ITD led to colony formation of 32D cells, even in the absence of IL-3 and FL (Figure 4). Taken together, these results indicate that signaling of Flt3-ITD may elicit a signal quality different from the wild-type isoforms of Flt3. Flt3-ITD induce leukemogenesis of 32D cells in vivo Previously, we have shown that a mutation in the catalytic domain of Flt3 leads to factor-independent receptor autophosphorylation, proliferation, and survival of 32D cells.24 However, when we injected cells carrying these mutations into syngeneic mice, they developed obvious disease very late, with a latency period of several months (M.M., R.S., H.S., unpublished data). Therefore, we investigated whether Flt3-ITD mutants enhance leukemogenesis of 32D cells in vivo. The 32D/Flt3-ITD1, 32D/Flt3-ITD2, 32D/Flt3-WT, and 32D/mock cells were intravenously injected into syngeneic mice. In all but one of the mice injected with 32D/Flt3-ITD1 or 32D/Flt3-ITD2, a disease resembling leukemia developed, with massive enlargement of spleen and liver and infiltration of the bone marrow by blastlike cells (Figure 5). Most animals injected with these cells died within 5 weeks after injection (Figure 6). Mice injected with 32D/mock or 32D/Flt3-WT did not develop obvious disease up to 3 months after injection. After this time, 2 of 10 animals injected with 32D/Flt3-WT developed a similar disease, as did the animals injected with the ITD-containing cells, and died. Constitutive activation of the Flt3 receptor and ligand-dependent activation of MAPK and Akt in 32D/Flt3-ITD To analyze the signaling properties of Flt3-ITD in comparison to Flt3-WT, we first studied the autophosphorylation status of the Flt3 mutants in transfected 32D cells. The cells were IL-3 and serum starved for 12 hours and FL stimulated in the absence of serum for 10 minutes. From the cell lysates, Flt3 was immunoprecipitated and immunoblotted with an antiphosphotyrosine moAb or an anti-Flt3 polyclonal antibody. Interestingly, we detected aberrant and constitutive autophosphorylation of the lower band of the mutant Flt3 isoforms, which corresponds to its immature, possibly intracellular form (Figure 7).29 In contrast, the mature form seems to retain its ligand-specific autophosphorylation pattern, at least in some cases. Also, the balance between the immature and the mature forms of the receptor was disturbed, because only weak expression of the mature form of the receptor was seen on Flt3 immunoblots. To examine downstream signals, total cell lysates of all 4 ITD mutant cell lines were subjected to immunoblotting with antibodies specific to the activation status of signal transduction molecules. The loading conditions of the gels were verified by antibodies detecting the molecules independent of their activation state. To detect MAPK activation, we used an anti-pErk antibody, which is specific for Erk1/2 phosphorylated on tyrosine 204. To detect the activation of PI3K-dependent pathways, we analyzed the activation of Akt, a serine/threonine protein kinase downstream of PI3K, which has been implicated in cell survival. The anti-pAkt\\u2013specific antibody detects Akt, which is phosphorylated at serine 473. As shown in Figure 7, Erk1/2 and Akt were activated ligand dependently in 32D/Flt3-ITD cells, with slightly increased basal levels of Figure 3. Ligand-independent resistance against radiation-induced apoptosis. Cells were starved for 3 hours and exposed to 5 Gy g-irradiation. Cells that were Annexin-V and propidium iodide-negative were counted as viable cells. Viability was calculated as a percentage of these cells over the total cell population. FL concentration was 100 ng/mL. Figure 2. Ligand-independent 3H thymidine incorporation by Flt3-ITD mutants. Cells were starved for 12 hours from IL-3 and serum and stimulated with FL (100 ng/mL or indicated concentrations), IL-3 (10 ng/mL), or no cytokine. Data are shown as percentage of thymidine incorporation compared with thymidine incorporation of the respective cell line under IL-3 stimulation. Note the different scale of the y-axis in the 2 diagrams of B. 3910 MIZUKI et al BLOOD, 1 DECEMBER 2000 z VOLUME 96, NUMBER 12 From www.bloodjournal.org by guest on May 28, 2017. For personal use only. phosphorylation. Only one ITD construct, ITD2, led to significant constitutive activation of the Erk proteins. Taken together, the ITD mutations induced factor-independent autophosphorylation of the immature form of Flt3. In the absence of ligand, they did not induce strong activation of Erk proteins or of Akt. Constitutive activation of STAT5 in 32D/Flt3-ITD To analyze STAT5 activation, we first used an antibody specific for STAT5a/b phosphorylated on tyrosine 694. Flt3-WT induced very weak phosphorylation of STAT5 after FL stimulation (Figure 8A) However, the ITD mutants induced strong constitutive phosphorylation of STAT5. To confirm the activation of STAT5 in a different assay system, we performed electrophoretic mobility shift assays using a STAT5-specific probe (Figure 8B). On stimulation with IL-3, both WT and ITD mutants showed similar DNA binding activity of STAT5. Only 32D/Flt3-ITD cells showed constitutive DNA binding activity, and even FL stimulation could not induce STAT5 DNA binding in 32D/Flt3-WT. The DNA binding activity of 32D/Flt3-ITD was confirmed to be STAT5 by supershift analyses with an anti-STAT5 antibody. Effects of dominant negative Ras and STAT5 on ITD-induced colony formation To examine the consequences of STAT5 activation in 32D/Flt3- ITD, we analyzed the effects of dnSTAT5 on the colony formation of 32D/Flt3-ITD. After transfection of STAT5Y694F (dnSTAT5), the colony formation of 32D/Flt3-ITD1 was reduced by approximately 50%, compared with the transfection of a control vector (Figure 9A). As shown in Figure 7, we observed weak MAPK activation by some of the ITD mutations in the absence of exogenously added Flt3 ligand. Also, inactivation of STAT5 did not abrogate the Flt3-ITD\\u2013induced colony growth completely. Therefore, we were interested in examining the relative role of the Ras/MAPK pathway for the function of the ITD mutations. We used the Lac-inducible system, in which the expression of dnRas (Ras S17N) was repressed by the Lac repressor. Repression was relieved by IPTG, leading to highly inducible expression of the dominant negative construct, as shown in Figure 9B. The colony formation induced by the ITD forms of Flt3 was strongly inhibited by dnRas induction, as shown in Figure 9B for ITD1. Similar results were obtained with ITD2 (data not shown), which had caused considerable constitutive activation of MAPK in the pErk blot (Figure 7). IPTG alone did not show any effect on cell lines transfected with the empty control vector. Figure 4. Clonal growth of 32D cells in methylcellulose. (A) Transfected 32Dcl3 cells were plated at a concentration of 300 cells per dish. Colonies were counted on day 6. The assays were plated as quadruplicates. The numbers given show the results of one of at least 3 independent experiments per construct, which all gave similar results. IL-3; 1 ng/mL, FL; 20 ng/mL. (B) Transfected cells were plated at 3000 cells per dish under the indicated growth conditions. Photographs of the dishes were taken on day 6. IL-3, 1 ng/mL; FL, 200 ng/mL. Figure 5. Development of leukemia-like disease in 32D/Flt3-ITD\\u2013injected mice. Female C3H/HeJ mice, aged 8 to 10 weeks, were used for the in vivo tumorigenesis experiments. The 1 3 106 cells were inoculated intravenously. The figure shows an animal at autopsy 28 days after injection of 32D cells containing ITD2. (A) The abdomen of the mouse showing a grossly enlarged liver. (B) The explanted spleen (right) is shown; for size comparison, the spleen of an uninjected animal (left) was photographed simultaneously. (C) Bone marrow smear, Pappenheim stain. Figure 6. Kaplan-Meier plot of the mice injected with 32D/Flt3. The fraction of surviving mice is plotted over the time after injection. All but one mouse injected with ITD1 or ITD2 developed the disease (described in the text and in Figure 5) within 4 to 5 weeks after injection and died. Mice injected with wild-type Flt3 survived without obvious disease until day 77 and day 92, when 2 of 10 mice developed the identical disease and also died. BLOOD, 1 DECEMBER 2000 z VOLUME 96, NUMBER 12 FLT3 MUTATIONS INDUCE TRANSFORMATION OF 32D CELLS 3911 From www.bloodjournal.org by guest on May 28, 2017. For personal use only. Discussion In this report, we demonstrate that internal tandem repeat mutations of Flt3, which are readily detectable in about 20% of cases with AML18,30 and which indicate a poor prognosis of the disease,21-23 elicit transforming ability in the factor-dependent murine myeloid cell line 32D. Our analyses of the autophosphorylation pattern of the Flt3 mutations indicate that ITD mutations cause constitutive autophosphorylation of Flt3. We also demonstrate that the observed activation of Flt3 is sufficient to mediate proliferation and survival in the absence of ligand, even if the mature form of the receptor is only marginally autophosphorylated. Furthermore, the mutant receptors alter cellular behavior in a way not seen with ligandactivated wild-type Flt3. The mechanism of Flt3 activation by mutation in the juxtamembrane domain remains elusive. Constitutive activation of the close relative Kit by point mutations and small deletions in the juxtamembrane region has been studied extensively,31,32 showing that ligandindependent dimerization occurs in the cellular membrane. Mutations in the kinase domain of Kit cause intracellular dimerization and activation of the receptor.33 Recently, it has been reported that in the Kit juxtamembrane domain, methionine 552 to isoleucine 563 form a putative alpha-helical conformation, and that substitution of any of 4 different amino acids sitting along the hydrophobic side of the alpha-helix leads to receptor activation.34 We also were interested in juxtamembrane deletion mutants of Flt3 and constructed a deletion mutant, in which all 4 potential tyrosine autophosphorylation sites were eliminated. This mutant also showed ligand-independent receptor autophosphorylation, proliferation, and apoptosis resistance, suggesting that the conformational change of the juxtamembrane domain, due to either addition or deletion, leads to receptor activation and similar biologic effects (M.M., R.F., H.S., manuscript in preparation). Taken together, ITD mutations in the Flt3 juxtamembrane region seem to disturb the Figure 7. Constitutive tyrosine phosphorylation of Flt3 and ligand-dependent activation of MAP kinase and Akt. After a 12-hour starvation period, 1 3 107 cells were stimulated for 10 minutes in 1 mL medium with or without 100 ng/mL FL for 10 minutes as indicated. Immunoprecipitates with anti-Flt3 were performed where indicated, and the immunoprecipitates or total cell lysates were separated by SDS-PAGE. After blotting, the blots were stained with the indicated activation-specific antibodies. pErk recognizes Erk1 (p44), and Erk2 (p42), phosphorylated on tyrosine 204; pAkt is specific for Akt phosphorylated on serine 473. Subsequently, the blots were stripped and stained with the indicated protein-specific antibodies to demonstrate equal loading: tErk is specific for total Erk2 (p42) and does not recognize Erk1. Figure 8. Constitutive activation of STAT5 by Flt3-ITD. (A) Detection of STAT5 activation by immunoblot. Total cell lysates were separated by SDS-PAGE and immunoblotted with anti-pSTAT5, which reacts with STAT5 phosphorylated on tyrosine 694, or antitotal STAT5 as indicated. (B) Electrophoretic mobility shift assay of STAT5. Nuclear extracts were incubated with a double-stranded 32P-labeled oligonucleotide containing the consensus STAT5 binding sequence. Results from 2 constructs are shown: ITD1 (lanes 1-5) and Flt3-WT (lanes 6-10). The stimulation conditions of the cells before lysis are indicated: no growth factors added (N, lanes 1 and 6), 10 ng/mL IL-3 for 10 minutes (IL-3, lanes 2, 3 and 7, 8), and 100 ng/mL FL (FL, lanes 4, 5 and 9, 10). To verify the presence of STAT5 in the shifted oligonucleotide/ protein complex, a supershift assay was performed with an anti-STAT5a/b antibody as indicated (lanes 3, 5 and 8, 10). Note that no shift or supershift is visible in unstimulated and FL-stimulated wild-type Flt3 cells, whereas shifted complexes containing STAT5 can be seen in ITD1 cells. Figure 9. Effects of dominant negative constructs on colony formation induced by Flt3-ITD. (A) Equal amounts of an empty control vector and a dominant negative STAT5 (Y694F) were electroporated into 32D cells containing ITD1. One day after electroporation, cells were plated at a concentration of 1 3 105 cells per dish in the presence of 0.6 mg/mL G418 and no cytokine. Numbers of colonies were counted at day 8. Data shown are representative of 3 independent experiments. (B) Inducible expression of dn-Ras. ITD1/dnRas were polyclonal cell lines, which were made by transfection of Flt3-ITD1 into the 32D/dnRas clone. Cells were treated with 0.5 mmol/L IPTG for 24 hours, and total cell lysates were subjected to Western blot with anti-Ras Ab. These cells were subjected to a colony assay as described in Figure 4, under the presence or absence of IPTG. 3912 MIZUKI et al BLOOD, 1 DECEMBER 2000 z VOLUME 96, NUMBER 12 From www.bloodjournal.org by guest on May 28, 2017. For personal use only. conformation of a tertiary structure, which normally exerts inhibitory control over unoccupied receptor tyrosine kinases. The ITD mutations significantly shorten the latency period, which is required for Flt3 to transform 32D cells into a fully oncogenic cell line in vivo. The low, but detectable transforming ability of induced wild-type Flt3 expression is in accordance with the ability of c-kit to induce leukemogenesis in 32D cells, with a long latency period.35 The additional effect of the ITD mutations most likely depends on the ligand-independent activation of Flt3. We found that all types of Flt3-ITD do not only confer constitutive activation of the major biologic functions of Flt3, but also induce additional functions in 32D cells. Of particular interest is the ability of ITD mutations to support clonal growth. The surprising difference between activated wild-type Flt3 and the ITD mutations could be attributable to a complementary signal necessary for the Flt3 response in semisolid media, but not in suspension culture. The ITD mutations obviously bypass the necessity for this. They seem to provide an intracellular signal different from activated wild-type receptors. We analyzed intermediates of several signaling pathways and could indeed find qualitative differences in the signals provided by wild-type and ITD receptors. Our results show that Flt3 activatesAkt, which has been shown to be an enzyme regulated by phosphoinositide products of PI3 kinase. The importance of PI3 kinase signaling for Flt3-mediated proliferation and survival is controversial, because the binding of the enzyme to Flt3 has not been demonstrated with certainty. On the other hand, it has been recently shown that Flt3 activates PI3 kinase by alternative mechanisms.36,37 Thus, our results showing the activation of Akt by wild-type Flt3 are in line with the latter experiments. We and others have shown that the Ras/Raf/MAP kinase pathways are involved in proliferation and survival induced by ligand-activated wild-type Flt3.38-40 Although the ITD mutants induced autonomous proliferation and survival of 32D cells, constitutive activation of MAP kinase and Akt was only weak in most of the ITD-containing cell lines. In contrast, STAT5 is strongly and constitutively activated by all ITD mutations tested, whereas STAT5 phosphorylation by wildtype Flt3 is very weak and does not translate into its enhanced DNA binding after Flt3 stimulation. After the initial submission of this paper, the constitutive activation of STAT5 by ITD mutations and the lack of STAT5 activation by wild-type Flt3 has been independently described by others,41 confirming our results of differential signaling between wild-type and ITD-mutated Flt3. Our results provide the first evidence for the relative importance of the different signaling pathways for Flt3/ITD function: both Ras and STAT5 are important for Flt3/ITD-induced colony formation of 32D cells. Transfection of dominant negative STAT5 does not inhibit colony growth completely. One explanation could be that other signaling pathways, possibly other STAT proteins, can substitute for STAT5. The relative importance of Ras-dependent signaling pathways is unexpected, given the low constitutive activation of MAPK by the Flt3 mutations. On the other hand, wild-type Flt3 strongly activates the Ras/MAPK pathway, but does not induce colony formation (Figures 4 and 7). Obviously, basic activity of the Ras/MAPK pathway is a necessary, but not sufficient, condition for 32D cells to grow in colonies. In recent reports, it was shown that STAT proteins are coactivated by Ras-dependent signals: expression of dominant negative Ras in a T-cell line inhibited IL-2\\u2013induced STAT5 activity on a b-casein promotor.42 Also, full STAT3 activation has been shown to be dependent on Rac, which is downstream of Ras.43 Thus, by inhibiting Ras function, we inhibit the activation of STAT5 and other STAT proteins of potential importance for Flt3/ITD-induced colony formation, a possible explanation for its relative importance for transformation of 32D cells. Taken together, our results show that the ITD mutations activate signaling pathways differently from wild-type Flt3, which results in a different phenotype of 32D cells in vitro and in vivo. In conclusion, we provide evidence that frequently occurring Flt3 mutations from patients with AML are leukemogenic in 32D cells. The leukemogenic effect is based on several intracellular signals of these receptor mutants, which involve the basal Ras activation and the constitutive activation of STAT5. Future work will have to elucidate these signaling pathways in still more detail. This will hopefully result in a more sophisticated picture of how proliferation and survival of leukemic blasts are regulated and how they may be subjected to specific therapeutic interventions.  Internal tandem duplications of the FMS-like tyrosine kinase 3 (FLT3) gene are one of the most frequent gene mutations in acute myeloid leukemia (AML) and are associated with poor clinical outcome. The remission rate is high with intensive chemotherapy, but most patients eventually relapse. During the last decade, FLT3 mutations have emerged as an attractive target for a molecularly specific treatment strategy. Targeting FLT3 receptor tyrosine kinases in AML has shown encouraging results in the treatment of FLT3 mutated AML, but in most patients responses are incomplete and not sustained. Newer, more specific compounds seem to have a higher potency and selectivity against FLT3. During therapy with FLT3 tyrosine kinase inhibitors (TKIs) the induction of acquired resistance has emerged as a clinical problem. Therefore, optimization of the targeted therapy and potential treatment options to overcome resistance is currently the focus of clinical research. In this review we discuss the use and limitations of TKIs as a therapeutic strategy for the treatment of FLT3 mutated AML, including mechanisms of resistance to TKIs as well as possible novel strategies to improve FLT3 inhibitor therapy.  Keywords: Acute myeloid leukemia, FLT3 mutations, molecular tailored therapy, tyrosine kinase inhibitors, mechanism of resistance Go to: Introduction Acute myeloid leukemia (AML) is a malignant clonal disorder of the hematopoietic system characterized by an enormous heterogeneity of acquired genetic and epigenetic changes in hematopoietic precursor cells and by impaired mechanisms of self-renewal, proliferation and differentiation. To date, more than 200 structural and numerical cytogenetic abnormalities have been described in this disease, with incidences ranging from <0.1% to 10% [1]. More recently, a variety of gene mutations, deregulated expression of genes, and non-coding RNAs (microRNAs) and epigenetic changes have also been identified in AML over the past few years [2,3].  In a model of the pathogenesis of AML proposed a decade ago, it was postulated that there are at least two classes of mutations that are necessary for the development of AML, with one class affecting the growth factor pathways and the other class resulting in a block in differentiation [4]. More recently, with the advent of whole genome sequencing of AML specimens, a variation on this two-step model has been suggested, in which an initiating driver mutation, such as NPM1, DNMT3A, IDH2 or TET2 , confers a survival advantage to a hematopoietic stem/progenitor cell. This is followed by a cooperating driver mutation, which results in full-blown transformation to AML [5]. This model will undoubtedly evolve, in light of the evidence that AML is polyclonal at presentation, but changes its clonality and mutational profile over time in the setting of chemotherapy and eventual relapse [6]. The most common cooperating mutation in both models is an internal tandem duplication mutation of the FMS-like tyrosine kinase 3 gene (FLT3-ITD mutation). FLT3-ITD mutations constitutively activate the receptor tyrosine kinase (RTK) activity and its downstream signaling pathways, leading to dysregulation of cellular proliferation [7\\u20139].  Since RTK mutations play a crucial role in the pathogenesis of AML and are relatively common, they seem to be an ideal target for a molecularly tailored treatment approach. Targeting FLT3 RTK signaling in particular is very attractive given that FLT3 mutations are found in approximately one-third of patients with AML [10].  In this article, we discuss the use and limitations of tyrosine kinase inhibitors (TKIs) as a therapeutic strategy for the treatment of FLT3 mutated AML. Mechanisms of resistance to TKIs are highlighted as well as possible novel strategies to improve FLT3 inhibitor therapy.  Go to: FLT3 mutated acute myeloid leukemia FLT3, located on chromosome 13q12, is grouped into the class III RTK family, and was first described by Nakao et al. in 1996 [11]. The FLT3 gene plays an important role in growth and differentiation of hematopoietic stem cells [10]. FLT3 mutations are found in about one-third of all patients with AML, and are one of the most frequent genetic abnormalities found in AML [2]. At present, three different activating FLT3 gene mutations are known: FLT3-ITD mutations, which can be detected in approximately 20% [11\\u201313], point mutations in the activation loop of the second tyrosine kinase domain (FLT3 -TKD), detectable in about 6\\u20138% [14,15], and point mutations in the juxtamembrane (JM) as well as extracellular domain of the receptor, which are very rare (approximately 2%) [16]. The most common mutation, FLT3-ITD, leads to a ligand-independent dimerization and constitutive activation of the tyrosine kinase domain, thus prompting factor-independent cell growth [7]. These mutations are associated with constitutive activation of multiple downstream pathways, such as phosphatidyl-inositol 3-kinase/v-akt murine thymoma viral oncogene homolog 1 (PIK3/AKT), mitogen-activated protein kinase/extracellular signal regulated kinase (MAPK/ERK) and signal transducer and activator of transcription 5 (STAT5) [8,9]. Previously, it was thought that FLT3-ITD mutations insert exclusively within the JM domain. Recently, however, we and others have shown that about one-third of FLT3-ITD mutations insert outside the JM domain, in particular in the \\u03b21-sheet of the tyrosine kinase domain 1 (TKD1; Figure 1) [17\\u2013 19]. In cell culture analyses, a prototypic FLT3-ITD with insertion site in the \\u03b22-sheet of the TKD1 ( FLT3-ITD627E) mediated phosphorylation of FLT3 and STAT5, suggesting that non-JM domain FLT3-ITD mutations confer constitutive activation of the receptor [18]. Additionally, FLT3-ITD627E induced transformation of hematopoietic 32D cells and led to a lethal myeloproliferative disease in a syngeneic mouse model [18]. A dramatic up-regulation of the anti-apoptotic myeloid cell leukemia 1 (MCL-1) protein expression could be shown after expression of the FLT3-ITD627E receptor in a cell culture model [20]. Increased binding capacity of the adaptor protein Grb2 to the FLT3-receptor was shown to be involved in MCL-1 up-regulation and was independent from TKI (midostaurin)-induced inhibition of the receptor kinase [20].  Figure 1 Figure 1 Schematic structure of the FLT3 receptor displaying the frequency of FLT3-ITD insertion sites in the juxtamembrane (JM) domain as well as in the tyrosine kinase (TK) domain 1 (according to [17]). Besides the insertion site, FLT3-ITD mutations show a broad variability in the mutant to wild-type (WT) allelic ratio, in the size of the insertion (ranging from three to over 100 base pairs), and the number of affected clones per patient [12, 17, 19, 21\\u201328].  Patients with AML harboring FLT3-ITD mutations have characteristic pretreatment features including increased white blood cell count, a higher percentage of blood and bone marrow blasts, and a more frequent diagnosis of de novo rather than secondary AML [13, 21]. FLT3-ITD mutations are most frequently associated with a normal karyotype as well as with the translocations t(6;9)(p23;q34) [29] and t(15;17)(q22;q12) [30]. In cytogenetically normal AML, FLT3-ITD mutations confer a worse prognosis due to a high relapse rate and adverse overall survival (OS) [13, 21, 22, 27]. Furthermore, previous studies suggested a prognostic role for the mutant to WT allelic ratio, as quantified by Gene-Scan analysis, and the size of the ITD [21\\u201327]. So far, data on the allelic ratio consistently show an association of high allelic burden with worse prognosis, whereas the prognosis of patients with a low FLT3-ITD allelic ratio (allelic ratio either < 0.5 or < 0.78) is discussed controversially [21, 22, 27]. The data published by Gale et al . are in contrast to another study, where no significant difference in outcome was found between lower level mutants and WT FLT3 , although the exact cut-offs for the allelic ratio varied [21,27]. One possible explanation for this finding could be that in these patients the FLT3-ITD mutation occurred in a late stage of leukemogenesis, including a lower addiction to FLT3-ITD signaling. And even the responsiveness to FLT3 TKIs seems to depend on the FLT3 allelic ratio [31]. Patients at diagnosis seem to present more often with lower allelic ratios, which are relatively less addicted to FLT3-ITD signaling for survival as compared to those at the time point of relapse [31]. In an in vitro analysis, relapsed samples and samples with a high mutant allelic ratio were more likely to be responsive to cytotoxicity from FLT3 TKIs as compared to the samples obtained at diagnosis or those with a low mutant allelic ratio [31]. However, the results probably indicate that the presence of a FLT3-ITD with even a low allelic ratio cannot be excluded altogether from prognostic risk stratification. For instance, if the presence of a FLT3-ITD provides resistance to chemotherapy by enhancing DNA repair and salvage of damaged cells [32], then mutant-carrying cells will have a survival advantage on treatment with conventional chemotherapy, irrespective of their relative proportion in the total population.  Besides the allelic ratio, the prognostic impact of the FLT3-ITD size is also a matter of debate. Both longer and shorter FLT3-ITD mutations have been associated with an inferior outcome, or have been shown to have no prognostic impact [23\\u2013 27]. Moreover, we were able to show that a FLT3-ITD insertion site in the \\u03b21-sheet of the TKD1 was associated with an inferior outcome as compared to all other insertion sites for younger adult patients with AML who had been treated with intensive chemotherapy, even after allogeneic hematopoietic stem cell transplant [17, 33]. Recently, these results have been confirmed by Schnittger and colleagues, who could show that FLT3-ITD insertion sites localized more C-terminally within the FLT3 gene were associated with an adverse outcome [19].  Furthermore, the molecular background of cooperating mutations, such as NPM1 and DNMT3A , may influence the prognostic impact of FLT3-ITD positive AML. Yet again, the prognostic impact of a concurrent NPM1 mutation in FLT3-ITD positive AML is discussed controversially. In a publication by Gale and co-workers, a better outcome in FLT3-ITD positive patients harboring a concurrent NPM1 mutation was stated [27], whereas according to other authors the \\u201cprotective effect\\u201c of NPM1 in AML with a higher FLT3-ITD allelic ratio (\\u2265 0.5) seemed at least to be diminished or to get lost [34]. Recently published data suggest that FLT3-ITD retains its negative prognostic impact in intermediate-risk AML, even in the context of other genetic abnormalities, such as DNMT3A and TET2 [35].  For FLT3 -TKD mutations, point mutations, small insertions or deletions can be found in exon 20 of the FLT3 gene, most commonly a substitution of aspartic acid by tyrosine at codon 835, which affect the activation loop of the carboxy terminal part of the TKD [2]. FLT3 -TKD mutations stabilize the activation loop of the open adenosine-5-triphosphate (ATP)-binding configuration, thus leading to constitutive activation of the FLT3 gene. When transduced into murine hematopoietic stem cells, FLT3 -TKD mutations induce an oligoclonal lymphoid disorder, suggesting differences in cell signaling between FLT3 -TKD mutants and FLT3-ITDs [36]. Indeed, strong STAT5 activation could only be demonstrated for FLT3-ITD mutations [36,37]. Currently, the prognostic value of a FLT3 -TKD mutation is still unclear [2,38,39].  Go to: Treatment with FLT3 tyrosine kinase inhibitors Activation of signaling pathways via RTKs plays a central role in the pathogenesis of AML, and inhibition of these tyrosine kinases using small molecules represents an attractive therapeutic concept. One option to interfere with FLT3 activity is to inhibit its kinase activity. TKIs compete with ATP for binding to the active pocket of the kinases, resulting in the inability to autophosphorylate or phosphorylate substrate proteins by transferring the terminal phosphate of ATP. Thus, signal transduction initiated by the mutated RTK is interrupted [40]. Several small molecule kinase inhibitors with activity against FLT3 have been evaluated in patients with AML as single agents and in combination with intensive chemotherapy, e.g. midostaurin (PKC412), lestaurtinib (CEP-701), tandutinib (MLN-518), sunitinib (SU11248) and sorafenib (BAY 43\\u20139006) [41]. These compounds were not specifically developed as FLT3 inhibitors, but exhibit activity against other RTKs, as summarized in Table I. In single-agent phase I and II trials in relapsed or refractory AML, responses were generally limited and not durable. Clinical experience using FLT3 TKIs is outlined below and summarized in Table II.  Table I Table I Overview of preclinical characteristics of FLT3-tyrosine kinase inhibitors used in clinical trials. Table II Table II Overview on reported data from clinical trials investigating the efficacy of FLT3 tyrosine kinase inhibitor treatment. Midostaurin  In a phase II trial, midostaurin has been evaluated as single-agent therapy in 20 patients with relapsed or refractory FLT3-mutated AML, including one patient with advanced myelodysplastic syndrome (MDS), at a dose of 75 mg orally three times daily [42]. Fourteen of the 20 patients (70%) showed a decrease of peripheral blast counts by 50%, and seven of the 20 patients (35%) experienced a greater than 2-log reduction in peripheral blast counts for at least 4 weeks (median response duration, 13 weeks; range, 9 \\u2013 47 weeks). In six of the 20 patients (30%) bone marrow blast counts were reduced by 50%, and two of these achieved a complete clearance of bone marrow blasts [42]. In combination with chemotherapy, midostaurin induced high complete remission (CR) rates in FLT3 mutated AML [43], and based on these favorable results the impact of midostaurin in AML with FLT3-ITD was evaluated in a large international randomized phase III study (Table III). This trial reached its accrual goal of over 700 patients with FLT3 mutated AML in October 2011, and first results are expected in 2014. An important issue relates to the scheduling and duration of treatment with the TKI. Administration of midostaurin simultaneously with chemotherapy appeared to be more toxic than a sequential approach, where midostaurin was given after chemotherapy [43]. Therefore, in the completed randomized phase III trial, midostaurin was administered sequentially following chemotherapy. Currently, the German - Austrian AML Study Group (AMLSG) is evaluating the impact of midostaurin given in combination with intensive induction, consolidation including allogeneic hematopoietic stem cell transplant (HSCT) and single-agent maintenance therapy on event-free survival in a single-arm, multicenter phase II study in adult patients with AML with FLT3-ITD (Table III).  Table III Table III Overview of currently ongoing clinical trials evaluating FLT3-tyrosine kinase inhibitor treatment in acute myeloid leukemia. Lestaurtinib  The TKI lestaurtinib has been evaluated as a single-agent therapy in a phase I \\u2013 II trial in patients with relapsed or refractory FLT3 mutated AML [44]. Fourteen patients with AML were treated with lestaurtinib at an initial dose of 60 mg orally twice daily. Of those, five patients (36%) had clinical evidence of biologic activity and measurable clinical response, including significant reductions of myeloid blast cells in peripheral blood and bone marrow. However, responses were generally short, lasting from 2 weeks to 3 months. In a follow-up phase II study, lestaurtinib was administered as first-line single-agent treatment for older patients with AML not considered fit for intensive chemotherapy, irrespective of FLT3 mutation status. Lestaurtinib was given orally for 8 weeks, initially at a dose of 60 mg twice daily, escalating to 80 mg twice daily, and was generally well tolerated. Clinical activity, manifest as transient reductions of myeloid blast cells in peripheral blood and bone marrow or longer periods of transfusion independence, was seen in three of five FLT3 mutated patients (60%) and five of 22 evaluable FLT3 WT patients (23%) [45]. In a randomized trial of FLT3 mutated AML in first relapse, 224 patients received chemotherapy alone or followed by 80 mg of lestaurtinib twice daily. Twenty-nine patients in the lestaurtinib arm achieved less than 5% myeloid blast cells in bone marrow as compared to 23 patients in the control arm (26% vs. 21%; p = 0.35). In this study, lestaurtinib treatment after chemotherapy did not increase response rates or prolong survival. Correlative studies from this trial, however, indicated that only a limited number of patients achieved FLT3 target inhibition with lestaurtinib [46]. Similar to midostaurin, the scheduling of lestaurtinib administration seems to be important, since lestaurtinib induced cytotoxicity only if it was used simultaneously or immediately following exposure to chemotherapy. This effect appeared to be attributable to the cell cycle arrest induced by lestaurtinib [47].  Sunitinib  Pharmacodynamic and pharmacokinetic effects of the multi-targeted oral TKI sunitinib have been evaluated in a single-dose phase I study in 29 patients with AML. The dose of sunitinib was escalated from 50 to 350 mg, in increments of 50 mg and cohorts of three to six patients. In this clinical trial, only five of the 29 (17%) patients harbored a FLT3 mutation: three patients had a FLT3-ITD and two patients a FLT3 -TKD. Inhibition of FLT3 phosphorylation was dependent on dose and plasma drug levels. The primary end point, strong inhibition of FLT3 phosphorylation in > 50% patients, was reached in 200 mg and higher dose cohorts [48]. Furthermore, 15 patients with refractory AML were treated in a phase I study with sunitinib. Contrary to the encouraging results of midostaurin or sorafenib, single-agent therapy with sunitinib induced only partial responses (PRs) of short duration (4 \\u2013 16 weeks) in this cohort of patients with AML [49]. However, much better results have been reported recently on a phase I/II clinical study, which evaluated the feasibility of standard induction and consolidation therapy in combination with orally administered sunitinib in elderly patients with AML with activating FLT3 mutations. Median age in this trial was 70 years (range, 60 \\u2013 78 years). Interestingly, the CR rate in AML with FLT3-ITD was 53% (8/15) and was 71% (5/7) in those with FLT3 -TKD. All 13 patients who achieved a CR received repetitive cycles of high-dose cytarabine consolidation therapy, and seven proceeded to single-agent sunitinib maintenance therapy (median duration, 11 months; range, 1 \\u2013 24 months). Within this study, median survival was 18.8 months, median relapse-free survival 11 months, and two patients are in sustained CR [50].  Sorafenib  The anti-leukemic activity of sorafenib was investigated in a phase I study in 16 patients with refractory or relapsed AML. Sorafenib was administered orally with dose levels from 200 to 400 mg twice daily. In this study, no dose limiting toxicity was observed. Sorafenib significantly reduced the percentage of myeloid blast cells in peripheral blood and bone marrow in patients with FLT3-ITD AML (n = 6/6), but showed only modest activity in FLT3 WT patients (n = 3/7) and no clinical response in patients with a FLT3 -TKD mutation (n = 0/3) [51]. A recently published study on 65 patients with FLT3-ITD AML treated with sorafenib monotherapy either after chemotherapy or allogeneic HSCT further highlights these encouraging results [52]. In a pharmacodynamic phase I dose escalation study, evaluating sorafenib in 15 patients with relapsed or refractory leukemia, of whom only two harbored a FLT3-ITD mutation, no significant clinical responses were noted [53]. Sorafenib was further evaluated in combination with intensive chemotherapy in a combined phase I/II study [54]. In this study, sorafenib induced high CR rates in younger FLT3 mutated as well as FLT3 WT patients with AML. The survival rate at 1 year was 74% (median follow-up of 54 weeks; range, 8 \\u2013 87 weeks). Accompanying plasma inhibitory assays in 10 patients demonstrated an on-target effect on FLT3 kinase activity due to complete inhibition of phosphorylated FLT3 [54]. So far, there are only limited results for sorafenib treatment in older patients with AML (> 60 years). A completed randomized double-blinded study of 197 older patients with AML could not show a beneficial effect of sorafenib in combination with standard chemotherapy on clinical outcome [55]. In this study, however, there might have been biases, since the study was not selected for the target population and the proportion of FLT3-ITD was very low in the study cohort (28 of 197 patients; 14%). In the context of allogeneic HSCT, sorafenib has shown clinical activity in some cases of FLT3-ITD positive AML [52,56,57]. A case report of a patient with FLT3 -ITD positive AML after allogeneic HSCT illustrates that sorafenib in conjunction with graft-versus-leukemia effects could induce molecular negativity of FLT3-ITD after a second relapse [56]. Encouraging results had also been reported on the monotherapy with sorafenib in 65 patients with FLT3-ITD AML, who had been relapsed or were refractory after chemotherapy or allogeneic HSCT [52]. Within this study, a response could be achieved in 83% of the patients (n = 54/65), including the achievement of a CR (with or without normalization of peripheral blood counts; n = 15). Furthermore, patients after prior allogeneic HSCT had a significantly lower rate of resistance toward sorafenib treatment as compared to those patients after chemotherapy (38% vs. 47%), as well as a longer remission duration (median, 197 days vs. 136 days; p = 0.03), suggesting that sorafenib therapy in the context of allogeneic HSCT may synergize with allogeneic immune effects [52]. However, these encouraging results could not be obtained in 16 patients with FLT3-ITD AML treated at the M. D. Anderson Cancer Center, who were relapsed after allogeneic HSCT [58]. The patients received either sorafenib alone (n = 8) or in combination with chemotherapy (n = 8). Although the number of circulating blasts decreased in 80% of cases, only three patients achieved a partial remission. Two patients were bridged to a second allogeneic HSCT, but both relapsed within 3 months and the median OS was only 83 days [58]. Currently, several studies evaluating the feasibility as well as efficacy of sorafenib maintenance after allogeneic HSCT are under way (Table III).  Quizartinib  While the first-generation TKIs showed only limited activity as a single agent, the second-generation TKI quizartinib (AC220) was effective as monotherapy in early phase I and II studies of patients with relapsed or refractory AML. Initially, quizartinib was identified due to screening of a scaffold-based library of compounds against several kinases [59]. Based on binding affinity, quizartinib was identified as a compound with high potency and selectivity for FLT3 [60]. In a phase I study, quizartinib was administered once daily as an oral solution in an outpatient setting for 14 days every 28 days at a starting dose of 12 mg, with dose escalations of 50% in subsequent cohorts. A later amendment included the ability to administer quizartinib continuously for 28 days without interruption. Observed adverse side effects were mainly gastrointestinal symptoms as well as prolongation of the QTc interval. The phase I study included 76 patients, of whom 18 (24%) were FLT3-ITD positive, 45 (59%) were FLT3 WT and 13 (17%) were indeterminate. Ten of 18 (56%) FLT3-ITD patients as compared to nine of 45 (20%) FLT3 WT patients responded. The median duration of response was 14 weeks. Of note, responses were observed in cohorts treated with doses as low as 18 and 40 mg/day. The FLT3 WT group comprised three patients with a FLT3 -TKD mutation, none of whom responded [60]. Very recently, the phase II study of quizartinib has reached its accrual goal of approximately 300 patients with AML. The trial consisted of two treatment cohorts: the first one comprised older patients (\\u2265 60 years) who had relapsed after one first-line chemotherapy regimen within 1 year or were primary refractory to first-line chemotherapy. The second cohort consisted of patients who were \\u2265 18 years of age and had relapsed or were refractory after one second-line (salvage) regimen or HSCT. Most patients in this study were FLT3-ITD positive, but a small number in each cohort lacked the mutation. The primary objective of the trial was to evaluate the composite CR rate, defined as the combination of CR, CR with incomplete platelet recovery and CR with incomplete hematologic recovery. Within both treatment cohorts, quizartinib as a single agent has shown promising efficacy, leading to a composite CR rate (CRc) of 46 \\u2013 54% in relapsed/refractory FLT3-ITD positive AML [61,62]. Notably, remission rates were highest in patients who had relapsed after allogeneic HSCT, and a high percentage (37%) of cohort II patients could be successfully bridged to allogeneic HSCT [62]. Patients with FLT3 wild-type also responded to quizartinib, but to a lower extent (CRc rates of 31% and 32% for cohort I and cohort II, respectively). Overall, quizartinib was well tolerated, with manageable toxicities; adverse events included mainly gastrointestinal symptoms, reversible QT prolongation and myelosuppression, possibly related to KIT inhibition [61,62]. These data represent the highest level of single-agent activity observed to date for FLT3-targeted therapy in patients with relapsed/ refractory FLT3-ITD positive AML. The encouraging efficacy results as well as an acceptable safety profile in this population with high-risk AML support continued clinical evaluation in mono- and combination therapy as well as for maintenance after allogeneic HSCT (currently ongoing clinical trials are summarized in Table III).  Go to: Allogeneic hematopoietic stem cell transplant In addition to TKIs, allogeneic HSCT seems to be clinically effective in patients with AML with FLT3 -ITD: both relapse-free and OS rates are significantly improved by allogeneic HSCT [33,63\\u2013 68]. Notably, the relapse risk after allogeneic HSCT could be reduced to roughly 30% [64,66,67,69]. The reported data suggest the existence of a strong anti-leukemic effect after myeloablative, but also after reduced-intensity conditioning in combination with a graft-versus-leukemia reaction. In contrast, in a single-center experience, a lower disease-free survival and increased risk of relapse after allogeneic HSCT in FLT3-ITD patients was reported as compared to patients with wild-type FLT3 [70]. However, the data might have been biased due to the low patient number and long median time to transplant, requiring several courses of chemotherapy before allogeneic HSCT. Within this survey of patients after allogeneic HSCT, any source of donor (including umbilical cord blood and haploidentical) had been used [70], which contrasts with other publications where the source of donor was restricted to matched-related or matched-unrelated donors [63,64,66,68,69]. Furthermore, almost half of the patients received reduced-intensity conditioning [70]. Therefore, these data should be interpreted cautiously. An overview of the outcome results for FLT3-ITD patients after allogeneic HSCT is given in Table IV. Yet again, the molecular characteristics of FLT3-ITD may further influence its prognostic impact even in the setting of allogeneic HSCT [71,72]. Recently presented data in FLT3-ITD positive AML suggest that a high allelic ratio and/or a FLT3-ITD insertion site in the \\u03b21-sheet of the FLT3 gene and, in particular, the presence of both seem to be associated with a very unfavorable prognosis, which could not be overcome by an allogeneic HSCT performed in first CR [72]. In contrast, allogeneic HSCT performed in first CR had a beneficial effect on relapse-free and OS in all other patients with FLT3-ITD positive AML [72]. Whether the recent introduction of FLT3 kinase inhibitors will significantly improve outcome after chemotherapy remains to be answered by ongoing clinical trials. TKIs may increase the proportion of patients with FLT3-ITD who achieve CR and reach allografting in the best condition. Therefore, FLT3 inhibitors may be a strategy early on in therapy to improve remission rate and to maintain patients in remission after allogeneic HSCT. Several ongoing clinical trials are now investigating this strategy (Table III).  Table IV Table IV Outcome after hematopoietic stem cell transplant in FLT3-YID AML. Go to: Mechanisms of resistance during treatment with tyrosine kinase inhibitors During therapy with FLT3 TKIs the induction of secondary FLT3 -TKD mutations seems to emerge in a substantial proportion of patients with FLT3-ITD AML, which is associated with loss of response [73,74]. The emergence of a FLT3-TKD during treatment with midostaurin had first been described in an index patient with AML by Heidel and colleagues [75] and has now been shown in several patients with AML who had been treated within the AC220-002 trial [73]. The FLT3 -TKD mutation might possibly lead to a change in the RTK structure, thus hampering the binding of the TKI and therefore leading to resistance similar to that described for resistance to imatinib in chronic myeloid leukemia (CML) [76]. Recently, it has been shown that different FLT3 TKIs, e.g. midostaurin, sorafenib and SU5614, displayed distinct and non-overlapping resistance profiles in FLT3-ITD expressing cell lines [77]. Interestingly, the profile of resistance mutations emerging with SU5614 was limited to exchanges in the second part of the kinase domain, predominantly exchanges of D835. In contrast, midostaurin exclusively produced mutations within the tyrosine kinase domain 1 at codon N676 [77], which is in line with the findings reported by Heidel and colleagues [75]. Similarly, in a cell based model, saturation mutagenesis of FLT3-ITD followed by selection of transfected cells in FLT3 TKIs revealed a growing number of FLT3 -TKD point mutations that are likely to contribute to TKI resistance [78]. Therefore, combinations of FLT3 inhibitors may be useful to prevent the induction of resistance mutations. Very recently, preliminary encouraging results were presented for the TKI crenolanib [79], which showed activity against platelet-derived growth factor receptor (PDGFR) and FLT3 and is currently being tested in clinical trials on solid tumors. Strikingly, crenolanib displayed cytotoxic activity against primary AML cells from patients who had developed D835 resistance mutations during FLT3 TKI treatment. In vitro , the 50% inhibitory concentration (IC50 ) of crenolanib for inhibition of FLT3-ITD in plasma was 34 nM, and from pharmacokinetic studies of crenolanib in patients with solid tumors, steady state trough plasma levels of roughly 500 nM were found to be safe and tolerable [79]. A phase II study of crenolanib in relapsed or refractory patients with AML with FLT3 -TKD mutations at codon D835 started recruitment quite recently (ClinicalTrials.gov Identifier: NCT01522469). Similarly, the multi-targeted kinase inhibitor ponatinib (AP24534), primarily created to inhibit BCR \\u2013 ABL including the gatekeeper resistance mutation T315I [80], but also certain other tyrosine kinases including FLT3, showed promising activity not only in FLT3-ITD cell lines but also in transfected subclones harboring additional FLT3 -TKD point mutations (N676D, F691I or G697R) [81,82]. Additionally, there seems to be evidence of clinical activity in patients with FLT3-ITD positive AML: in a first phase I study on patients with relapsed or refractory AML the response rate was 43% (n = 3/7). All responses observed were in FLT3-ITD positive patients who were naive to FLT3 inhibitors [83]. Further evaluation of the activity of ponatinib in patients with AML with FLT3-ITD and/or acquired TKD resistance mutations may be warranted.  Besides ponatinib, the oral multi-targeted TKI PLX3397 showed potential activity against FLT3 , including the FLT3-TKD point mutation F691; however, it conferred cross-resistance to other quizartinib-resistant FLT3 -TKD point mutations, such as D835V/Y as well as Y842C/H [84]. Evaluation of the safety and efficacy of PLX3397 in a phase I/II study in adult patients with relapsed or refractory AML is currently ongoing (ClinicalTrials.gov Identifier: NCT01349049).  Another mechanism of resistance might be related to the FLT3-ITD insertion site, since an insertion site outside the JM domain of the FLT3 receptor was shown to be associated with rewired signaling, resulting in an up-regulation of the anti-apoptotic protein MCL-1 [20]. This resulted in a significant reduction of apoptosis as well as resistance toward treatment with midostaurin. Decreasing MCL-1 expression levels by small interfering RNA knockdown resulted in rescue of TKI-sensitivity, indicating the pivotal role of MCL-1 in mediating resistance [20]. Clinically, a FLT3-ITD insertion site within the \\u03b21-sheet of the TKD1 was associated with an inferior outcome as compared to all other insertion sites, even after allogeneic HSCT [17,33]. Moreover, overexpression of survivin with concomitant activation of enhanced signaling pathways has been found to mediate resistance to FLT3 inhibition as well [85]. Enhanced signaling seems to be one mechanism to escape from inhibition of TKI treatment, since FLT3 TKI-resistant cell lines and primary samples showed continued activation of downstream PI3K/AKT and/or RAS/MEK/ MAPK signaling pathways as well as continued expression of genes involved in FLT3-mediated cellular transformation [86]. Inhibition of these downstream signaling pathways by PI3K or MEK inhibitors restored partial sensitivity to FLT3 TKIs [86]. Therefore, an approach combining FLT3 TKIs with anti-FLT3 antibodies and/or inhibitors of important pathways downstream of FLT3 may reduce the chances of developing resistance.  Stromal cells of the bone marrow environment may also contribute to resistance against TKIs due to a possible cyto-protective effect [87,88]. In a cell culture experiment, FLT3 inhibition did not induce apoptosis in FLT3-ITD AML blast cells after being cultured for several days on stroma [88]. Therefore, the bone marrow stroma may provide survival signals to FLT3-ITD AML blast cells, thus enabling survival despite the presence of potent FLT3 inhibition. ERK signaling was strongly activated in primary FLT3-ITD AML blast cells when the cells were co-cultured with stroma, and this activation was unaffected by FLT3 ligand or by FLT3 inhibition [88]. These observations may account for the difference in response to FLT3 inhibition observed between peripheral blood and bone marrow blast cells. Recent data suggest that a combination of FLT3 inhibitors and Janus kinase (JAK) inhibitors might be effective in overriding stromal protection and potentiating FLT3 inhibition [89].  One further clinically relevant aspect may simply be related to the timing of TKI therapy in relation to conventional chemotherapy regimens. The sequence of combination of TKI with chemotherapy may be important, since in vitro studies indicate that the use of TKI prior to chemotherapy results in antagonism, whereas the use of TKI adjacent to chemotherapy results in synergistic effects [47]. Furthermore, it has been shown that FLT3 ligand (FL) levels increase dramatically following intensive chemotherapy [90]. On the one hand, such increased FL levels appear to shift the cytotoxicity IC 50 of FLT3 inhibitors upward by 2 \\u2013 4-fold; on the other hand, FL up-regulation immediately after chemotherapy may be an important driver of residual FLT3-mutant cells, particularly if the inhibitor is only given intermittently. Clinical responses to FLT3 TKIs appear to be dependent upon sustained inhibition, as determined by the plasma inhibitory activity for FLT3, but this is a difficult obstacle to overcome when the agents are administered immediately following chemotherapy, when FL levels are highest \\u2013 and yet it may be the most important time period to try [91].  Go to: Conclusions Our progress in unraveling the heterogeneity of AML has allowed us to identify a number of potential molecular targets such as FLT3 , and has provided insight into the basis for the enormous diversity in the response to treatment. Complete and sustained inhibition of FLT3 -mutated AML will probably require a combination of agents, both targeted and conventional chemotherapy. The ongoing investigations into the mechanisms of resistance to FLT3 inhibition will enable us to gain a deeper insight into the pathobiology of FLT3-ITD mutated AML and will probably continue to lead to the development of even more novel target-specific agents .\",\n          \"Bruton\\u2019s tyrosine kinase (BTK), a member of the TEC family of kinases, plays a crucial role in B-cell maturation and mast cell activation. Although the structures of the unphosphorylated mouse BTK kinase domain and the unphosphorylated and phosphorylated kinase domains of human ITK are known, understanding the kinase selectivity profiles of BTK inhibitors has been hampered by the lack of availability of a high resolution, ligand-bound BTK structure. Here, we report the crystal structures of the human BTK kinase domain bound to either Dasatinib (BMS-354825) at 1.9 A\\u02da resolution or to 4-amino-5-(4-phenoxyphenyl)-7H-pyrrolospyrimidin- 7-yl-cyclopentane at 1.6 A\\u02da resolution. This data provides information relevant to the development of small molecule inhibitors targeting BTK and the TEC family of nonreceptor tyrosine kinases. Analysis of the structural differences between the TEC and Src families of kinases near the Trp-Glu-Ile motif in the N-terminal region of the kinase domain suggests a mechanism of regulation of the TEC family members. Keywords: Bruton\\u2019s tyrosine kinase; BTK; Dasatinib; Celera compound; TEC-family; crystal structure Introduction Bruton\\u2019s tyrosine kinase (BTK) is a member of the TEC kinase family, nonreceptor tyrosine kinases that play important roles in T-cell receptor-, B-cell receptor-, and Fcc-receptor-mediated signaling. BTK participates in signal transduction from B-cell antigen receptors resulting in phospholipase C-c2-mediated calcium mobilization1 which, in turn, affects pre-B-cell functional maturation and expansion. Since BTK is required for B-cell function, it is an important target for the potential treatment of inflammatory diseases that involve B-cell activation. Mutations in the human BTK gene are responsible for Abbreviations: BTK, Bruton\\u2019s tyrosine kinase; BTK-KD, BTK kinase domain; ITK, interleukin-2 inducible T-cell kinase; PH, pleckstrin homology; SH, Src homology; TEC, tyrosine kinase expressed in hepatocellular carcinoma; XLA, X-linked agammaglobulinemia. Disclosure: All authors (with the exception of M. J. Romanowski) are employees of Biogen Idec and own company stock. Michael J. Romanowski\\u2019s current address is Novartis Institutes for Biomedical Research, Cambridge, MA 02139. *Correspondence to: Laura F. Silvian, Biogen Idec, Inc., 12 Cambridge Center, Cambridge, MA 02142. E-mail: laura.silvian@biogenidec.com Published by Wiley-Blackwell. VC 2010 The Protein Society PROTEIN SCIENCE 2010 VOL 19:429\\u2014439 429 X-linked agammaglobulinemia (XLA), a male immunodeficiency that results in a deficit of mature B cells and serum immunoglobulin.2,3 Several compounds that inhibit BTK kinase activity in biochemical assays have been described in the literature and differ in their kinase selectivity profiles. One weak compound, LFM-A13 (a-cyano-bhydroxy-b-methyl-N-(2,5-dibromophenyl)-propenamide) is a BTK inhibitor with an IC50 of 2.5 lM in a biochemical assay, but also inhibits PLK3 and JAK2.4\\u20136 However, it was found to be somewhat specific for BTK, exhibiting 100-fold higher IC50 values for related tyrosine kinases such as JAK1, HCK, EGFR, and insulin-receptor kinase (IRK).7 Another compound, Dasatinib ([N-(2-chloro-6-methylphenyl)- 2-(6-(4-(2-hydroxyethyl)piperazin-1-yl)-2-methylpyrimidin-4-ylamino)thiazole-5-carboxamide] or BMS- 354825) [Fig. 1(A)], originally used to target BCRAbl, has been shown to bind to BTK with an IC50 of 5 nM8 but also binds to other kinases such as SRC family members (HCK, SRC, and CSK), and ephrin receptors, FGR, PDGFRa, and YES.9 BTK was identified as a target of Dasatinib through pull-down experiments in the CML cell line K562.8 The reversible Celera compound, 3-cyclopentyl-1-(4-phenoxyphenyl)-1H-pyrazolo pyrimidin-7-amine,10 was recently described by Pan et al. 11 as a potent inhibitor of unphosphorylated BTK (8.2 nM IC50 in a biochemical assay). However, it also inhibits Lck and Src with IC50 values of 2 and 70 nM, respectively.10 It is chemically similar to the commercially available 4-amino-5-(4- phenoxyphenyl)-7H-pyrrolo[2,3-d]pyrimidin-7-yl-cyclopentane [B43; Fig. 2(A)] described as a potent inhibitor of Lck.12 Finally, an irreversible inhibitor from Pharmacyclics11 is currently in Phase I for B-cell lymphomas. It is expected to bind irreversibly to Cys481 in the BTK kinase domain active site (0.72 nM IC50 in a biochemical assay) and its selectivity profile is better than the reversible binder because it exhibits greater selectivity against Lck, which lacks this cysteine (>1000-fold selectivity in a biochemical assay). Future design of potent, specific BTK inhibitors would be facilitated by the structures of these compounds bound to BTK, to discern whether there are regions surrounding the ligand that are unique to this kinase. BTK is composed of several domains: an N-terminal pleckstrin homology (PH) domain, a prolinerich TEC homology domain, two SRC homology domains (SH3 followed by SH2), and a C-terminal kinase domain (BTK-KD). Mutations in all domains of human BTK have been found to lead to XLA and missense mutations have been found in all domains except for the SH3 domain.13 Structures have been Figure 1. BTK-KD Y551E/Dasatinib crystal structure. A: Chemical structure of Dasatinib. B: Electron density (2Fo-Fc map at 1 sigma) for Dasatinib within a surface representation of the BTK protein in the human BTK-KD-Y551E/Dasatinib complex. C: Overall view of the BTK kinase domain bound to Dasatinib. Inhibitor (magenta), amino-terminal lobe (green), carboxy-terminal lobe (gray), C-helix (red), hinge (blue), and activation loop (yellow). D: Close-up of the active site and residues within 5 A\\u02da of the bound Dasatinib. The surface is colored according to the convention of base (blue) and acid (red). 430 PROTEINSCIENCE.ORG Crystal Structures of Human BTK Kinase Domain solved for the kinase domains of apo-murine BTK7 and human ITK,14 but a high-resolution structure of a full-length protein with regulatory domains is not available. Low-resolution structures of BTK solved by small angle X-ray scattering have revealed an extended, linear arrangement of the SH3, SH2, and kinase domains, which contrasts with structures of autoinhibited full-length Src and Abl kinases in which a more compact arrangement of the SH2 and SH3 domains allows for the SH2 domain to bind near the C-terminal tail of the kinase domain.15 Structural studies of the Src family of tyrosine kinases have revealed that these proteins can adapt two conformations: an autoinhibitory state of the protein, referred to as an \\u2018\\u2018assembled regulatory domain\\u2019\\u2019 conformation, and an active, more open, structure, where the SH2 domain does not interact with the unphosphorylated C-terminal tail.16 Here, we describe the 1.94 A\\u02da resolution crystal structure of the human BTK-KD Y551E mutant bound to Dasatinib and a 1.6 A\\u02da resolution crystal structure of the unphosphorylated human BTK-KD bound to B43. We observe that the two structures differ in the orientation of the C-helix, similar to conformational changes observed in Src kinase family members that are locked into active or inactive states. Both BTK-KD structures reveal ordered density for the WEX motif at the N-terminus of the kinase domain, where X is a hydrophobic residue. The location of the tryptophan side chain at the base of the C-helix provides an explanation for how the WEX motif acts as an important regulatory element for the TEC family of kinases, similar to its role in regulation of the Src family of kinases, and suggests that the two families have a similar mechanism of regulation. Results Protein purification and characterization BTK-KD and BTK-KD Y551E were purified to  95% purity using a simple, three-step process utilizing two successive glutathione-Sepharose chromatography steps followed by size exclusion chromatography. Mass spectrometry indicated that the majority of the wild-type BTK-KD and BTK-KD Y551E was intact and unphosphorylated (wild-type calculated mass \\u00bc 32,641.6 Da, measured mass \\u00bc 32,643 Da; mutant calculated mass \\u00bc 32,607.5 Da, measured mass \\u00bc 32,609 Da), although 2 and 8%, respectively, were missing the first 4 N-terminal residues (wildtype calculated mass \\u00bc 32,317.2 Da, measured mass Figure 2. BTK-KD/B43 crystal structure. A: Chemical structure of B43. B: Electron density (2Fo-Fc map at 1 sigma) for B43 within a surface representation of the BTK protein in the human BTK-KD-B43 complex. C: Overall view of the BTK kinase domain bound to B43. Inhibitor (green), amino-terminal lobe (green), carboxy-terminal lobe (gray), C-helix (red), hinge (blue), and activation loop (yellow). D: Close-up of the active site and residues within 5 A\\u02da of the bound B43. The surface is colored according to the convention of base (blue) and acid (red). The active site cysteine is shown with a yellow surface. Marcotte et al. PROTEIN SCIENCE VOL 19:429\\u2014439 431 \\u00bc 32,319 Da and mutant calculated mass \\u00bc 32,283.2 Da, measured mass \\u00bc 32,285 Da). Crystal structures The crystal structures of BTK-KD Y551E/Dasatinib and BTK-KD/B43 complexes were determined using 1.94 A\\u02da diffraction data (Rfree \\u00bc 25.8%) and 1.6 A\\u02da diffraction data (Rfree \\u00bc 23.1%), respectively (Table I). The electron density maps clearly revealed the positions of the ligands. Figure 1(B) depicts the electron density for Dasatinib and Figure 2(B) reveals the electron density for B43. The human BTK-KD-Y551E/Dasatinib complex structure exhibits a \\u2018\\u2018C-helix in,\\u2019\\u2019 active conformation in which the catalytic lysine forms H-bonds to Glu445 of the C-helix [Fig. 1(C,D)]. The bilobal fold of the BTK KD is similar to that reported for other tyrosine kinase structures: the N-terminal lobe (residues 392\\u2013473) consists of five antiparallel b-sheets and two a-helices and contains the glycine-rich loop that covers the active site. The C-terminal lobe (residues 482\\u2013659) consists of nine a-helices and two b-strands and the activation loop. Finally, the hinge region (residues 474\\u2013481) connects the two lobes of the kinase and contains the active site cysteine (Cys481). The ordered region of the Dasatinib cocrystal structure encompasses residues 392\\u2013658 but electron density for parts of the activation loop (residues 542\\u2013558) and residues 435\\u2013441 is not visible. However, the DFG-portion of the activation loop is visible and is in the so-called \\u2018\\u2018DFG-in\\u2019\\u2019 conformation. The unphosphorylated BTK-KD/B43 cocrystal structure, in contrast, displays a \\u2018\\u2018C-helix-out\\u2019\\u2019 inactive conformation [Figs. 2(C) and 3(B)]. Most of the kinase is ordered except the tip of the glycine rich loop (residues 410\\u2013414). The activation loop displays a \\u2018\\u2018DFG-in\\u2019\\u2019 conformation and is completely ordered. It is composed of two alpha helices, in which Arg544 within the first helix of the activation loop forms a salt bridge to Glu445 of the C-helix. Dasatinib interactions Dasatinib makes several hydrogen bonds to the hinge and occupies a hydrophobic pocket behind the Thr474 gatekeeper residue [Fig. 1(B,D)]; similar to the previously reported structures of Dasatinibbound Abl (PDB ID: 2GQG),17 Lyn (PDB ID: 2ZVA),18 and cSrc (PDB ID: 3G5D).19 Its H-bond interactions to the hinge are described in Figure 1(D), including interactions with the backbone carbonyl and amide of 477, water-mediated interactions with the Tyr476 side chain, and an interaction between the Thr474 side chain and the compound amide nitrogen. The glycine-rich loop of BTK-KD curls toward Dasatinib to form a hydrophobic interaction with Phe413 [Fig. 1(D)]. The ortho-chloro, methyl phenyl substituent of Dasatinib is twisted to enter a hydrophobic pocket, composed of both hydrophobic and hydrophilic residues Met449, Val458, Leu460, Ile472, Lys430, Glu445, and Ser538. Finally, Glu445 of the C-helix forms a salt bridge with the catalytic Lys430; the epsilon amino group of this lysine is poised to make a pi-cation interaction with the Dasatinib ortho-chloro, methyl phenyl ring [Fig. 1(D)]. B43 interactions The B43 compound makes several hydrogen bonds to the hinge and occupies a hydrophobic pocket behind the Thr474 gatekeeper residue [Fig. 2(D)]. The 4-amino pyrrolopyrimidine of B43 occupies the position of the adenine ring of ATP and the cyclopentyl ring occupies the space generally occupied by the ATP ribose in typical protein kinase structures. The 4-amino pyrrolopyrimidine makes many interactions with the hinge; the exocyclic amine directly interacts Table I. Data Collection and Refinement Statistics BTK-KD/B43 complex BTK-KD Y551E/Dasatinib complex Data collection Space group P21212 P21212 Cell dimensions a (A\\u02da ) 72.5 73.4 b (A\\u02da ) 104.3 104.6 c (A\\u02da ) 38.0 38.1 Wavelength (A\\u02da ) 1.54 0.98 Resolution (A\\u02da ) 50\\u20131.6 50\\u20131.94 Rsym (%)a 8.0 (36.3)a 8.3 (61.9) I/r(I) a 18.5 (4.6)a 32.1 (3.3) Multiplicity 3.8 (4.1)a 6.8 (6.6) Total no. reflections/no. unique reflections 405,732/39,020 490,862/22,496 Completeness (%)a 98.1 (98.5)a 99.2 (99.2) Refinement Resolution (A\\u02da ) 24.5\\u20131.6 28.4\\u20131.94 No. reflections 36,293 21,141 Rwork (Rfree) b (%) 19.2 (23.1) 19.9 (25.8) No. molecules per asymmetric unit 1 1 No. atoms Protein 2161 2016 Ligand 28 33 Water 495 164 Average B-factors (A\\u02da c ) Protein atoms 25.8 23.4 Ligand 30.5 41.7 Water 53.5 35.7 R.m.s.d Bond lengths (A\\u02da ) 0.007 0.024 Bond angles () 1.14 2.05 Ramachandron plot % Allowed 98.5 98.2 % Generously allowed 1.5 1.4 % Not allowed 0 0.4 Rsym \\u00bc R|Ihkl  <Ihkl>|/RIhkl. a The value in parentheses is for the highest resolution bin (approximate interval, 0.1 A\\u02da ). b Rwork \\u00bc RIhkl| |Fo  |Fc| |/RIhkl|Fo| for all data except 5%, which is used for the Rfree calculation. 432 PROTEINSCIENCE.ORG Crystal Structures of Human BTK Kinase Domain with the gatekeeper Thr474 hydroxyl and the backbone carbonyl of 475, the N-3 of the pyrimidine accepts a hydrogen bond from the backbone amine of Met477, and the N-1 of the pyrimidine forms a water-mediated hydrogen bond network to the hydroxyl of Tyr476 and the backbone carbonyl of Ala 478 (Fig. 2). The distal phenyl group of the phenoxyphenyl is twisted 38 out of plane of the phenylether, such that it enters a hydrophobic pocket composed of only hydrophobic residues, Phe442, Met449, Leu460, Ile472, Phe540, and Leu542, and is in proximity to Asp539 of the \\u2018\\u2018DFG\\u2019\\u2019 motif. Phe540 of this motif forms a face-to-edge pi-stacking interaction with the phenoxyphenyl group of B43. Asp539 of the DFG-motif forms a salt bridge with the catalytic Lys430 but does not form direct hydrogen bond interactions with the compound. Discussion Relevance of structures for drug discovery The structures of the human BTK-KD Y551E/Dasatinib and BTK-KD/B43 complexes we report here differ from the publicly available structure of apo-murine BTK-KD (PDB ID: 1K2P) and are arguably more relevant for drug discovery for diseases in which inhibition of BTK may be desired. When the apo-mouse BTK structure is superimposed on the human BTK-KD/B43 structure (RMSD of 1.19 A\\u02da over 230 aligned a-carbons), the biggest differences are observed in the activation loop and in the glycine-rich loop. The activation loop of the mouse apoBTK-KD structure adapts an extended configuration with Tyr551 pointed toward solvent. In the mouse apo-BTK structure, the glycine loop also caves into the active site and occludes the ATP binding pocket. Because the mouse and human BTK-KD\\u2019s are 98.3% identical, and only four amino acids are replaced in the mouse sequence (Q379K, K433R, K625R, and T653S), it is likely that the kinase domain flexibility observed in the apo-murine BTK-KD structure is due to a lack of occupancy of a compound in the active site, rather than due to an intrinsic structural difference between the mouse and human species. Selectivity of compound inhibition For both Dasatinib and the reversible Celera compound, the size and hydrogen bonding nature of the gatekeeper residue of a given kinase generally correlates with its degree of biochemical inhibition9 (Table II). Most of the kinases that are inhibited by 10 lM Dasatinib with a Kd < 1 nM, or that are inhibited by 10 lM Celera compound with less than 5% residual activity, have a threonine gatekeeper. A valine residue in this gatekeeper position is tolerated for the Celera compound binding, but is not as well tolerated for Dasatinib binding to the Ret and KDR kinases (Table II). Because the threonine gatekeeper forms H-bond interactions with both compounds, it is possible that the H-bonding binding energy plays a greater role in binding Dasatinib compared to the Celera compound. An alternative explanation for the poor binding of Dasatinib to valine gatekeeper containing kinases KDR and Ret is that there are differences in side chains within 5 A\\u02da of the compound (Table III). In particular, one residue in the back pocket that forms close hydrophobic interactions with Dasatinib in BTK is Met449, which is replaced by a leucine in KDR and Ret. Because the back pocket in the Dasatinib cocrystal structure Figure 3. Differences between the \\u2018\\u2018inactive\\u2019\\u2019 B43 and \\u2018\\u2018active\\u2019\\u2019 Dasatinib-bound conformation of BTK. A: Overlay of BTK-KD bound to B43 (green), ITK bound to staurosporin (magenta, PDB ID: 1SNU) and P38a (yellow, PDB ID: 1P38). This indicates that the larger phenylalanine gatekeeper of ITK compared to the Thr474 gatekeeper of BTK prevents B43 from binding. It also indicates that the shorter hinge of P38a blocks the B43 binding site. B: Overlay of B43 structure (green) and Dasatinib structure (magenta) indicates that they superimpose well near the hinge but differ near the C-helix, resulting in a shift of >10 A\\u02da in the conformation of the Glu445 side chain and the residues that surround the hydrophobic back pocket. Marcotte et al. PROTEIN SCIENCE VOL 19:429\\u2014439 433 is composed of mixed hydrophobic and hydrophilic residues, Dasatinib may have a greater reliance on Met449 compared to B43, whose back pocket is completely surrounded by hydrophobic residues. Either explanation (i.e., different energy considerations for H-bonding to a threonine gatekeeper for different compounds or differences in the makeup of side chains that line Dasatinib\\u2019s unique hydrophobic pocket), could explain why Dasatinib does not bind as well to Ret and KDR. The exception to the rule of requiring a small gatekeeper for compound binding is p38a, EGFR, and NIMA (never in mitosis gene A)-related kinase 11 (NEK11) kinases, which have threonine gatekeepers, but are only moderately inhibited by both small molecules. P38a kinase has a shorter hinge, and thus its reduced affinity can be ascribed to a smaller binding site [Fig. 3(A)]. Similarly, there are differences in the other residues within 5 A\\u02da of the two small molecules (Table III), which could account for the differences in affinity for NEK11 and EGFR. Both differ in the residues that make up the hydrophobic pocket behind the threonine gatekeeper, with EGFR substituting a cysteine for Val458 and NEK11 substituting a larger phenylalanine for Leu460 and a leucine for Met449. Furthermore, NEK11 contains a glycine substitution for Ser538, a residue which is likely to enhance the flexibility of the DFG motif which follows (Table III). In contrast, Abl, Lyn, and Src, kinases which are inhibited well by both small molecules, show no significant variation in the residues that make up the hydrophobic pocket. In these kinases Ser538 is replaced with alanine, a residue with similar conformational flexibility, and is not likely to effect the flexibility of the activation loop (Tables II and III). As the size of the gatekeeper residue increases from threonine to phenylalanine [as in ITK, Fig. 3(A)], methionine (as in Igf1R, JAK1, JNK3, and SYK) (Table II), or leucine (as in Aurora A) (Table II) neither inhibitor reduces kinase activity to the full extent because both binding modes require insertion of the compound past the bulky gatekeeper into the hydrophobic back pocket. Table II. Gatekeeper Residue in Various Kinases, % Activity for Celera Compound and Kd Values for Dasatinib Kinase Gatekeeper % Activity in the presence of 10 lM Celeraa Dasatinib Kd (nM) b Ack1 Ser 3 6 Abl Thr 3 0.5 Bmx Thr 1 0.0 BTK Thr 1 1.0 CSK Thr 0 0.2 EGFR Thr 20 100 FGR Thr 3 0.5 FYN Thr 3 0.7 Hck Thr 5 0.3 Kit Thr 4 0.6 Lck Thr 13 0.2 Lyn Thr 4 0.6 P38a Thr 57 30 Src Thr 1 0.2 Yes Thr 1 0.3 SIK Thr 17 ND RIPK2 Thr 4 30 cRAF Thr 11 600 EphB1 Thr 65 0.4 NEK11 Thr 64 7000 RET Val 0 200 KDR/VEGFR2 Val 6 3000 RSK1 Leu 80 ND RSK2 Leu 83 ND AurA Leu 101 2000 IGF1R Met 87 ND JAK2 Met 88 ND JNK3 Met 76 ND Syk Met 50 3000 Itk Phe 18 ND ND, not determined. a Upstate kinase profiler. b Taken from Ref. 9. Table III. Residues Within 5 A\\u02da of Compounds in the BTK-KD/B43 and BTK-KD Y551E/Dasatinib Structures NTD GRL UPH LWH HYP AVL 428b 430c 408b 409d 413d 416b 472d 474a,c 475b 476b 477b 479d 480b 481c 458c 460b 449b 538b 539b 540b 542b BTK A K L G F V I T E Y M N G C V L M S D F L NEK11 Ve K L G F V I T E YCe Ge Re De V Fe Le Ge DFVe EGRF A K L G F V Le T Qe Le M Fe G CCe LMTe DFL ABL A K L G Ye V I T E FMYe G Ne V L MAe DFL LYN A K L G F V I T E Y M Ke G Se V L MAe DFL SRC A K L G F V I T E Y M Ke G Se V L MAe DFL RET A K L G F V Le Ve E YAe Ye G Se I e L Le SDFL KDR A K L G F V Ve Ve E Fe Ce Fe G Ne VLLe SDFL NTD, N-terminal domain; GRL, glycine rich loop; UPH, upper hinge; LWH, lower hinge; HYP, hydrophobic pocket; AVL, activation loop. a Gatekeeper residue. b Residue interacts with both B43 and Dasatinib in BTK. c Residue only interacts with Dasatinib in BTK. d Residue only interacts with B43 compound in BTK. e Differences in sequence from BTK sequence. 434 PROTEINSCIENCE.ORG Crystal Structures of Human BTK Kinase Domain Overall conformational changes In the BTK-KD Y551E/Dasatinib complex structure, the Gly-rich loop of BTK partially shields Dasatinib from solvent, as found in the Abl kinase structure (data not shown). Because of the curled-in glycinerich loop in the BTK structure, a hydrophobic interaction is formed between Phe413 and the exocyclic methyl of the Dasatinib pyridine ring [Fig. 1(D)], and between Gln412 and Asp539 of the \\u2018\\u2018DFG\\u2019\\u2019 motif. In every other respect, however, the overall conformation of the BTK-KD/Dasatinib structure is similar to the known Dasatinib-bound conformations of Lyn,18 cSrc,19 and Abl,17 including the compound\\u2019s H-bond interactions with the hinge and the position of its C-helix. The nonvisible residues differ between the two BTK structures, which can be correlated with the different interactions made with different compounds or by the mutation in the activation loop. The tip of the glycine-rich loop is disordered in the B43 structure but is ordered in the Dasatinib structure, while the activation loop is disordered in the Dasatinib structure but is well ordered in the B43 structure. The curled glycine-rich loop forms van der Waals contacts with Dasatinib, between a methyl substituent and the Phe413 side chain, whereas no similar interactions take place between B43 and the BTK-KD protein residues and the glycine-rich loop is disordered. One might expect that the Y551E mutation in the activation loop of the BTK-KD Y551E/Dasatinib structure is responsible for the activation loop disorder; the mutated residue is electrostatically incompatible with the conformation of the activation loop seen in the BTK-KD/B43 structure. In the BTK-KD/B43 structure, the Y551 residue is in close proximity to an Asp521 side chain; this is likely to be electrostatically repelled by mutation of the tyrosine to a glutamate. While it is often hard to pinpoint why flexible regions of crystal structures are disordered, it appears that formation of critical molecular interactions produces ordered electron density for the more flexible regions of BTK. Comparison of the structures of the human BTK-KD-Y551E/Dasatinib complex and the BTKKD/B43 complex reveals a change of conformation from catalytically \\u2018\\u2018active\\u2019\\u2019 to \\u2018\\u2018inactive.\\u2019\\u2019 The Dasatinib complex is more similar to the ATP-bound conformation of most kinases, in which a conserved glutamate from the C-helix forms a salt bridge to the catalytic lysine. In fact, no crystals could be formed with the unphosphorylated, wild-type BTK kinase construct, prompting us to make the Y551E mutant as a mimic of the phosphorylated wild-type protein. In contrast, the BTK-KD/B43 complex shows an outward shift of the C-helix [Fig. 3(B)] relative to its position in the Dasatinib structure, the conserved salt bridge from the glutamate to the catalytic lysine breaks, and a large hydrophobic pocket opens behind the gatekeeper residue. The ability of different kinases to adapt a C-helix out conformation might enable the design of specific inhibitors that targets this larger hydrophobic pocket. Furthermore, Cys481 in the active site of BTK-KD could also be exploited to gain kinase selectivity in which a small molecule may be irreversibly bound to this cysteine via a covalent bond.11 To determine the overall similarity of the BTKKD/B43 structure to other kinases, the B43 complex structure was submitted to the Dali-lite server for structure alignment and scoring20 (http://ekhidna. biocenter.helsinki.fi/dali_server/). The top hits, inactive Hck (PDB ID: 1QCF), inactive SRC (PDB ID: 2SRC), inactive ABL (PDB ID: 2G1T), ITK (PDB ID: 1SM2), and mouse BTK (PDB ID: 1K2P), could be aligned with the human BTK over more than 260 a-carbons and with an rmsd of 2.0 A\\u02da or better. The highest scoring hits, excluding the TEC family of kinases, were all inactive conformations of tyrosine kinases from the Src and Abl families, consistent with their overall sequence similarities to human BTK. The conformation of the activation loop and C-helix in the human BTK-KD/B43 structure is very similar to the inactive Src structure (PDB ID: 2SRC) with an rmsd 1.64 A\\u02da over 257 a-carbons; in Src the activation loop forms two alpha helices and occludes access of the substrate peptide. The overall conformation of the BTK-KD Y551E/Dasatinib structure is similar to the active c-Src (CSK) structure (PDB ID: 1Y57) where the activation loop is swung out and the C-helix moves toward the active site (Fig. 4). Comparison of Src and TEC family kinases The phosphorylation-triggered regulation of BTK and Src differ. Unlike the Src family, the TEC family of nonreceptor tyrosine kinases lacks a conserved tyrosine in the C-terminus that could be phosphorylated to then bind to the SH2 domain.16 BTK is regulated by the phosphorylation of two tyrosine residues, Tyr223 in the SH3 domain and Tyr551 in the activation loop of the kinase domain, both of which participate in kinase activation.21 In a recent study of BTK autophosphorylation, the Y551F mutant was shown to have a 5 to 10-fold lower enzymatic activity than the wild-type protein, indicating that this tyrosine plays an important role in BTK activation.22 Furthermore, mutation of a conserved tryptophan in the N-terminal W-E-X motif, in which X is a hydrophobic residue, also appears to effect the activities of the two kinase families differently. In Src, mutation of the Trp to Ala increases kinase activity while in BTK, mutation of the Trp to Ala reduces kinase activity.23 The human BTK structures described here include ordered density for the WEI motif (residues 392\\u2013395), an area which was disordered in the Marcotte et al. PROTEIN SCIENCE VOL 19:429\\u2014439 435 available murine BTK structure (PDB ID: 1K2P) and the human ITK structure (PDB ID: 1SM2). This enables a structural comparison of the TEC family and the Src family kinases in this conserved region. The Trp side chain shifts from being solvent exposed in the \\u2018\\u2018inactive\\u2019\\u2019 BTK-KD/B43 complex structure to being wedged into a pocket behind the inward C-helix in the \\u2018\\u2018active\\u2019\\u2019 BTK-KD Y551E/Dasatinib complex structure [Fig. 4(A,B)]. A structural superposition of the two BTK structures with the inactive SRC (PDB ID: 2SRC) 24 and an active CSK structure (PDB ID: 1Y57) 16 show that the side chain of Trp395 superimposes in the active structures of both kinase families. In the \\u2018\\u2018inactive\\u2019\\u2019 conformations, a lysine or methionine side chain from the rotated C-helix sterically occludes the tryptophan side chain, and the Trp side chains are not superimposable [Fig. 4(B)]. The similar structural shift observed suggests that the Src and Tec families utilize a similar means of activating the conformation of the kinase domain. However, the inactive conformations of the Trp side chain differ and the differential energy required in switching from inactive to active conformations may differ; this could explain why mutation of this residue to alanine produces different regulatory effects on the two proteins. This study reports the structures of human BTKKD with Dasatinib or 4-amino-5-(4-phenoxyphenyl)- 7H-pyrrolo[2,3-d]pyrimidin-7-yl-cyclopentane bound at the active site. Analysis of the binding modes reveals that the size and H-bonding potential of the gatekeeper residue in different kinases and the nature of the residues that make up the hydrophobic pocket behind the gatekeeper residue are critical in dictating if either compound binds. We propose that the presence of a distinct hydrophobic pocket in the B43 structure suggests that one could design compounds with limited selectivity for kinases that can adapt the \\u2018\\u2018C-helix-out\\u2019\\u2019 conformation. Movements in the C-helix, presumably induced by phosphorylation of the activation loop residue Tyr551, can shift BTK into an active conformation and could generate a second metal binding site containing Glu445 that Lin et al. have shown is important for optimal catalytic activity of the Tyr551-phosphorylated BTK.25 Finally, we show that, like the Src-family of kinases, BTK can adapt a similar conformational rotamer of Trp395 in its \\u2018\\u2018active\\u2019\\u2019 conformation, which is correlated with a similar movement of the C-helix. With a growing recognition that BTK plays a key role in many B cell lymphomas and autoimmune diseases, these structures will assist with selective drug design.  Bruton\\u2019s tyrosine kinase (Btk) is required for B cell development and B cell antigen receptor (BCR) function. Cross-linking of BCR induces phosphorylation of Btk at Tyr551 and Tyr223. However, the functional requirement of these phosphorylation for BCR signaling remains unclear. We demonstrate here that mutation of Tyr551, not Tyr223, abrogates the BCR-induced calcium mobilization. Not only Lyn, but also Syk was required for tyrosine phosphorylation of Btk in BCR signaling. These results suggest that transphosphorylation of Btk on Tyr551 is essential for BCR function and that this phosphorylation is mediated through the concerted actions of Lyn and Syk.  The B cell antigen receptor (BCR)1 is composed of surface immunoglobulin noncovalently associated with a pair of Ig\\u03b1/Ig\\u03b2 disulfide-linked heterodimers, which are essential for signal transduction. Stimulation of the BCR induces the enzymatic activation and tyrosine phosphorylation of three distinct families of nonreceptor cytoplasmic protein tyrosine kinases (PTKs), the Src family, Syk, and Btk. The Src family kinases are rapidly activated after BCR engagement, and their activation correlates with the initial tyrosine phosphorylation of the immunoreceptor tyrosine-based activation motif on the BCR Ig\\u03b1 and Ig\\u03b2 subunits (reviewed in Refs.1-4).  Temporally, activation of Src family kinases is followed by Btk and Syk (5). This sequential activation potentially places Btk and Syk downstream of Src family kinases. Utilizing co-overexpression system in fibroblasts and COS cells, it has been demonstrated that Lyn transphosphorylates Btk on Tyr551 in the catalytic domain, a site homologous to the Src family kinase consensus autophosphorylation site (6, 7). This results in a 5\\u201310-fold increase in Btk enzymatic activity (7). The increase in activity also leads to increased autophosphorylation at Tyr223 in the SH3 domain of Btk (8). The identical phosphopeptides were generated after cross-linking of the BCR, indicating that these sites are also tyrosine phosphorylated in B cells (7). Although the importance of phosphorylation of Tyr551 and Tyr223 for fibroblast transformation has been examined (8, 9), functional significance of phosphorylation of Tyr551 and Tyr223 of Btk in BCR signaling remains elusive.  To genetically define the functional relationship among Lyn, Syk, and Btk in BCR signaling, we established each PTK-deficient DT40 B cells (10, 11). Our previous results have shown that the BCR-induced calcium mobilization is abrogated in Btk-deficient DT40 cells and that the loss of Btk does not significantly affect the activation of Lyn and Syk in BCR signaling (11). Here we show that BCR-induced tyrosine phosphorylation of Btk is abolished in Lyn/Syk double-deficient DT40 cells, suggesting that Btk acts downstream of Lyn and/or Syk in BCR signaling. Moreover, this phosphorylation is partially inhibited in Lyn- or Syk-deficient cells, indicating contribution of both Lyn and Syk to Btk phosphorylation. The Btk Y223F mutant was able to restore the BCR-induced calcium mobilization, whereas the Y551F mutant could not. Thus, these results suggest that phosphorylation of Tyr551 of Btk through Lyn and Syk is essential for BCR signaling.  Previous Section Next Section EXPERIMENTAL PROCEDURES  Cells, Antisera, and DNA Transfection DT40 cells were cultured in RPMI 1640 supplemented with 10% fetal calf serum, penicillin, streptomycin, and glutamine. Anti-chicken IgM mAb M4 and anti-phospholipase C (PLC)-\\u03b32 Ab were described previously (10). The anti-phosphotyrosine mAb (4G10) and anti-T7 mAb were obtained from Upstate Biotechnology, Inc. and Novagen, respectively. T7-tagged and mutant Btk cDNAs were created by polymerase chain reaction, and the resulting constructs were confirmed by DNA sequencing. These cDNAs were cloned into pApuro expression vector (10). For DNA transfection into DT40 cells, DNA was linearized, electroporated, and selected in the presence of puromycin (0.5 \\u03bcg/ml). The expression of Btk was analyzed by Western blotting.  Immunoprecipitation and Immunoblot Analysis DT40 cells were stimulated by mAb M4 for indicated time. Cells were solubilized in Nonidet P-40 lysis buffer (1% Nonidet P-40, 150 mM NaCl, 20 mM Tris, pH 7.5, 1 mM EDTA) containing 50 mM NaF, 10 \\u03bcM molybdate, and 0.2 mM sodium vanadate supplemented with protease inhibitors described previously (10). Cell lysates were sequentially incubated (1 h at 4\\u2009\\u00b0C for each incubation with Ab and protein A-Sepharose). For immunoblotting, samples were separated on SDS-PAGE and transferred to nitrocellulose membrane (Amersham Corp.). Filters were incubated with mAb 4G10 or anti-T7 mAb. After washing, filters were developed using a sheep anti-mouse IgG Ab conjugated to horseradish peroxidase and enhanced chemiluminescence (ECL).  For in vitro kinase assay, the immunoprecipitates were washed with 20 mM Hepes, pH 8, and 150 mM NaCl after washing with lysis buffer. Added to each sample was 50 \\u03bcl of kinase buffer (20 mM Hepes, pH 8, 10 mMmagnesium acetate, 10 mM MnCl2) in the absence or presence of ATP (1 \\u03bcM). Recombinant glutathioneS-transferase fusion protein containing a cytoplasmic domain of mouse Ig\\u03b1 (glutathione S-transferase/Ig\\u03b1) was made and used as an exogenous substrate (6). The reactions were allowed at 30\\u2009\\u00b0C for 10 min and terminated by the addition of sample buffer.  Calcium and Phosphoinositide Analysis For calcium analysis, cells (5 \\u00d7 106) were resuspended in phosphate-buffered saline containing 20 mM Hepes, pH 7.2, 5 mM glucose, 0.025% bovine serum albumin, and 1 mM CaCl2 and loaded with 3 \\u03bcMFura-2/AM at 37\\u2009\\u00b0C for 45 min. Cells were washed twice and adjusted to 106 cells/ml with continuous monitoring of fluorescence spectrophotometer (model F-2000; Hitachi) at an excitation wavelength of 340 nm and an emission wavelength of 510 nm. Calibration and calculation of calcium level were done as described (12).  For phosphoinositide analysis, cells (106/ml) were labeled with myo-[3H]inositol (10 \\u03bcCi/ml, 105 Ci/mmol) for 4\\u20135 h in inositol-free RPMI 1640 supplemented with 10% dialyzed fetal calf serum, then stimulated in the presence of 10 mM LiCl with mAb M4. The soluble inositol phosphates were extracted with trichloroacetic acid at indicated time points, and applied to AG1-X8 (formate form) ion exchange columns (Bio-Rad) preequilibrated with 0.1 M formic acid. The columns were washed with 10 ml of H2O and 10 ml of 60 mMammonium formate, 5 mM sodium tetraborate. Elution was performed with increasing concentrations of ammonium formate (0.1\\u20130.7M).  Previous Section Next Section RESULTS  We have used a genetic approach to determine the requirements for Btk tyrosine phosphorylation following BCR engagement. We utilized three mutants of the chicken B cell line DT40, generated by inactivation of either lyn, syk, or both genes by homologous recombination (10, 11). Since the Ab raised against chicken Btk does not immunoprecipitate efficiently, we expressed an epitope-tagged version of Btk (designated T7-Btk) into wild type and these mutant DT40 cells. Clones expressing similar levels of T7-Btk in these deficient DT40 cells were selected. These clones were lysed prior to and following BCR ligation, and T7-Btk was immunoprecipitated with anti-T7 mAb. As shown in Fig. 1, Btk was inducibly tyrosine-phosphorylated following BCR stimulation in wild type DT40 cells, consistent with previous reports (5, 13-15). In contrast to wild type cells, Lyn/Syk double-deficient DT40 cells failed to exhibit any Btk tyrosine phosphorylation following BCR ligation, indicating requirement of Lyn and/or Syk for the BCR-induced tyrosine phosphorylation of Btk. In Lyn-deficient DT40 cells, tyrosine phosphorylation of Btk at 1 and 3 min after BCR ligation was significantly reduced, whereas this phosphorylation at 10 min reached almost the same level as that in wild type cells. Compared with Lyn-deficient DT40 cells, Syk-deficient cells showed a complementary time course of the BCR-induced phosphorylation of Btk; phosphorylation of Btk was only observed at 1 and 3 min after BCR stimulation. These data suggest that either Lyn or Syk alone is capable of phosphorylating Btk at least to some extent and that the concerted actions of Lyn and Syk are required for full phosphorylation of Btk in BCR signaling.  Figure 1 View larger version: In this page In a new window Download as PowerPoint Slide Figure 1 BCR-induced tyrosine phosphorylation of Btk in various DT40 mutant cells. At the indicated time points after M4 stimulation (4 \\u03bcg/ml), cells were lysed in 1% Nonidet P-40 lysis buffer and immunoprecipitated with anti-T7 mAb. Immunoprecipitates were loaded onto 8% SDS-PAGE, and the blotted membrane was incubated with mAb 4G10. After the filter was stripped, the same blot was reprobed with anti-T7 mAb. pY, phosphotyrosine.  It has been demonstrated recently that transphosphorylation of Btk at Tyr551 results in increased its enzymatic activity, leading to autophosphorylation of a second tyrosine Tyr223 in a fibroblast system (8). To determine that this sequential phosphorylation of Btk occurs also after BCR stimulation, we transfected Y551F and Y223F mutants of Btk into Btk-deficient DT40 cells (Fig. 2 A). Stimulation of BCR did not enhance tyrosine phosphorylation of Btk(Y551F) (Fig. 3). This result suggests the possibility that Tyr551 of Btk is only target of Lyn/Syk-dependent phosphorylation after BCR stimulation in DT40 cells. Alternatively, phosphorylation at Tyr551 is a critical step for subsequent phosphorylation of Btk in BCR signaling context. In contrast to Btk(Y551F), Btk(Y223F) showed increased tyrosine phosphorylation upon BCR stimulation, although this phosphorylation was only observed at 1 min after receptor cross-linking (Fig. 3). Thus, these results implicate that a primary target of tyrosine through concerted actions of Lyn and Syk is Tyr551 of Btk in BCR signaling, leading to phosphorylation of Tyr223. Since Btk(Y223F) did not show significant tyrosine phosphorylation at 3 and 10 min after receptor stimulation, our finding also suggests that phosphorylation of Tyr223 is required for sustained Btk phosphorylation in BCR signaling.  Figure 2 View larger version: In this page In a new window Download as PowerPoint Slide Figure 2 Schematic representation of Btk cDNA constructs (A) and expression of mutant Btk in Btk-deficient DT40 cells (B). DT40 cells expressing indicated constructions were lysed with Nonidet P-40 lysis buffer (2 \\u00d7 106 cells/lane) and immunoprecipitated with anti-T7 mAb. Immunoprecipitates were divided, and half of them were used for Western blotting with anti-T7 mAb. The remaining half were used for in vitro kinase assay as described under \\u201cExperimental Procedures\\u201d and blotted with mAb 4G10. In the absence of ATP, in vitro kinase samples of these immunoprecipitates showed no significant tyrosine phosphorylation. pY, phosphotyrosine.  Figure 3 View larger version: In this page In a new window Download as PowerPoint Slide Figure 3 BCR-induced tyrosine phosphorylation of Btk(Y223F) and Btk(Y551F). At the indicated time points after M4 stimulation (4 \\u03bcg/ml), Btk-deficient cells expressing these constructs were lysed in 1% Nonidet P-40 lysis buffer and immunoprecipitated with anti-T7 mAb. Immunoprecipitates were loaded onto 8% SDS-PAGE, and the blotted membrane was incubated with mAb 4G10. After the filter was stripped, the same blot was reprobed with anti-T7 mAb. pY, phosphotyrosine.  Utilizing Btk-deficient DT40 cells expressing Btk(Y551F) and Btk(Y223F), we analyzed the effects of these mutations on BCR signaling. Wild type and kinase-negative T7-Btk were also transfected into Btk-deficient DT40 cells as a positive and a negative control, respectively (Fig. 2 A). To determine whether these mutations affect tyrosine kinase activity, Btk immunoprecipitates were used forin vitro kinase assays with glutathioneS-transferase/Ig\\u03b1 as an exogenous substrate. Btk(R525Q) exhibited no kinase activity, indicating that the immunoprecipitates are largely free of contaminating tyrosine kinases. When the extent of tyrosine phosphorylation is normalized to the amount of protein present in each kinase assay, both Btk(Y551F) and Btk(Y223F) had similar transphosphorylation activity compared with wild type Btk (Fig.2 B). We showed previously that the BCR-induced PLC-\\u03b32 activation is abrogated in Btk-deficient DT40 cells, leading to loss of calcium mobilization (11). Thus, we examined whether these mutations were able to restore these defects or not. As shown in Fig.4 A, DT40 cells expressing Btk(Y223F) exhibited normal calcium mobilization, whereas Btk(Y551F) was able to mobilize only small amount of calcium upon receptor cross-linking. Consistent with these data, cross-linking of BCR on DT40 cells expressing Btk(Y223F) stimulated inositol 1,4,5-trisphosphate (IP3) production and tyrosine phosphorylation of PLC-\\u03b32, although these parameters were lower than wild type Btk (Fig. 4,B and C). This might reflect the lower expression level of Btk(Y223F) than wild type Btk (Fig. 2 B). The BCR-induced IP3 production and tyrosine phosphorylation of PLC-\\u03b32 in DT40 cells expressing Btk(Y551F) was essentially the same as those in DT40 cells expressing Btk(R525Q) (Fig. 4 and data not shown). These results demonstrate that phosphorylation of Tyr551, not Tyr223, is essential for BCR signaling.  Figure 4 View larger version: In this page In a new window Download as PowerPoint Slide Figure 4 Functional restoration of BCR signaling by Btk(Y223F) and Btk(Y551F). Calcium mobilization (A), IP3 generation (B), and tyrosine phosphorylation of PLC-\\u03b32 (C). C, Btk-deficient DT40 cells expressing these mutants were stimulated for 3 min with M4 (4 \\u03bcg/ml). Cells were lysed in 1% Nonidet P-40 lysis buffer and immunoprecipitated with anti-PLC-\\u03b32 Ab. Immunoprecipitates were loaded on 6% SDS-PAGE, and the blotted membrane was incubated with mAb 4G10. After the filter was stripped, the same blot was reprobed with anti-PLC-\\u03b32 Ab. pY, phosphotyrosine.  Previous Section Next Section DISCUSSION  In Lyn-deficient DT40 cells, BCR-induced tyrosine phosphorylation of Btk was significantly inhibited at 1 and 3 min after stimulation (Fig. 1), indicating that Btk phosphorylation is mediated by Lyn in BCR signaling in these early time points. However, at 10 min after stimulation, this phosphorylation reached almost the same level in wild type cells, suggesting that tyrosine phosphorylation of Btk at 10 min after BCR stimulation is independent of Lyn. In contrast to Lyn-deficient DT40 cells, Syk-deficient cells show the profound inhibition of the BCR-induced tyrosine phosphorylation of Btk at 10 min after receptor cross-linking, implicating that this Btk phosphorylation is mediated by Syk. Taken together, these data suggest that the initial Btk phosphorylation and sustained phosphorylation are mediated by coordinated actions of Lyn and Syk after BCR cross-linking.  Our conclusion is somewhat inconsistent with the previous reports using COS cell and fibroblast expression systems (6, 7). In these systems, Lyn is able to phosphorylate Btk, whereas Syk is not. One of the possibilities about this disparity between the heterologous systems and DT40 B cell system is that Syk is maximally activated through transphosphorylation at Tyr518 and/or Tyr519 by Lyn in DT40 cells (16). In contrast, overexpressed Syk itself may not be fully activated in heterologous systems. To test this possibility, we transfected T7-Btk into Syk-deficient cells expressing Syk mutant in which Tyr518/Tyr519 is changed to Phe518/Phe519. In this mutant DT40 cell, the BCR-induced tyrosine phosphorylation of Btk was similar to that in Syk-deficient DT40 cells (data not shown), implicating that the requirement of Syk is due to up-regulated Syk through transphosphorylation of Tyr518/Tyr519. It is also possible that Btk may not be a direct substrate of Syk. Assuming that another PTK acts downstream of Syk in BCR signaling, our data might be accounted for by the involvement of this Syk-regulated PTK in tyrosine phosphorylation of Btk.  It has been reported that two tyrosines, Tyr551 and Tyr223, are phosphorylated after cross-linking of BCR. Btk(Y551F) exhibited no tyrosine phosphorylation upon BCR cross-linking, whereas Btk(Y223F) showed an increase of tyrosine phosphorylation at 1 min after receptor stimulation, suggesting that Tyr551 phosphorylation is a prerequisite for subsequent phosphorylation of Btk in BCR signaling events. Thus, these results support the previous contention that phosphorylation of Btk at Tyr551 is followed by its autophosphorylation at Tyr223 (7, 8). Since Btk(Y223F) did not show significant tyrosine phosphorylation at 3 and 10 min after receptor stimulation, phosphorylation of Tyr223 appears to be required for sustained Btk phosphorylation. Recent crystallographic analysis of Itk (Btk/Tec family PTK expressed in T cells) may provide insights into the function of phosphorylation of Tyr223 of Btk (17). Based on this analysis, the proline-rich domain adjacent to the SH3 domain of Btk/Tec family kinases contains an SH3 ligand (Fig. 2 A), allowing intramolecular interaction. Interestingly, Tyr223of Btk is located within the interface of this interaction. Thus, phosphorylation of Tyr223 may disrupt this intramolecular interaction, thereby changing the conformation of Btk. This conformational change might be required for sustained phosphorylation of Btk in BCR signaling.  Our functional data clearly indicate that Tyr551 is essential for BCR signaling, whereas Tyr223 is dispensable for BCR-induced PLC-\\u03b32 activation. Since phosphorylation of Tyr551 was already reported to increase the kinase activity of Btk with 5\\u201310-fold (7), one of the consequence of phosphorylation of Tyr551 is increased kinase activity upon BCR cross-linking. Although we carried out in vitro kinase assay on Btk immunoprecipitates in wild type DT40 cells, the BCR-induced activation of its in vitro kinase activity could not be reproducibly observed. Previous reports suggest that the magnitude of the BCR-induced activation of Btk is significantly smaller than that of maximum Btk phosphorylation by Lyn in heterologous systems (6, 7). Indeed, in our hand, this 5\\u201310-fold activation of Btk in COS cells could be reproducibly detected. Thus, the most likely explanation is that the increase of Btk enzymatic activity through phosphorylation of Tyr551 in DT40 cells is too small for our detection system.  In the case of Syk, it has been demonstrated that in addition to recruitment of Syk to phosphorylated Ig\\u03b1/Ig\\u03b2, phosphorylation at tyrosine (Tyr519 and/or Tyr518) within its activation loop is critical for BCR signal transduction (16). In this report, we show that phosphorylation of Tyr551 on Btk activation loop is also an obligatory mechanism for its participation in BCR signaling. Thus, cytoplasmic PTK cascade through phosphorylation of tyrosine located in the activation loop may be one of the general mechanisms for cytoplasmic signal transduction.  The Tec family kinases are tyrosine kinases that function primarily in hematopoietic cells. The catalytic activity of the Tec kinases is positively influenced by the regulatory domains outside of the kinase domain. The current lack of a full-length Tec kinase structure leaves a void in our understanding of how these positive regulatory signals are transmitted to the kinase domain. Recently, a conserved structure within kinases, the \\u2018regulatory spine\\u2019, has been identified that assembles and disassembles as a kinase switches between its active and inactive states. Here we define the residues that comprise the regulatory spine within Tec kinases. Compared to previously characterized systems, the Tec kinases contain an extended regulatory spine that includes a conserved methionine within the C-helix and a conserved tryptophan within the SH2-kinase linker of Tec kinases. This extended regulatory spine forms a conduit for transmitting the presence of the regulatory domains of Tec kinases to the catalytic domain. We further show that mutation of the gatekeeper residue at the edge of the regulatory spine stabilizes the regulatory spine resulting in a constitutively active kinase domain. Importantly, the regulatory spine is preassembled in this gatekeeper mutant rendering phosphorylation on the activation loop unnecessary for its activity. Moreover, we show that the disruption of the conserved electrostatic interaction between Btk R544 on the activation loop and Btk E445 on the C-helix also aids in the assembly of the regulatory spine. Thus, the extended regulatory spine is a key structure that is critical for maintaining the activity of Tec kinases.  Keywords: Itk and Btk, regulatory spine, kinase activation, gatekeeper residue, phosphorylation Go to: INTRODUCTION Protein kinases catalyze the transfer of a phosphate group from ATP to a hydroxyl containing amino acid side chain; either Ser/Thr for serine-threonine kinases or Tyr for tyrosine kinases[1]. The activity of protein kinases is exquisitely regulated within the cell[1; 2; 3; 4]. The switch from the catalytically inactive state of a kinase to the active state is often accompanied by the phosphorylation of a key residue within the activation loop, a large flexible loop that lies between the two lobes of the kinase domain structure[1]. Phosphorylation of the activation loop residue can trigger concerted movements in other mobile elements within the kinase domain such as the C-helix and the DFG motif that brings about the assembly of the catalytically critical residues in the kinase active state[1]. The active state conformations associated with different kinases are nearly identical across the many kinases for which high resolution structures have been solved[2; 3; 5; 6; 7]. It is not surprising that the active states of distinct kinases are very similar since the phospho-transfer chemistry carried out by different kinases is the same. In contrast, it is becoming clear that the structural features associated with the inactive state of various kinases differ widely.  The search for features that are conserved within the structures of active kinases has recently led to the identification of a structure termed the \\u2018regulatory spine\\u2019[8; 9]. The regulatory spine was first identified by Local Spatial Patterns alignment analysis using a set of serine-threonine and tyrosine kinase structures that included PKA as a model kinase[8; 9]. The regulatory spine defines a stretch of amino acid residues that is assembled only in the active state of kinases. Unlike consensus sequences that consist of a continuous stretch of amino acids in the primary structure, the regulatory spine consists of disparate residues that span the N- and C-terminal lobes of the kinase domain. Assembly of the regulatory spine has been proposed to be a crucial step in the activation of protein kinases and this structure is disrupted in structures of kinases in the inactive state. Moreover, the regulatory spine has been proposed as the mechanism by which allosteric effects can be propagated to the kinase active site[8; 9].  The Tec kinases are immunologically related tyrosine kinases which consist of five mammalian members: Itk, Btk, Tec, Txk and Bmx[10; 11]. Tec kinases share similar domain architecture as Src, Abl and Csk family of kinases, in that they have the SH3-SH2-kinase domain cassette[10]. Despite domain similarities, we and others have shown that the regulation of Tec kinases is distinct from Src and Abl kinases[10; 12; 13; 14]. Unlike the Src and Abl kinases, the N-terminal regulatory domains of Tec kinases positively regulate the activity of the kinase domain[14]. While the isolated kinase domains of Src and Abl are active, the isolated kinase domains of the Tec family exhibit poor catalytic activity[13; 14; 15]. Regulation of the Tec kinases is in fact more similar to the Csk family of enzymes, whose N-terminal regulatory domains are essential for the catalytic activity of the kinase domain[16]. Moreover, while the structures of full-length Src and Csk kinases are available, the structure of a full-length Tec kinase remains elusive[2; 17; 18]. This leaves a significant gap in our mechanistic understanding of the regulation of the Tec family kinases.  In this manuscript, we define the regulatory spine within the Tec family kinases and show that mutation of the spine residues leads to different effects in the context of the isolated kinase domain verses full-length Itk. Moreover, we show that the assembly of a stable regulatory spine within members of the Tec kinase family is critically dependent on the presence of the SH2-kinase linker region. The residues that comprise the regulatory spine within PKA are insufficient to promote the assembly of a stable regulatory spine within the Tec kinases. We extend the Tec regulatory spine to include a conserved methionine within the C-helix and a conserved tryptophan residue within the SH2-kinase linker of Tec kinases. Together with the spine residues originally identified within the PKA kinase domain, the conserved methionine and tryptophan residues form a continuous structure that links the kinase active site to the N-terminal regulatory domains of Tec kinases. Furthermore, we show that the stabilization of the regulatory spine by the gatekeeper threonine to methionine mutation eliminates the need for phosphorylation on the activation loop for Btk kinase activity. Thus, the extended regulatory spine is a structure that is critical for the regulation of Tec kinases and if stabilized appropriately, is sufficient for activation in the absence of activation loop tyrosine phosphorylation and in the absence of the non-catalytic Tec regulatory domains.  Go to: RESULTS The regulatory spine controls Tec kinase activity  The regulatory spine within PKA consists of five residues: L95, L106, Y164, F185, and D220[9]. These residues are spread throughout the primary sequence of the PKA kinase domain: L95 is located on the C-helix, L106 is on the N-terminus of the \\u03b24 strand, Y164 is part of the \\u2018HRD\\u2019 motif, F185 is from the \\u2018DFG\\u2019 motif and D220 is on the F-helix within the C-terminal lobe[9]. Alignment of the structure of the Btk and Itk kinase domains with that of PKA shows that the corresponding regulatory spine residues within the Btk and Itk kinase domains should consist of: Btk M449, L460, H519, F540 and D579 and Itk M409, L420, H479, F500 and D539 respectively (Fig. 1a).  Figure 1 Figure 1 Identification of the regulatory spine in Tec kinases To test the role of the predicted spine residues in the Tec kinases, we wished to take advantage of the rapid bacterial expression and purification system that has been developed for the isolated kinase domains of Itk and Btk[19]. One issue that arises, however, is the fact that the isolated kinase domains of the Tec kinases exhibit poor catalytic activity and so the expected loss of function mutations in the regulatory spine would be difficult to characterize given the already low activity of the wild type kinase domains. In order to take advantage of the ease of the bacterial expression system, we needed an isolated Btk and Itk kinase domain with higher catalytic activity. It has been demonstrated previously that introduction of a bulky hydrophobic residue such as isoleucine or methionine at the gatekeeper position activates multiple kinases[20]. Indeed, we find that the Itk F434M and Btk T474M isolated kinase domains are more active when compared to wild-type isolated kinase domain (manuscript in preparation). We therefore probed the importance of the predicted regulatory spine residues in Itk and Btk by mutating them individually to alanine in the context of the Itk F434M and Btk T474M isolated kinase domain mutants. Activity measurements are carried out by monitoring the phosphorylation levels of a peptide substrate in a radioactive assay or by detecting the level of autophosphorylation on the activation loop tyrosine (Y511 in Itk and Y551 in Btk) by western immunoblotting. While phosphorylation of the activation loop tyrosine is achieved by the activity of the Src family kinases Lck and Lyn respectively in vivo, under in vitro conditions Tec kinases autophosphorylate on the activation loop tyrosine[21; 22; 23].  Disruption of the regulatory spine residues by mutation to alanine is predicted to disrupt kinase activity. Indeed, mutation of Btk M449, H519, F540 and D579 and Itk M409, H479, F500 and D539 to alanine within the isolated kinase domain of Itk F434M and Btk T474M leads to decreased phosphorylation on the activation loop tyrosine and drastically reduces the catalytic activity (Fig. 1b, c, d and e). Itk kinase residues such as M409, F500 and H479 are highly conserved with other kinases with well-defined roles in catalysis[8]. Itk M409 is involved in binding the substrate ATP[12]. Itk F500 is part of the \\u2018DFG\\u2019 motif at the start of the activation loop segment and is responsible for stabilizing the conformation of the preceding aspartate, and the C-helix[8]. Itk H479 is part of the \\u2018HRD\\u2019 motif where it serves as a scaffold for D499 and F500[8]. It is therefore not surprising that mutation at these sites leads to a loss in activity. However, Itk D539 is not part of any previously characterized regulatory motif and therefore the drastic reduction in Itk activity upon mutation of this residue to alanine highlights the importance of the regulatory spine in regulating Itk activity. Moreover, since the regulatory spine residues are conserved within the Tec kinase family, this structure would be predicted to be critical for regulating the activity of all Tec kinase family members.  Unexpectedly, mutation of Itk L420 and Btk L460 to alanine, which based on the previous PKA work is predicted to be part of the regulatory spine, failed to inactivate the isolated kinase domains of Itk F434M or Btk T474M (Fig. 1b, c, d and e). In fact, the isolated kinase domains of both Itk L420A/F434M and Btk L460A/T474M double mutants showed an increase in activity when compared to the isolated kinase domain of Itk F434M and Btk T474M, respectively. To ensure that the activating effect of the Leu to Ala mutation was not an artifact of working with the activated (gatekeeper mutant) isolated kinase domains of Itk and Btk, the regulatory spine residues were mutated to alanine in the context of wild-type isolated kinase domains of Itk and Btk. Since the catalytic activity of the isolated kinase domain of Tec kinases is generally poor, the detection of activation loop phosphorylation levels by western immunoblotting is difficult and for Itk, in particular, phosphorylation Y511 is below the detection limit by western immunoblotting. Hence, the activity of the wild-type and regulatory spine mutants of the isolated kinase domains of Itk and Btk are monitored by their ability to phosphorylate a peptide substrate in the radioactive assay.  The results of the regulatory spine mutations in the context of the wild type kinase domains mirror the results for the activating gatekeeper mutants described above. Mutation of Btk M449, H519, F540 and D579 to alanine within the isolated wild type kinase domain of Btk leads to decreased catalytic activity (Fig. 1f). Circular dichroism spectra of the Btk mutants overlay well with that of wild-type Btk showing that the decreased activity of the Btk mutants are not due to unfolding of the kinase (Supp. Fig. 1). Since the activity of the isolated Itk wild-type kinase domain is not significantly above background levels, the resulting activity of the Itk spine mutants: Itk M409A, H479A, F500A and D539A can not be interpreted except to say activity is no greater than wild type (Fig. 1g). However, consistent with our earlier results with the gatekeeper mutant, the Itk L420A and Btk L460A mutants are both more active than wild-type isolated kinase domain of Itk and Btk, respectively (Fig. 1f and g). Interestingly, previous studies on the regulatory spine within the Abl kinase have shown that mutation of Abl L320 to glycine (corresponding to Itk L420 and Btk L460), leads to only a slight decrease in kinase activity of full length Abl[20]. This is also consistent with modeling studies of Abl that have shown that Abl L320 has a modest impact on regulatory spine formation[20]. Thus, Itk L420 and Btk L460 do not play a major role in the assembly of the regulatory spine within the context of the isolated kinase domain of Itk and Btk.  The regulatory spine is not assembled in the structure of phosphorylated Itk kinase domain  Activation of Tec kinases requires the phosphorylation of a conserved tyrosine within the activation loop[12; 21; 23]. Separate high-resolution structures of the Btk kinase domain with Y551 either phosphorylated (active) or unphosphorylated (inactive) have recently been reported[24]. A comparison of the inactive and active states of Btk kinase domain shows clear differences in the region of the regulatory spine (Fig. 2a). The regulatory spine residues assemble into a linear arrangement in the structure of the active Btk kinase, while interactions between the regulatory spine residues (in particular L460, M449 and F540) are disrupted in the structure of inactive Btk kinase. The structural differences in this region between active and inactive Btk are consistent with the role of the regulatory spine as defined previously for PKA (Fig. 2a).  Figure 2 Figure 2 The regulatory spine is assembled only in the active state of kinases Crystal structures of the isolated kinase domain of Itk that is either unphosphorylated or phosphorylated on Y511 in the activation loop are also available[12]. Surprisingly, the available Itk kinase domain structures overlay quite well, with little or no conformational differences between them regardless of the phosphorylation state of the activation loop. Inspection of the Itk kinase domain structures shows that the regulatory spine residues in the Itk kinase domain structures adopt a disrupted configuration similar to the inactive Btk kinase domain (Fig. 2b). An additional hallmark of an active kinase domain structure is the formation of a crucial ion pair between a conserved lysine positioned within the \\u03b23 strand of the kinase domain and a conserved glutamate on the C-helix[1]. The distance between the corresponding residues (Itk K390 and E405) within the structures of both phosphorylated and unphosphorylated Itk kinase domain is on the order of 8.0 \\u00c5 which is significantly greater than the 3 to 4 \\u00c5 distance that is observed in the structures of other active kinases (Fig. 2b)[12]. Indeed, the distance between the same conserved ion pair within Btk, (Btk K430 and E445) changes from 14.6 to 3.8 \\u00c5 upon activation of the kinase (Fig. 2a). These observations suggest that both the phosphorylated and unphosphorylated Itk kinase domains adopt a conformation consistent with the \\u2018inactive\\u2019 enzyme. The inability of the Itk kinase domain to fully assemble into an active conformation, despite being phosphorylated on Y511 in the Itk activation loop, points to additional requirements for activation.  A notable difference between the Itk and Btk crystal structures is that the construct used for the crystallization of the Btk kinase domain included the SH2-kinase linker, while that of the Itk kinase did not (Fig. 2a & b)[12; 24]. We hypothesize that the absence of the SH2-kinase linker within the Itk construct used for crystallography might prevent the assembly of a stable regulatory spine in the kinase domain despite phosphorylation on the activation loop Y511.  Extension of the Tec kinase regulatory spine  We have previously shown that the SH2-kinase linker, the 17 amino acids between the SH2 domain and the kinase domain, is critical for the activity of both Btk and Itk, exerting a positive effect on the catalytic function of the kinase domain[14]. More specifically, a conserved tryprophan (Itk W355, Btk W395) in the SH2-kinase linker, as well as a methionine residue in the C helix (Itk M410, Btk M450) are crucial residues in the Tec family kinase regulatory apparatus (Fig. 3a)[14]. Like the regulatory spine mutants described in Figure 1, point mutations of Itk W355, Btk W395 or Itk M410 to alanine all result in a significant drop in catalytic activity (Fig. 3b). Examination of the crystal structures of Btk shows that Btk W395 and Btk M450 are located at the \\u2018top\\u2019 of the regulatory spine and, in the active Btk structure, serve to extend the hydrophobic packing of the spine residues well into the N-terminus of the kinase domain (Fig. 3c & d). In the inactive Btk structure, repositioning of the W395 and M450 side chains disrupts the extended regulatory spine (Fig. 3c & d) in much the same manner that repositioning of M449 in the inactive Btk structure disrupts the core of the spine (Fig. 2a). The active and inactive Btk structures consist of the same amino acids (residues 382\\u2013659) and differ with respect to the activation loop tyrosine (Y551). In the active structure the activation loop tyrosine is mutated to glutamate (Y551E) to mimic phosphorylated Btk, and for the inactive structure wild type Btk in the unphosphorylated form was used for crystallization[24].  Figure 3 Figure 3 Extension of the regulatory spine within Tec kinases to include the SH2-kinase linker Compared to the regulatory spine that has been defined for PKA[9], the Tec kinase family requires at least two additional residues (Itk W355 & M410) to fully assemble the regulatory spine. The tryptophan residue is located in the SH2-kinase linker region providing an explanation for the positive regulatory role of this non-catalytic region[14]. The available Itk kinase domain structures are entirely consistent with this finding; without the contribution of the SH2-linker region, the regulatory spine of Itk does not assemble into the active conformation. As a result, the structures of the phosphorylated and unphosphorlated Itk kinase domain fragments are very similar to each other and neither resembles an active kinase. Future structures of Tec kinases with and without the SH2-kinase linker in various activation (phosphorylation) states will be required to fully probe this hypothesis.  Itk L420 and Btk L460 are essential for the formation of the extended regulatory spine within full-length Tec kinases  The conserved tryptophan in the SH2-kinase linker is critical for the activity of full-length Tec kinases[14]. Upon activation, the tryptophan side-chain forms part of a hydrophobic pocket that is lined by the conserved methionine on the C-helix and leucine on the \\u03b24 strand (Itk M410, L420 and Btk M450, L460) (Fig. 3d). Although Itk L420 and Btk L460 were shown to not be a part of the regulatory spine in the context of the isolated kinase domain (Fig. 1), the structure of the Btk kinase domain (which contains the SH2-kinase linker region) suggests that this leucine would be critical for the formation of the extended regulatory spine in the context of the full-length kinase. We therefore tested the role of this leucine in the context of full-length Itk.  As shown in Figure 3e, mutation of Itk L420 to alanine within the context of full-length Itk disrupted the activity of Itk. Consistent with our earlier results, mutation of Itk M409, F500 and D539 to alanine within full length Itk also inactivated the kinase. Problems with cloning prevented us from testing the activity of the full-length Itk H479A mutant. Together, Itk M409, L420, H479, F500, D539, M410 and W355 form a conserved allosteric signaling network that is absolutely required for the activity of Itk. Based on sequence conservation, the extended regulatory spine defined here likely controls the activity of the entire Tec kinase family.  Phosphorylation on the activation loop tyrosine is not required for the activity of the Btk T474M mutant  It is well established that activation loop phosphorylation is required for activation of tyrosine kinases[1; 8] and specifically for the Tec kinases, phosphorylation on the activation loop tyrosine results in at least a ten-fold increase in activity[21; 25]. Structures of active and inactive kinase domains from a number of different kinases demonstrate a conserved electrostatic network that switches between the active and inactive conformations (Fig. 4a)[1; 8]. In the inactive state of Btk, the conserved glutamate (Btk E445) on the C-helix (of the Lys-Glu ion pair) is associated with a conserved arginine (Btk R544) on the activation loop[24; 26]. Activation of Btk by phosphorylation on Btk Y551 (the activation loop tyrosine) leads to a specific interaction between pY551 and R544, and a concomitant loss of the association between the R544 and E445[24; 26]. Btk E445 on the C-helix then swings toward the kinase active site and associates with the conserved lysine side chain (Btk K430) to bring about the assembly of the active state of the kinase[24; 26]. The conformational adjustment of the C-helix brings M449 and M450 in line with the other residues of the regulatory spine (Fig. 4a). Thus, phosphorylation on the activation loop tyrosine is an initiating step in the process of regulatory spine assembly.  Figure 4 Figure 4 The regulatory spine is preassembled in the Btk T474M gatekeeper mutant Since mutation of the gatekeeper residue to methioine appears to stabilize the regulatory spine within Btk, we next probed the requirement of activation loop phosphorylation in the context of the T474M gatekeeper mutation. The activity of the BtkT474M/Y551F double mutant was compared to that of the Btk T474M mutant by monitoring phosphorylation of an exogenous substrate. We find that mutation of the activation loop Btk Y551 to phenylalanine has no effect on the activity of the Btk T474M mutant (Fig. 4b & c). The Btk T474M/Y551F double mutant and the Btk T474M single mutant are both active, whereas the wild type Btk kinase requires activation loop phosphorylation; the Btk Y551F single mutant exhibits poor activity (Fig. 4b). These data are consistent with the idea that phosphorylation on the activation loop tyrosine initiates the assembly of the regulatory spine by disrupting the association between Btk E445 and Btk R544. Once pY551 competes with E445 for association with R544, the E445/K430 salt bridge is formed, bringing the C-helix (and importantly M449) into the regulatory spine structure. Under conditions that pre-organize the regulatory spine structure (as in the Btk T474M mutant), phosphorylation on the activation loop Y551 is no longer required to trigger the conformational changes that accompany Btk activation.  We further tested the importance of this switched electrostatic network by mutating Btk R544. We reasoned that mutation of Btk R544 will activate the kinase as inward movement of the C-helix is restrained by the E445:R544 interaction and loss of this electrostatic interaction should facilitate movement of the C-helix. Indeed, mutation of Btk R544 to serine in the context of wild-type isolated kinase domain of Btk leads to a two-fold increase in activity (Fig. 4d & e). In contrast, mutation of Btk R544 to serine in the context of Btk T474M isolated kinase domain does not further activate the kinase (Fig. 4d & e). These results again suggest that pre-assembly of the kinase regulatory spine (by mutation) can overcome the regulatory interactions that normally control kinase activity. Thus, for the wild-type kinase under physiological conditions, phosphorylation on the activation loop sets into motion a cascade of events: (1) formation of the pY551:R544 interaction with concomitant disruption of the Btk E445:R544 interaction, (2) inward movement of the C-helix and formation of the E445:K430 interaction and (3) assembly of the regulatory spine, that culminates in the formation of an active kinase (Fig. 4a).  Go to: DISCUSSION The structures of the isolated kinase domain of Btk in both the active and inactive states have provided significant insight into Tec kinase activation[24; 26]. However, the lack of a full-length structure of any Tec kinase leaves numerous unanswered questions regarding the regulation of Tec kinases. While the SH2 domain and the SH2-kinase linker region of Tec kinases have been shown to be required for the activity of Tec kinases[12; 13; 14], the mechanism by which these regulatory domains positively influence catalytic activity has not been clear. Here we identify an intramolecular allosteric signaling network that extends from the SH2-kinase linker into the kinase domain of Tec kinases. This intramolecular connectivity defines how the non-catalytic Tec regulatory domains, through the SH2-kinase linker region, impinge on the kinase domain and stabilize the regulatory spine. In the absence of the Tec regulatory domains, whether achieved by deletion or conformational changes within the full-length molecule, the regulatory spine is disrupted and kinase activity is inhibited.  Comparing our results to those previously published for PKA shows that the Tec kinases require an extended spine structure and it is this longer regulatory spine that couples kinase activity to regions outside of the kinase domain. Specifically, a conserved methionine in the C-helix and a conserved tryptophan in the SH2-linker swing in to \\u2018cap\\u2019 the spine in the active Btk structure and are critical for the activity of the Tec kinases. Mutation of these extended spine residues disables kinase activity even if the \\u2018core\\u2019 residues defined for PKA are intact.  Further support for the importance of the entire regulatory spine comes from examination of the genetic mutations in Btk that are associated with X-linked agammaglobulinemia (XLA) in humans, a disease that is characterized by impaired B cell development[27]. Several of the extended regulatory spine residues identified in this study are mutated in XLA patients[27]. In addition to nonsense mutations at Btk M449, M450 and W395, which cause premature truncation of the protein, there are also missense mutations at Btk M450, F450 and D579 (M450I, F450S and D579V). These mutations likely destabilize the regulatory spine and prevent Btk kinase activation.  Activation loop tyrosine phosphorylation has long been known to influence kinase activity. In addition to priming the kinase domain for the assembly of the regulatory spine, phosphorylation on the conserved Tyr of the activation loop is also thought to stabilize the substrate-binding site of the activation loop[1]. However, our work has shown that in the context of the activating gatekeeper Btk T474M mutation, loss of Y551 phosphorylation (by mutation to phenylalanine) has no affect on substrate phosphorylation. Pre-assembly of the regulatory spine (mutation of T474M) removes any requirement for activation loop phosphorylation. These results are consistent with earlier studies that focused on phosphorylation of Y551 in Btk[25]. That earlier work did show that phosphorylation on Btk Y551 significantly alters catalytic activity of Btk; kcat of the Btk Y551F mutant is greatly diminished compared to wild-type Btk presumably due to loss of the trigger for regulatory spine assembly. However, the affinity for a peptide substrate is unaffected by Y551 phosphorylation; Km of a peptide substrate for Btk Y551F mutant is the same as wild-type Btk. As well, an alternative direct substrate docking mechanism has been identified for the Tec kinases, in which an SH2 domain within the substrate docks onto the kinase domain outside of the active site and facilitates substrate phosphorylation[28; 29; 30]. It is therefore possible that the peptide substrate-binding region on the activation loop may only play a minor role in substrate recognition in Tec kinases and that activation loop phosphorylation plays one significant role: initiation of spine assembly by altering electrostatic interactions in and around the active site.  We have previously shown that the regulation of Tec family kinases is similar to that of Csk (C-terminal Src kinase)[14]. Unlike the Src and Abl family of kinases, the SH3 and SH2 regulatory domains of Csk and Tec kinases positively influence the catalytic activity of the kinase domain[14; 16]. Like Itk and Btk, the isolated kinase domain of Csk has poor catalytic activity[16; 31; 32]. In order to define the elements within the Csk kinase domain that are responsible for the poor catalytic activity, a mutagenesis study was carried out in which Src:Csk kinase domain chimeras were created[33]. A Src:Csk chimeric kinase domain where the N-terminal lobe of Csk was replaced with the N-terminal lobe of Src was shown to exhibit activity comparable to that of full-length Csk[33]. Further mutagenesis of the N-terminal lobe, identified the C-helix and the \\u03b2 turn between the \\u03b24 and \\u03b25 strands (the \\u2018top\\u2019 of the regulatory spine) as key structural elements that are required for the activity of the isolated kinase domain of Csk[33].  There is remarkable correlation between the elements that are identified in the Csk mutagenesis study and the extended regulatory spine of Tec kinases identified in this study. Since the Csk and Tec family kinases are both positively regulated by domains outside of the kinase domain, it is not surprising that they would share the key determinants of catalytic activity. Indeed, as suggested in the original identification of the kinase regulatory spine[9], this feature likely plays a critical role in all active kinases. Our work confirms this notion for the Tec family kinases, and illustrates that extension of the regulatory spine by two residues couples this regulatory feature to allosteric events occurring outside of the kinase domain. Moreover, our data suggest that despite a high degree of sequence conservation, the five core regulatory spine residues do not assemble into the active conformation in the context of the isolated Itk or Btk kinase domains. The energetics of spine assembly seem to be tailored for each specific regulatory environment; the Tec kinase domains are maintained in their inactive state (spine assembly does not occur even when the activation loop is phosphorylated) until additional regions of the protein (in this case the SH2-linker) are present to drive complete spine assembly forward and fully activate the kinase.\",\n          \"More than 50% of transitional cell carcinomas of the bladder show loss of heterozygosity of a region spanning the TSC1 locus at 9q34 and mutations of TSC1 have been identified in 14.5% of tumours. These comprise nonsense mutations, splicing mutations, small deletions and missense mutations. Missense mutations are only rarely found in the germline in TSC disease. Therefore, we have examined six somatic missense mutations found in bladder cancer to determine whether these result in loss of function. We describe loss of function via distinct mechanisms. Five mutations caused mutually exclusive defects at mRNA and protein levels. Of these, two mutations caused pre-mRNA splicing errors that were predicted to result in premature protein truncation and three resulted in markedly reduced stability of exogenous TSC1 protein. Primary tumours with aberrant TSC1 pre-mRNA splicing were confirmed as negative for TSC1 expression by immunohistochemistry. Expression was also significantly reduced in a tumour with a TSC1 missense mutation resulting in diminished protein half-life. A single TSC1 missense mutation identified in a tumour with retained heterozygosity of the TSC1 region on chromosome 9 caused an apparently TSC2- and mTOR-independent localization defect of the mutant protein. We conclude that although TSC1 missense mutations do not play a major role in causation of TSC disease, they represent a significant proportion of somatic loss of function mutations in bladder cancer.  Go to: INTRODUCTION Tuberous Sclerosis Complex (TSC) is an autosomal dominant tumour suppressor gene syndrome with an incidence of 1 in 6000\\u201310 000 births. TSC is characterized by the development of benign growths, called hamartomas, in the kidneys, heart, brain and skin, and patients present clinically with a variety of developmental disorders (1). TSC is caused by mutations affecting either of the tumour suppressor genes TSC1 or TSC2. TSC1 on chromosome 9q34 encodes hamartin (2) and TSC2 on chromosome 16p13.3 encodes tuberin (3). Approximately half of large TSC families show linkage to 9q34 and half to 16p13.3 (4\\u20136). Tumour development in TSC patients is thought to occur as the result of a somatic \\u2018second-hit\\u2019 in either TSC1 or TSC2, according to Knudson\\u2019s tumour suppressor model (7). Loss of heterozygosity (LOH) of TSC1 or TSC2 has been reported in some TSC hamartomas, such as renal angiomyolipomas. However, loss of the wild-type allele in brain lesions is rare, suggesting the possibility of tissue-specific haploinsufficiency of TSC genes (8\\u201310).  Co-localization and co-immunoprecipitation of TSC1 and TSC2 in mammalian cells (11,12) and direct binding in yeast two-hybrid assays provide a tentative explanation for the similar disease phenotype in TSC patients with mutations in either TSC1 or TSC2 genes (2,13). Functionally, the TSC1/TSC2 complex is positioned at the centre of multiple growth signalling pathways and is a key integrator of signals controlling protein translation and cell growth (14). Activation of the TSC1/TSC2 complex in growth-limiting conditions attenuates signalling through mTOR via specific GTPase activating protein (GAP) activity of TSC2 towards RHEB (15,16).  While epithelial malignancy is not a common feature of TSC, studies in this laboratory and others have implicated loss of function of TSC1 in bladder tumorigenesis (17\\u201319). Loss of heterozygosity (LOH) for markers on chromosome 9 is observed in more than 50% of bladder tumours of all grades and stages (20) and sub-chromosomal LOH analyses have identified the TSC1 locus at 9q34 as a common critical region of deletion between markers D9S149 and D9S66 (19,21\\u201323). To date, we have screened 154 bladder tumours by fluorescent single strand conformation polymorphism (F-SSCP) analysis and direct sequencing, and found an overall mutation frequency of 14.5%. The mutation spectrum comprises nonsense (35%), missense (26%), frameshift (26%), in-frame deletions (3%) and splicing (10%) mutations (24) (Platt et al., in preparation). In all cases but one, TSC1 missense mutations were tumour-specific somatic events. TSC1 is the only gene on 9q that has been found to be mutated in bladder tumours, and may therefore be the critical gene on this chromosome arm implicated in >50% of all bladder tumours.  Missense mutations of TSC1 have not routinely been confirmed as functionally inactivating in TSC disease, though two recent reports provide evidence that in a few cases these are likely to be disease-causing (25,26). Here we sought to determine whether the TSC1 missense mutations identified in bladder tumours constituted inactivating mutations. We anticipated that discrete amino acid changes of mutant proteins might allow the identification of functionally important residues. Wild-type and mutant TSC1 constructs were retrovirally delivered into TSC1-null bladder tumour cell lines and functionally characterized. All somatic TSC1 missense mutations perturbed TSC1 function by causing aberrant splicing, protein instability or protein mislocalization. Defects were confirmed in primary tumours by RT\\u2013PCR analysis of mutant transcripts and immunohistochemical analysis.  Go to: RESULTS Missense mutations of TSC1 identified in bladder tumours  Previously, we identified 8 mutations including 2 missense mutations in a series of 62 bladder tumours (24). Screening of an additional 92 tumours (Platt et al., in preparation) identified an additional 15 mutations of which 4 were missense mutations. In total, therefore, 6 missense mutations have been identified in 154 tumours, representing 26% of all mutations found (Table 1). Mutations were determined as tumour-specific by genotyping of paired tumour and blood samples. 1250C>T (Thr417Ile) was previously described in the germline in TSC disease in Japanese patients, but was not confirmed as causative of TSC (27,28). 1250C>T was also identified in our laboratory in the patient\\u2019s constitutional DNA, and the tumour sample retained heterozygosity for microsatellite markers at the TSC1 gene locus (24). Threonine 417 was previously identified as a site of CDK1-dependent phosphorylation (29). To determine the biological significance of this variant, 1250C>T (Thr417Ile) was characterized here alongside tumour-specific missense mutations. Five of six other tumours with TSC1 missense mutations showed LOH for markers at the TSC1 locus (Table 1). There was no relationship between mutation and tumour stage or grade. Missense mutations were predominantly N-terminal but did not localize to a common functional domain (Fig. 1).  Figure 1. Figure 1. Positions of amino acid substitutions in relation to described functional domains of hamartin. Table 1. Table 1. TSC1 missense variants identified in bladder tumours To assess possible functional implications of amino acid substitutions, conservation of missense mutant residues was determined in TSC1 orthologs. His68, Phe158 and Phe216 were conserved among Rattus, Mus, Drosophila, Fugu and Gallus orthologs. His105, His206 and Thr417 differed only in Drosophila, which shares 31% identity with human TSC1. Ser35 differed in Drosophila, Fugu and Gallus orthologs. According to the BLOSUM62 scoring matrix (30), His68Arg and His105Arg substitutions are considered conservative. Ser35Cys, His206Asp and Thr417Ile are less conservative and Phe158Cys and Phe216Asp are least conservative (Table 1).  Re-expression of missense mutant TSC1 proteins in TSC1-null urothelial cells  We anticipated that missense mutations might lead to loss of function at the amino acid level. Thus, our initial approach was to express wild-type and missense mutant TSC1 cDNAs in TSC1-null bladder cell lines and investigate function of the mutant proteins. Missense mutant proteins were C-terminally FLAG- or GFP-tagged, expressed in 97-1 and HCV29 cells and characterized for TSC2 binding activity, and mTOR suppressive activity in nutrient-starved conditions.  Expression of mutant proteins was investigated by immunoblotting following retroviral transduction and selection of neomycin resistant mass cell populations. Expression levels of mutant proteins were markedly different; Ser35Cys, His105Arg, Phe216Asp and Thr417Ile were expressed at high levels similar to wild-type protein, whereas His68Arg and Phe158Cys were much reduced and His206Asp was not detected (Fig. 2A). Repeated independent infections of these two cell lines and of telomerase-immortalized normal human urothelial cells (TERT-NHUC) and assessment of mass populations of cells following selection resulted in entirely reproducible expression levels of all mutant proteins (data not shown). The transcription of missense RNA was confirmed by real time RT\\u2013PCR and uniform levels of wild-type and missense transcripts were detected (Fig. 2B). The consistently reduced or absent protein expression of His68Arg and Phe158Cys and His206Asp missense mutant forms despite the presence of RNA expression suggested possible effects of these missense changes on protein stability.  Figure 2. Figure 2. (A) Immunoblot showing levels of wild-type and mutant TSC1-FLAG proteins and endogenous TSC2 in 97-1 cell lines. (B) Measurement of TSC1 RNA levels by real time RT\\u2013PCR analysis of wild-type and mutant TSC1-FLAG mRNA transcript levels in 97-1 cell ... Mutant TSC1 proteins retain interaction with TSC2  The interaction between TSC1 and TSC2 appears to be important in maintaining the stability of each of the proteins. TSC1/TSC2 binding specifically augments TSC2 expression by limiting its ubiquitination (31,32). TSC2 GAP activity towards RHEB defines TSC1/2 mediated control of mTOR signalling, and phosphorylation of downstream effectors of mTOR is constitutive and refractory to amino acid withdrawal in cells lacking TSC1 or TSC2 (33\\u201335). Where expressed, missense mutant TSC1 proteins stabilized TSC2 levels in 97-1 cells (Fig. 2A) and co-immunoprecipitated with it (Fig. 3A). However, our data cannot exclude minor effects on interaction or TSC2 stability.  Figure 3. Figure 3. (A) Immunoblot showing TSC1 and TSC2 in 97-1 Neo control and FLAG-tagged wild-type and mutant TSC1 cell line lysates immunoprecipitated with anti-TSC1 and non-specific mouse IgM antibodies. (B) Immunoblot showing expression levels of GFP-tagged wild-type ... While His206Asp-FLAG was undetected in transduced 97-1 cells, we were able to achieve low-level expression of His206Asp-GFP in HCV29 cells, suggesting some stabilizing effect of the GFP tag. TSC2 co-immunoprecipitation and S6 phosphorylation assays were not accurately quantitative, but expression of all missense mutant proteins in TSC1-null cells reduced S6 phosphorylation in amino acid starved conditions, relative to vector alone (Fig. 3B), indicating that none of these mutant forms of TSC1 abolish interaction with TSC2.  Aberrant RNA splicing caused by TSC1 missense mutations  High-level expression and TSC2 binding activity of TSC1 Ser35Cys, His105Arg and Phe216Ala proteins suggested no functional effect of amino acid substitutions caused by 104C>G, 314A>G and 648T>A missense mutations. However, the expression of proteins from exogenous missense cDNAs did not allow the assessment of possible effects of mutations at the pre-mRNA level. We speculated that TSC1 missense mutations may cause loss of function through introduction of splicing errors in mutant transcripts in vivo. Therefore, wild-type and missense mutant TSC1 pre-mRNA sequences were screened for effects of mutations on splice site definition using the neural network algorithm (http://www.fruitfly.org/seq_tools/splice.html).  Interestingly, significant differences were seen between splice site scores of wild-type and 104C>G and 314A>G TSC1 transcripts. The 104C>G mutation, positioned 3 bp upstream of the TSC1 exon 3/4 junction, resulted in a reduced exon 3/4 splice motif score. The 314A>G mutation resulted in the introduction of a high-scoring 5\\u2032 donor splice site, immediately upstream of the A/G transversion, by generation of a novel consensus splice motif. No differences in splice site scores were observed between wild-type and 648T>A or 1250C>T transcripts or between wild-type and mutant transcripts that generated low exogenous protein expression.  To examine splicing of missense mutant TSC1 transcripts, fragments spanning the mutation site and flanking intron/exon junctions were amplified from respective tumour cDNAs by RT\\u2013PCR and sequenced. An RT\\u2013PCR product reproducibly amplified from cDNA of the tumour containing 104C>G showed increased electrophoretic mobility relative to a control fragment amplified from TERT-NHUC cDNA (Fig. 4A). Sequencing of the tumour product revealed use of a cryptic 5\\u2032 donor site in exon 4, the adjacent downstream exon to the 104C>G mutant exon (Supplementary Material, Fig. S1). The reduced splice motif score at the exon 3/4 junction was shown to have no effect on exon 3/4 splicing. If translated, this 104C>G mutant transcript would generate 26 new amino acids from an alternate reading frame and truncate prematurely at residue 79.  Figure 4. Figure 4. Semi quantitative RT\\u2013PCR analysis of TSC1 RNA in normal urothelial cells and TSC1 mutant tumour samples, showing increased electrophoretic mobility of RT\\u2013PCR products amplified from cDNA from tumours containing TSC1 104C>G (A) ... A faster migrating band was also reproducibly amplified from cDNA of the tumour containing 314A>G (Fig. 4B), compared to control TERT-NHUC cDNA. Sequence analysis of PCR products revealed a 50-nucleotide deletion in the transcript generated from the missense allele. The 314A>G mutation created a de novo 5\\u2032 donor splice site immediately upstream of 314A>G and resulted in the splicing of 50 nucleotides from the 3\\u2032 end of exon 5. The effect seen at the mRNA level was entirely consistent with the prediction made by in silico analysis. If translated, the 314A>G mutant transcript is expected to generate two new C-terminal amino acids and truncate prematurely at residue 107.  Exon skipping through introduction of a de novo 5\\u2032 splice site by point mutation is a relatively well-described mechanism of aberrant pre-mRNA splicing (36\\u201338). No normal transcript was amplified from cDNA from tumours containing TSC1 104C>G or 314A>G mutations, suggesting that all splicing occurs via the novel sites. The abundance of missense transcripts appeared lower than normal transcript by semi-quantitative RT\\u2013PCR, when standardized to HPRT. However, by real time RT\\u2013PCR, TSC1 transcript abundance was relatively higher in tumours with TSC1 missense mutations causing transcript splicing or protein stability defects, than in uncultured or cultured TERT-NHUC, when standardized to SDHA (data not shown).  Missense mutant TSC1 proteins exhibit reduced protein stability  The demonstration of consistently low-level exogenous protein expression, despite uniform mRNA transcript abundance, suggested diminished stability of TSC1 His68Arg, Phe158Cys and His206Asp mutant proteins. By inhibiting protein synthesis with cycloheximide, we determined that His68Arg, Phe158Cys and His206Asp mutant proteins were turned-over relatively faster than wild-type TSC1, in a proteasome-dependent manner (Fig. 5 A and B and data not shown). Moreover, by 35S labelling, we were able to show that His68Arg, Phe158Cys and His206Asp proteins had markedly shorter half-lives than wild-type TSC1 (Fig. 5C).  Figure 5. Figure 5. (A) Immunoblot showing turnover of wild-type and S35C and H68R missense TSC1 proteins in cycloheximide (CHX)-treated cells. Cells were cultured in full growth medium supplemented with 100 \\u00b5g/ml CHX or DMSO vehicle alone, and lysed at time-points ... Phe216Ala TSC1 protein shows altered localization  Previous studies have described a granular, cytoplasmic localization of endogenous TSC1 in vitro and in vivo and also of overexpressed TSC1 in COS-7 cells (39\\u201341). Localization of monomeric TSC1 and of TSC1 complexed with TSC2 is likely influenced by culture conditions given that components of the Akt-mTOR signalling cascade are membrane localized when activated. In addition, TSC2 is reported to shuttle into the nucleus in a cell cycle and phosphorylation-dependent manner (42\\u201344).  It was anticipated that the characterization of discrete missense amino acid changes may offer insight into potentially mTOR-independent or bladder-specific functions of TSC1. Our results indicated that TSC1 104C>G, 203A>G, 314A>G, 473T>G and 616C>G missense mutations cause loss of function by generic mechanisms of altered message or reduced protein stability. However, sequencing of RT\\u2013PCR products amplified from cDNA from the tumour containing the TSC1 648T>A mutation showed no altered splicing of the mutant transcript. Also, at the protein level, Phe216Ala was expressed at high levels similar to wild-type TSC1. The substitution of a phenylalanine residue for an alanine residue constitutes the loss of a high-molecular weight, hydrophobic benzyl group and is assigned a non-conservative \\u20132 BLOSUM62 score. The lack of effect on splicing or protein stability raised the possibility that this mutation may cause a defect in TSC1 protein function per se.  We determined the localization of GFP-tagged mutant proteins in nutrient replete and deficient conditions. Wild-type, Phe216Ala, Ser35Cys, His68Arg and Thr417Ile proteins were compared. In complete growth medium, wild-type TSC1 and Ser35Cys, His68Arg and Thr417Ile mutant proteins showed diffuse punctate cytoplasmic distribution with distinct cytoplasmic foci. In amino acid deficient medium, these proteins became localized almost exclusively to large cytoplasmic bodies. Intriguingly, the Phe216Ala substitution markedly altered the localization of TSC1; Phe216Ala was exclusively cytoplasmic and diffuse in full growth medium and did not redistribute to discrete bodies in amino acid deficient conditions (Fig. 6). However, it was shown to stabilize and to co-immunoprecipitate with endogenous TSC2, and to attenuate growth signalling through mTOR in starved conditions (Fig. 3B). These observations were reproduced in TSC1-transduced 97-1 and HCV29 cell lines and TERT-NHUC and suggest that localization of TSC1 to cytoplasmic foci is not a requirement for TSC2 binding or negative regulation of mTOR. Also, defective localization of Phe216Ala is likely TSC2- and mTOR-independent, possibly suggesting an independent function of TSC1 at cytoplasmic foci.  Figure 6. Figure 6. Localization of wild-type TSC1 (A) and TSC1 Ser35Cys (B), Phe216Ala (C) and Thr417Ile (D) mutant proteins in amino acid-starved 97-1 cells. Cells were cultured in full growth medium on highly optically clear microscopy dishes to sub-confluence and amino ... Unlike the other five tumours with missense mutations, the tumour containing the TSC1 648T>A mutation did not show LOH of 9q, as confirmed by analysis of pure microdissected tumour cell populations (data not shown). Both normal and mutant alleles were detected by sequencing of tumour cDNA.  TSC1 protein expression in bladder tumour tissues  To examine the effects of pre-mRNA splicing defects and protein stability defects caused by missense mutations at the tumour level, TSC1 mutant and wild-type tumours were screened for expression of TSC1 protein by immunohistochemistry. Sensitivity and specificity of a rabbit monoclonal anti-TSC1 antibody was confirmed by staining of paraffin-embedded pellets of TSC1-null HCV29 cells and HCV29 in which wild-type TSC1 had been re-expressed (Fig. 7A and D). Normal ureter showed strong cytoplasmic TSC1 expression in the urothelium (Fig. 7E). Tumours with wild-type TSC1 showed strong cytoplasmic staining (Fig. 7F), and a tumour with a TSC1 73\\u201377\\u0394 5 small deletion causing premature truncation at residue 27, showed no TSC1 expression (Fig. 7C). Both tumours with homozygous TSC1 missense mutations causing pre-mRNA splicing defects were negative for TSC1 expression (Fig. 7G and H). Figure 7H shows normal urothelium with strong TSC1 staining adjacent to immunonegative TSC1 314A>G mutant tumour cells. The 648T>A (Phe216Ala) mutant tumour showed moderate TSC1 expression consistent with normal stability of the Phe216Ala protein and retention of chromosome 9q heterozygosity in the tumour (Fig. 7I). Of the three missense mutations causing reduced protein stability (TSC1 203A>G, 473T>G and 616C>G), tumour material was available only for the 616C>G (His206Asp) mutant sample. The 616C>G mutation was identified from tumour material resected in 2003, and TSC1 expression was shown to be low in three tumour resections from this patient in successive years (Fig. 7J\\u2013L).  Figure 7. Figure 7. TSC1 immunostaining of HCV29 Neo (A) and TSC1 (D) cell pellets, normal ureter (negative (B) and positive (E) antibody controls) and TSC1 73\\u201377\\u0394 5 (C) and TSC1 wild-type (F) bladder tumours. TSC1 staining of TSC1 missense mutant bladder ... Go to: DISCUSSION We have demonstrated that bladder tumour-derived TSC1 missense mutations result in loss of TSC1 function and that this occurs via distinct mechanisms. An overall TSC1 mutation frequency of \\u223c14.5% is found in bladder cancer (24) (Platt et al., in preparation) and missense mutations comprise 26% of mutations found to date. The identification of deleterious missense mutations in bladder tumours indicates a causative role of loss of TSC1 function via this mechanism in bladder tumorigenesis. The vast majority of TSC1 mutations in TSC disease are predicted to be protein truncating in nature, and no significant genotype/phenotype correlations have been observed (2,5,45\\u201348). Non-chain terminating TSC1 mutations (missense or in-frame deletions) are rare in TSC disease (27,47,49,50) (www.LOVD.nl/TSC1). Although missense mutations have been reported previously, most have not been confirmed as disease-causing. Several have been revealed as rare polymorphisms or associated with other nonsense mutations in the same patient (46,50). Others have been disregarded as potentially disease-causing on the basis of conservative amino acid changes, or have not been characterized further (27,51). However, a recent publication by Jansen et al. (26) that identified three missense mutations (L916R, M224R and E412V) in affected individuals reported functional analyses on two of these. It was reported that E412V affected RNA splicing. Interestingly, in transfections of constructs of wild-type TSC1 and the M224R variant, the latter showed lower levels of protein expression that were unable to completely suppress S6 phosphorylation, results similar to those described here for missense variants with reduced protein half-life.  We have shown that two of six bladder tumour-derived missense mutations result in pre-mRNA splicing defects, three lead to reduced stability of mutant proteins and intriguingly, one mutation causes a TSC2- and mTOR-independent localization defect of the mutant protein. Wild-type TSC1 showed a granular cytoplasmic distribution in cells in full growth medium and became localized almost exclusively to large cytoplasmic bodies in amino acid starved cells. In contrast, TSC1 was cytoplasmic and diffuse in serum-starved cells. Preliminary results indicated that TSC1 bodies were dynamic, non-aggresomal structures (data not shown). Interestingly, the Phe216Ala substitution abolished the localization of TSC1 to these bodies. As expected from its position outside the recognized TSC2 binding domain, TSC1 Phe216Ala retained TSC2 binding activity and mTOR suppressive activity in amino acid starved conditions, suggesting that localization of TSC1 to cytoplasmic foci was not a requirement for TSC2 binding or for suppression of mTOR signalling. This may also suggest that TSC1 has a separable function at cytoplasmic foci that is independent of TSC2 and independent of negative regulation of mTOR by TSC1/TSC2. However, it is not clear whether the formation of intensely staining bodies in conditions of over-expression of TSC1 and/or TSC2 (39,52) is relevant to the normal physiological condition. Further investigation of the function of Phe216Ala is now required, ideally at levels of expression that are closer to normal and in both TSC1-null cells and those expressing endogenous wild-type TSC1.  Although the missense variant 1250C>T (T417I) was found in the patient\\u2019s germline and this patient showed no symptoms of TSC, we included this variant in our analyses as threonine 417 has previously been identified as a site of CDK1-dependent phosphorylation (29). We found no evidence for a functional defect and conclude that this represents a rare polymorphism. As this patient was Japanese, as were both TSC patients in whom this variant was reported previously (27,28), this rare variant may be confined to the Japanese population.  Evidence suggests that cis-acting mutations affecting splicing of some tumour suppressor genes can have causal roles in tumour initiation and progression (53). Predicted and confirmed effects of TSC-related and bladder tumour-derived TSC1 missense changes on pre-mRNA splicing reported here underscore the importance of using RNA-based techniques, together with conventional mutation detection methods, to effectively identify disease-causing mutations. One of the mutations that deleteriously affected splicing (104C>G) had an unexpected effect. The use of a cryptic 5\\u2032 donor site in an adjacent downstream exon is a non-conventional splicing event; disease-associated mutations at splice junctions typically result in skipping of the mutant exon (38,54,55). The 104C>G mutation was shown to reduce the exon 3 5\\u2032 splice site motif score and to disrupt an enhancer sequence spanning exon 3/4. 104C>G also disrupts a U1 snRNA-binding motif, spanning the 5\\u2032 splice site from position \\u20133 to +8 (Supplementary Material, Fig. S1). In combination with a normally low-scoring exon 4 3\\u2032 acceptor site, these factors may result in reduced splicing efficiency of introns 3\\u20134. Spliceosome assembly is directed by juxtaposed splicing elements, and were introns 4\\u20135 to splice first, splicing machinery at the 3\\u2032 acceptor site of exon 4 may stimulate recruitment of U1 snRNA to possible binding sites in exon 4, and result in use of a cryptic 5\\u2032 donor site.  In bladder tumours as in TSC disease, nonsense, deletion and frameshift TSC1 mutations result in premature protein truncation and loss of protein function. We have now shown that missense mutations cause loss of function by aberrant splicing or reduced protein stability. The identification of tumour-specific TSC1 mutations in the context of chromosome 9 LOH argues strongly for a direct role of loss of TSC1 function in the aetiology of these tumours. These data are consistent with TSC1 acting as a tumour suppressor gene in bladder cancer in accordance with Knudsen\\u2019s two-hit hypothesis. Biallelic inactivation of TSC1 or TSC2 may not be necessarily required in some TSC-related tumours (9). The discrepancy between frequency of TSC1 mutation and frequency of LOH in the TSC1 gene region in bladder tumours suggests that haploinsufficiency of TSC1 may contribute to tumour development in some cases. LOH of chromosome 9 is a particularly frequent event in bladder cancer and to date, TSC1 is the only gene on 9q found to be mutated in bladder tumours. The contribution of partial loss of TSC1 to clonal expansion of tumour cells with 9q LOH is unknown, and haploinsufficiency remains a possibility in those bladder tumours with 9q LOH and no TSC1 mutation. LOH at the TSC1 locus may accompany another event driving loss of chromosome 9. Deletion of the CDKN2A locus at 9p21, which occurs in up to 50% of bladder tumours (56,57), or of an as yet unidentified chromosome 9 tumour suppressor gene, may constitute such a driving force. Loss of one copy of TSC1 may therefore be an advantageous gratuitous hit or \\u2018passenger event\\u2019 (58).  TSC is not a cancer prone syndrome and TSC lesions very rarely progress to malignancy. That TSC patients do not have an increased risk of developing bladder cancer, or other sporadic cancers, may be explained by the requirement of cumulative genetic insults and the typically late onset of malignant disease. The timing and order of initiating and subsequent genetic events in hamartoma and bladder tumour development is likely to be critical in determining malignant potential. Why TSC1 appears to be involved in bladder cancer and not other epithelial cancers is an unresolved question.  Frequent LOH of 9q is found in other tumour types, including ovarian carcinoma, gallbladder carcinoma, nasopharyngeal carcinoma and non-small cell lung cancer (59\\u201365). However, reports of TSC1 and TSC2 mutation status in sporadic tumours other than bladder are very few. No mutations were observed in sporadic glial and glioneuronal tumours or renal cell carcinomas (66,67). Fifty-three and 39% of lung adenocarcinomas and precursor lesions, respectively, were found to have LOH of 9q (68). Subsequently, a screen of 47 lung adenocarcinomas identified three confirmed TSC1 mutations. However, LOH and mutations were not detected simultaneously (69).  In ovarian carcinoma, gallbladder carcinoma, nasopharyngeal carcinoma and lung adenocarcinoma, LOH or deletion is described in both TSC gene regions (63,68,70,71). If loss of TSC1/2 complex function is the pathogenic effect of loss of TSC1, it may be expected that bladder tumours would also show loss of TSC2 function. In an LOH screen of 16p, we have detected LOH for markers at the TSC2 locus in 16% of bladder tumours (Platt et al., in preparation). It will be important to screen for mutations in the retained copy of TSC2 in cases with LOH to determine whether there is also an involvement of TSC2 or whether there is an important independent role of TSC1 in urothelial cells.  Abstract We have surveyed the mutations of TSC1 and TSC2 from 38 (25 sporadic, 11 familial, and 2 unknown) Japanese patients with tuberous sclerosis complex. In 23 of 38 subjects, we detected 18 new mutations in addition to 4 mutations that had been previously reported. We also found 3 new polymorphisms. The mutations were not clustered on a particular exon in either of the genes. Seven TSC1 mutations found in 3 familial and 4 sporadic cases were on the exons (3 missense, 2 nonsense point mutations, a 1-base insertion, and a 2-bp deletion). Fifteen TSC2 mutations were found in 5 familial cases, 10 sporadic cases, and 1 unknown case. The 12 mutations were on the exons (8 missense, 1 nonsense point mutations, a 1-bp insertion, a 5- bp deletion, and a 4-bp replacement) and 3 point mutations were on the exon\\u2013intron junctions. Although the patients with TSC2 mutations tend to exhibit relatively severe mental retardation in comparison to those with TSC1 mutations, a genotype\\u2013phenotype correlation could not yet be established. The widespread distribution of TSC1/TSC2 mutations hinders the development of a simple diagnostic test, and the identification of individual mutations does not provide the prediction of prognosis. Key words Tuberous sclerosis complex \\u2022 TSC1 gene \\u2022 TSC2 gene \\u2022 Hamartin \\u2022 Tuberin \\u2022 Mutation Introduction Tuberous sclerosis complex (TSC) is an autosomal dominant disease with an estimated prevalence of 1 in 10,000 in the Caucasian population and 1 in 31,000 in the Japanese population (Ohno et al. 1981; Gomez 1988). Multiple organs are affected in TSC, including the central nervous system (mental retardation, epilepsy), skin (facial angiofibroma, shagreen patches, hypopigmented macules, periungual fibromas), heart (cardiac rhabdomyomas), and kidney (cysts, angiomyolipoma). There are two causative genes of TSC. The TSC1 (hamartin) is located on 9q34 and consists of 23 exons. The TSC2 (tuberin) is located on 16p13.3 and consists of 41 exons (European Chromosome 16 Tuberous Sclerosis Consortium 1993; Van Slegtenhorst et al. 1997). Loss of heterozygosity (LOH) of either TSC1 or TSC2 in the affected tissues indicates that both act as a tumor suppressor (Green et al. 1994; Henske et al. 1995, 1996; Sepp et al. 1996). Multiple isotypes of tuberin are generated by alternative splicing at two different exons (exon 25 and 31) of TSC2 (Xiao et al. 1997). Since the identification of the TSC1/ TSC2 genes, the mutations in Caucasian TSC patients have been extensively surveyed (Kumar et al. 1995 a,b; Vrtel et al. 1996; Wilson et al. 1996; Bajek et al. 1997; Jones et al. 1997; Maheshwar et al. 1997; Van Slegtenhorst et al. 1997; Au et al. 1997, 1998). The mutations in Japanese TSC patients, however, have remained unknown. In the present study, we surveyed the mutations in Japanese patients and assessed the relationship between the mutations and clinical symptoms. Materials and methods Patients and sample collection Thirty-eight (25 sporadic, 11 familial, and 2 unknown) Japanese TSC patients were clinically diagnosed according to the criteria (Gomez 1988). In some cases, ultrasound renogra392 phy, MRI, and CT brain scan gave supportive evidence for the diagnosis. Genomic DNAs were prepared from the blood samples as described (Nanba et al. 1995). Healthy Japanese volunteer DNA as normal controls were extracted by the Nucleic Acid Purification System (MagExtractor MFX-2000; Toyobo, Osaka, Japan) from 100\\u00edl blood. Polymerase chain reaction-single strand conformation polymorphism (PCR-SSCP) analysis All coding exons of TSC1 and TSC2 were analyzed for every patient. Sets of PCR primers for the 41 exons of TSC2 were designed based on the TSC2 sequence, except for the primers for exon 13 and 25, which were kindly obtained by Dr. J.R. Sampson at the University of Wales, Cardiff, UK (Maheshwar et al. 1996) (Table 1). The primer information for the 21 coding exons of the TSC1 gene was obtained from Dr. O Hino in the Department of Experimental Pathology, Cancer Institute, Tokyo, Japan, and Dr. D.J. Kwiatkowski in Brigham and Womens\\u2019 Hospital, Boston, USA, except for the primers for exon 9 (CACTGAGTTGACACTCTGAAG and CTGAACTAAGTCTTACTCCAG), which were originally designed. PCR contained 25ng of the genomic DNA as a template, 10pmol of each primer, 0.25\\u00edM of each dNTP, 0.1 unit of Ampli Taq Gold (Perkin-Elmer, Norwalk, CT, USA), and a buffer in a total volume of 5\\u00edl. Cycling conditions were as follows: an initial denaturation at 95\\u00b0C for 10 min followed by 30 cycles of denaturation at 95\\u00b0C for 1 min, annealing at 65\\u00b0C for 1min and elongation at 72\\u00b0C for 1min, with a final extension at 72\\u00b0C for 5min. All the PCR products were analyzed on 1.5% agarose gel. SSCP analysis was performed as described (Orita et al. 1989; Yuasa et al. 1997). Using a minigel (10 3 10 cm), the samples were analyzed under four different electrophoresis conditions, in a combination of two sets of gel mixtures (12% polyacrylamide gel with or without 5% glycerol) and two temperature conditions (4\\u00b0C or 22\\u00b0C). Table 1. Primers for TSC2 gene analysis Exon Product size (bp) 1 F: AAGGTTATGCCCACCAGAGA; R: AGACACAGGTAGCTCACTCA 296 2 F: CACTGGCCCCTTTTTCTTCT; R: CTCTAGTAGCTCAACTGGAT 150 3 F: CTTGGAGAGCACATCCTCAC; R: CAGCCTGACGTCACCCATCC 195 4 F: CTGATCCTGTGGCTTTTGTC; R: AGAAACCTCCAACCCAAGGT 199 5 F: TCGCAAACTGCGCGACTTCT; R: CAGAACAGGATCTCAATTCT 173 6 F: GACTGAGCTCGGTGCTCCCT; R: TGAAATGGGCCGGCCCTGGG 109 7 F: GAAGGAGGTGGGAAGGAAG; R: CATCCATGTGCTCTCAGGAA 278 8 F: CTTTGGGAGGAGATGGTGG; R: ATGTGTGCGTTGGTAAGGGA 286 9 F: CGTCTCTCTGGGGAACACTT; R: TCACTGCACACAGAAACCGC 176 10 F: TGCTGGCCGGGCTCGTGTTC; R: TGCTCCCCGGAGCTCCTGCC 198 11 F: GCCTGTGTCATCGTGCCTGG; R: AAAGGCCCCACCCAGGAGGG 194 12 F: CAACACCGGCTCTTCTTTTG; R: AGGCAAGGCTACAGAGGACC 153 13a F: GGGCTGTGGCGGGCACTC; R: AGAGGTCCGGCAGACCGAAGTC 245 14 F: TTGGCCTCCCTTGTGCCTGT; R: GCCCCGGGTACAACGGCTCT 203 15 F: GACTCAGAACCATGAGCCTG; R: AACAGACTCCAACACAACGC 251 16 F: CGTGGTGAGCTGCGTCCTCT; R: ACTGCGTGCGCAACCCCAGC 178 17 F: GGCTTTCACCATCCTCTTCC; R: TCATGGGCCGTGGGCAGGTC 160 18 F: GCCTCAGCTGCTTCTCTTGC; R: AACGTGGGACGGATGGTCCC 216 19 F: CCTGACGCCTCCTCTCCTCG; R: CCCAACCCCAGGCCGGCATG 185 20 F: TCATGCCTGGATTTGGTCAT; R: TCAAGCTCGCCTGCTCTGAC 187 21 F: GGGCCTGAGGTGTCCTGTCT; R: AGGTGCAGTGCTGCAGGGCG 246 22 F: GGCTCCCTGACCACCTCTCC; R: ACAGGGCCTGGGGCGACCAC 148 23 F: GTGCCAGCCCCCTTCTCATC; R: ACAGCCTTCACCGCCCTGAG 152 24 F: TCACTGTCTGGGTGTGCTC; R: TGCCCTGCCTCACATACTC 260 25a F: CCCTCCACTGGCTTGTTCTCC; R: CGGGCAAGACGATGAGGTCAT \\u2014b 26 F: CCTGGTCACGGCCTCTCCCT; R: CCAGCCCTGTAGTGCCGCCT 213 27 F: GGTTTCACGCTCCCTGTCTT; R: GGAGGAAAGGAAGGTGCAGT 200 28 F: ACCCTGTGCGTGGGATTCTC; R: AGGTGGCTGGGGCCAAGGCT 163 29 F: CCTCATGCCCGTCTTCTTC; R: AGATGGCGCTCAGGCCAC 267 30 F: AAGGTAGTCTGCCGCCTCCG; R: CCAAGGCCCGCCATGCCACT 251 31 F: CGTCGACACGGCCTTCCCTT; R: ACCGCTTCCCTCTGAGAGGC 118 32 F: AGCAGCCCCGTCTGTGTCCT; R: GTCCAGGCTCTGAGCCACAC 171 33 1F: CTCTTTGGGATGGTCCTTTC; 1R: CTGTGACCGGGCCTTAACCT 275 33 2F: CCTGGGGACAAGGCCGACGT; 2R: CCCGCCCGGAAGCAAGAGGC 317 34 F: CCACCATCCCCTCCCTGT; R: TAGCCAGAGCCCTCTGCAA 206 35 F: CTCTGGCTAAGCTCCCTGT; R: TGAGCACTTCATGCTGTAGG 326 36 F: GGGAGTGATGCCACCCTG; R: ATGGGAGCCGTGCCCTGA \\u2014b 37 F: CATCCGGCCCTGCTCACCCT; R: GCCTCACTGACGGCCCTCAC 187 38 F: TGACCCTTTCTCTTGTCCGG; R: TCTGCACTGCAGCCCCACCC 131 39,40 F: TGGCGTGACCACCAAGTCTC; R: GCTGAGGAGCCCCATATTCC 325 41 F: CAGACTTACTGCCCAAGCCGCCT; R: CCCCGCACCAAGCAGACAAA 342 a Data were kindly provided by Dr. J.R. Sampson (University of Wales, UK) b The exact sizes of the PCR products were not known 393 Sequencing Variant bands were reamplified and subcloned into pGEMT Vector (Promega, Madison, WI, USA). The DNA sequences were determined using the Thermo Sequenase Fluorescent Labeled Primer Cycle Sequencing Kit (Amersham, Buckinghamshire, England) with Cy-5-labeled M13 universal or reversal primers and ALFred Automatic Sequencer (Pharmacia Biotech, Tokyo, Japan). At least ten independent clones of each PCR product were sequenced in both directions. The sequence data were analyzed by Genetyx computer software (Software Development, Tokyo, Japan). Results SSCP analysis A total of 25 variants were detected by SSCP analysis (Fig. 1) from 23 patients (60% of the total 38 patients). Seven variants of TSC1 gene were detected in 3 familial cases and 4 sporadic ones. Fifteen variants of TSC2 gene were detected in 5 familial cases, 10 sporadic cases, and one unknown case. L574L of TSC1 and I531I of TSC2 genes were considered to be polymorphisms because the point mutations did not cause amino acid substitution. M322T of TSC1 was detected in the normal controls (6 of 80) and was supposed to represent a polymorphism (Table 2A). All other missense mutations were not detected in 60 or 100 normal controls. Any DNA sample from the parents was not available except for P28. Mutations in TSC1 gene Two nonsense, two frame shift, and three missense mutations were located in various exons of the gene. Two mutations (R509X, Q654E) in three patients were located in exon 15, and the R509X was found both in a sporadic patient (P26) and a familial patient (P3). Two mutations (W676X and T417I) were found in one patient (P31). Mutations in TSC2 gene One nonsense, 3 frame shift, and 8 missense mutations were identified in 16 patients (Table 2B). The two single nucleotide substitutions (2760 1 1 G to A, 5179 2 1 G to T) of an exon\\u2013intron junction were predicted to cause a splicing error. Although a substitution (5178 1 5 G to A) in the intron was not in the consensus sequence of the splicing, the mutation was located 5bp distant from the exon and was not detected in 100 normal controls. The mutation might cause a splicing error. L717R in a patient with pulmonary tuberous sclerosis was reported previously (Zhan et al. 1999). Both the V769E mutation (P2 and P21) and R1459X mutation (P6 and P8) were detected in 2 unrelated sporadic patients. Two missense mutations (H137R and D647N) Fig. 1. PCR-SSCP analysis of the TSC1/TSC2 genes. Arrows indicate the aberrant bands found in the patients\\u2019 samples. C, control samples from healthy subjects; P, patients\\u2019 samples 394 were found in the same patient (P32). P1657L in P28 was confirmed to be a de novo mutation by examination of the parents\\u2019 samples. Clinical symptoms and mutation types The clinical symptoms and the gene mutations are summarized in Table 3. The number of the patients with mental retardation (1 and 11) in TSC2 was significantly larger than that in TSC1 (\\u00f82 5 4.8, P 5 0.0285). The severity of mental retardation of the patients with TSC1 mutations appeared to be milder than that of the patients with TSC2 mutations. Complete information on renal and cardiac involvements was not available and could not be assessed. None of the mutation types (missense, nonsense, frame shift, and splicing) appeared to be correlated with a specific clinical symptom. In particular, the R509X mutation in TSC1 was detected in two patients, and the severity of mental retardation in these patients were quite different. Discussion Mutations in a given gene may be screened by various methods such as Southern blot analysis, SSCP, heteroduplex analysis, protein truncation test, and RT-PCR followed by SSCP (Kumar et al. 1995a,b; Vrtel et al. 1996; Wilson et al. 1996; Bajek et al. 1997; Jones et al. 1997; Maheshwar et al. 1997; Au et al. 1997, 1998). In the present study, we employed SSCP to survey the mutations of TSC1 and TSC2 genes because the method is simple enough to cover the multiple numbers of exons. To increase the mutation detection rate of SSCP, we used four electrophoresis conditions by a combination of different sets of temperature and glycerol content of the gel. Our mutation detection rate was 60% (23 of 38), which is comparable with those in other reports (Jones et al. 1997; Van Slegtenhorst et al. 1997; Au et al. 1998). There are several possible explanations for the failure to detect of TSC1/TSC2 mutations in the 15 patients. First, it can be false negative. One report showed that the SSCP Table 2A. Mutations in TSC1 Type of Nucleotide Patient Amino acid change Location mutation alteration S/F* 46 216PheFS-216X(a) exon 7 Frame shift 866delTT F 11 311SerFS-340X exon 10 Frame shift 1210insT F 31 T417l exon 12 Missense 1471 CtoT S 26,3 R509X(b) exon 15 Nonsense 1746 CtoT F,S 23 Q654E(b) exon 15 Missense 2181 CtoG S 31 W676X exon 16 Nonsense 2249 GtoA S 12 T899S exon 21 Missense 2918 CtoG S 46,27,3 L574L exon 15 Polymorphism 1947 TtoC 17,34,36 M322T exon 10 Polymorphism 1186 TtoC Table 2B. Mutations in TSC2 Type of Nucleotide Patient Mutation Location mutation alteration S/F* 32 H137R Exon 4 Missense 428 AtoG S 48 238LeuFS-337X Exon 7 Frame shift 725 insT S 30 F320L Exon 9 Missense 978 TtoG 43 337GluFS-351X Exon 10 Frame shift 1029GATCdel, CATGGCAT ins F 22 R611Q(c) Exon 16 Missense 1850 GtoA F 47 597HisFS-614X Exon 16 Frame shift 1808delACTAC S 32 D647N Exon 17 Missense 1957 AtoG S 1 L717R Exon 19 Missense 2168 TtoG S 21,2 V769E Exon 20 Missense 2324 TtoA F,S 33 \\u2014 Intron 24 Splicing 2760 1 1 GtoA S 27 V963M Exon 25 Missense 2905 GtoA S 6,8 R1459X Exon 33 Nonsense 4393 CtoT S,S 28 P1657L(d) Exon 38 Missense 5042 CtoT S 24 \\u2014 Intron 39 Splicing 5178 1 5 GtoA F 41 \\u2014 Intron 39 Splicing 5179 2 1 GtoT F 33 1531l Exon 14 Polymorphism 1611 CtoT *S, sporadic; F, familial The initiator ATG begins at base 222 of TSC1 and at base 19 for TSC2 The original amino acid and its position in the protein (numbered from the initiator Met as 1) are followed by a new amino acid for missense mutations, and by X for nonsense mutation a,b The identical change was reported by Van Slegtenhorst et al. (1997) c The identical change was reported by Wilson et al. (1996) d The identical change was reported by Maheshwar et al. (1997) 395 method to detect a point mutation on a 250-bp DNA fragment was 63% effective (Sheffield et al. 1993). Second, it may be true negative in those cases in which the mutation lies outside of the coding sequence, either in the 59-regulatory elements, within the intronic sequences, or in the 39- UTR. In addition, a large-scale rearrangement of genomic DNA may yield true negative results in the SSCP of the coding exons. Although there is no report of a large-scale mutation in TSC1, there are indeed some in TSC2 (Jones et al. 1997; The European Chromosome 16 Tuberous Sclerosis Consortium 1993; Au et al. 1997; Van Slegtenhorst et al. 1997). The mutation surveys of TSC1/TSC2 genes in Caucasian TSC patients revealed no clustering of the mutation in either gene (Wilson et al. 1996; Jones et al. 1997; Van Slegtenhorst et al. 1997; Au et al. 1998). In accordance with these previous findings, the 22 mutations (18 new and 4 reported) found in this study were distributed on various exons and there was no clustering of the mutations. Again in accordance with the previous findings, there was no obvious relationship between the type and location of the mutation and the clinical symptoms. However, the patients with TSC2 mutations appeared to have more severe mental retardation compared with the patients with TSC1 mutations. The same tendency has been noted (Jones et al. 1997). It is therefore at least clear that the clinical phenotype cannot be predicted from the genotype. Because of the heterogeneity of the mutations, it also appears impossible at present to develop a simple diagnostic test for TSC. Acknowledgements We thank Dr. Satoko Kumada, Dr. Akira Uchiyama (Tokyo Metropolitan Fuchu Ryoiku Center), Dr. Youich Sakakihara, Dr. Hidekazu Takeshita (Departement of Pediatrics, Graduate School of Medicine, University of Tokyo), Dr. Khotaro Table 3. Clinical data of mutation detected patients Patient Mental White Subependymal number Age Sex retardation AF macules nodules Seizures S/F Mutations Location of mutation 46 9 M 1 2 1 1 1 F Frame shift TSC1 exon 7 11 7 F 6 1 1 1 1 F Frame shift TSC1 exon 10 31 8 F 1 n.d. 1 1 1 S Missense/nonsense TSC1 exon 12; TSC1 exon 16 3 32 F 11 n.d. n.d. n.d. n.d. S Nonsense TSC1 exon 15 26 21 M 2 2 1 1 2 F Nonsense TSC1 exon 15 23 21 M 1 2 1 1 1 S Missense TSC1 exon 15 12 32 F 2 1 1 1 1 S Missense TSC1 exon 21 32 2 M n.d. n.d. 1 1 1 S Missenses TSC2 exon 4; TSC2 exon 17 48 24 M 11 1 1 1 1 S Frame shift TSC2 exon 7 30 nd nd n.d. n.d. n.d. n.d. n.d. n.d. Missense TSC2 exon 9 43 8 M 11 1 1 1 1 F Frame shift TSC2 exon 10 22 6 F 2 2 1 1 1 F Missense TSC2 exon 16 47 6 F 1 1 1 1 1 S Frame shift TSC2 exon 16 1 17 F 2 1 1 1 1 S Missense TSC2 exon 19 21 44 F 11 1 1 n.d. 1 F Missense TSC2 exon 20 2 32 M 11 1 1 1 1 S Missense TSC2 exon 20 33 48 M 11 n.d. n.d. n.d. 1 S Splicing TSC2 intron 24 27 13 M 2 2 1 1 1 F Missense TSC2 exon 25 6 8 M 11 n.d. n.d. n.d. n.d. F Nonsense TSC2 exon 33 8 9 F 11 1 1 1 1 S Nonsense TSC2 exon 33 28 3 F 1 n.d. 1 1 1 S Missense TSC2 exon 38 24 12 M 2 2 1 1 1 F Splicing TSC2 intron 39 41 7 M 2 1 1 1 1 F Splicing TSC2 intron 39 n.d., information not available; AF, angiofibroma; S, sporadic; F, familial Izumi, Dr. Masahiro Nakagawa (Third Department of Internal Medicine, Faculty of Medicine, Kagoshima University), Dr. Ryutaro Kira (Department of Pediatrics, Kyushu University), Dr. Hitoshi Sejima (Department of Pediatrics, Shimane Medical University), Dr. Masahito Miyazaki (Department of Pediatrics, School of Medicine, Tokushima University), Dr. Akira Kuramochi (Department of Dermatology, Saitama Medical School), and Dr. Sadahisa Ohta (Department of Neurosurgery, Tokyo Medical and Dental University) for collecting the samples of the TSC patients. This work was supported by the Ministry of Health and Welfare of Japan by a special grant for neurocutaneous disease, and by the Ministry of Education, Science, Sport and Culture of Japan by a grant-in-aid for scientific research.  Twenty-seven Japanese patients with the tuberous sclerosis complex (TSC), consisting of 23 sporadic and 4 familial cases, were tested for mutations in the TSC1 and TSC2 genes, using single-strand conformational polymorphism analysis and direct sequencing. Four possible pathogenic mutations were found in the TSC1 gene, including three frame shifts and a nonsense mutation in a familial case. All mutations were expected to result in a truncated hamartin gene product. The TSC2 gene analysis identified six possible pathogenic mutations only in the sporadic cases, including two frame shifts, one in-frame deletion, and three missense mutations. Two of the TSC2 mutations were expected to result in a truncated tuberin gene product. These results of the Japanese TSC patients were compatible with the reports from Europe and the United States, i.e., (1) TSC1 mutations are rarer in sporadic cases than in familial cases, (2) substantial numbers of sporadic cases arise from mutations in the TSC2 gene, and (3) mutations of the TSC1 gene may cause premature truncation of hamartin.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}",
              "type": "dataframe",
              "variable_name": "train"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-4d9add7c-03cd-468b-bebb-a09ca98fbecb\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>ID</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Variation</th>\n",
              "      <th>Class</th>\n",
              "      <th>Text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>0</td>\n",
              "      <td>FAM58A</td>\n",
              "      <td>Truncating Mutations</td>\n",
              "      <td>1</td>\n",
              "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>CBL</td>\n",
              "      <td>W802*</td>\n",
              "      <td>2</td>\n",
              "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2</td>\n",
              "      <td>CBL</td>\n",
              "      <td>Q249E</td>\n",
              "      <td>2</td>\n",
              "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>3</td>\n",
              "      <td>CBL</td>\n",
              "      <td>N454D</td>\n",
              "      <td>3</td>\n",
              "      <td>Recent evidence has demonstrated that acquired...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>4</td>\n",
              "      <td>CBL</td>\n",
              "      <td>L399V</td>\n",
              "      <td>4</td>\n",
              "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3316</th>\n",
              "      <td>3316</td>\n",
              "      <td>RUNX1</td>\n",
              "      <td>D171N</td>\n",
              "      <td>4</td>\n",
              "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3317</th>\n",
              "      <td>3317</td>\n",
              "      <td>RUNX1</td>\n",
              "      <td>A122*</td>\n",
              "      <td>1</td>\n",
              "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3318</th>\n",
              "      <td>3318</td>\n",
              "      <td>RUNX1</td>\n",
              "      <td>Fusions</td>\n",
              "      <td>1</td>\n",
              "      <td>The Runt-related transcription factor 1 gene (...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3319</th>\n",
              "      <td>3319</td>\n",
              "      <td>RUNX1</td>\n",
              "      <td>R80C</td>\n",
              "      <td>4</td>\n",
              "      <td>The RUNX1/AML1 gene is the most frequent targe...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3320</th>\n",
              "      <td>3320</td>\n",
              "      <td>RUNX1</td>\n",
              "      <td>K83E</td>\n",
              "      <td>4</td>\n",
              "      <td>The most frequent mutations associated with le...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3321 rows × 5 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4d9add7c-03cd-468b-bebb-a09ca98fbecb')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-4d9add7c-03cd-468b-bebb-a09ca98fbecb button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-4d9add7c-03cd-468b-bebb-a09ca98fbecb');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-e3a21aac-134d-4119-8b62-9ec39c1289bf\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-e3a21aac-134d-4119-8b62-9ec39c1289bf')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-e3a21aac-134d-4119-8b62-9ec39c1289bf button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_acb54d09-6a61-4b36-8e13-1b9a8d678b62\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('train')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_acb54d09-6a61-4b36-8e13-1b9a8d678b62 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('train');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "        ID    Gene             Variation  Class  \\\n",
              "0        0  FAM58A  Truncating Mutations      1   \n",
              "1        1     CBL                 W802*      2   \n",
              "2        2     CBL                 Q249E      2   \n",
              "3        3     CBL                 N454D      3   \n",
              "4        4     CBL                 L399V      4   \n",
              "...    ...     ...                   ...    ...   \n",
              "3316  3316   RUNX1                 D171N      4   \n",
              "3317  3317   RUNX1                 A122*      1   \n",
              "3318  3318   RUNX1               Fusions      1   \n",
              "3319  3319   RUNX1                  R80C      4   \n",
              "3320  3320   RUNX1                  K83E      4   \n",
              "\n",
              "                                                   Text  \n",
              "0     Cyclin-dependent kinases (CDKs) regulate a var...  \n",
              "1      Abstract Background  Non-small cell lung canc...  \n",
              "2      Abstract Background  Non-small cell lung canc...  \n",
              "3     Recent evidence has demonstrated that acquired...  \n",
              "4     Oncogenic mutations in the monomeric Casitas B...  \n",
              "...                                                 ...  \n",
              "3316  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
              "3317  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
              "3318  The Runt-related transcription factor 1 gene (...  \n",
              "3319  The RUNX1/AML1 gene is the most frequent targe...  \n",
              "3320  The most frequent mutations associated with le...  \n",
              "\n",
              "[3321 rows x 5 columns]"
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "train"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 269,
          "referenced_widgets": [
            "4887e407b61742a0bbbd9afb596a24ea",
            "374f7999b6044aab820c205222cd4180",
            "749b2a61b6ba424f9f4c28eaaedcde8a",
            "93e36dc05d794cf0b2fdc08d3b1a21c2",
            "855f2978e943473fa47c24042f3c8ab7",
            "593d005df53346dabc08434184e94dfa",
            "d93d4c49a28d40fcb4fc29095e0f08df",
            "6d41fa432c924ba59c878feb184d0599",
            "94fec15f4468460bb36164106f3974b6",
            "cca22a41f2f14b9faf8c14112f21d028",
            "336ee518677b4aa38215224668044067",
            "5a6281d834b6428b8a3495662b15cfae",
            "3bc0982573374f5fa808a9c538911014",
            "44328b117863496b92e77e17b08f9769",
            "969a0195f1ad4316b201c49844b2a9e8",
            "78e410d1459447e0a30a2260e6e42664",
            "688230e990f4463e9d2331d13f420698",
            "f6f1b644e1f64939ac73373045b05744",
            "8b2ec685bae84da58720089676789bc0",
            "93b5980a620949e98ae7c89adcadf08c",
            "c63bedd10d1e4441a980fec1c1b5283d",
            "a7b2f68e27594cf2972db7e0495d2e90",
            "7c1959c72c4947ef87e123d49c43c203",
            "b1fd1d076d544d9a86198dfbb4893c3b",
            "048d2ab5a72744489745fd98e004b97f",
            "a12ee47525d74b70ab71ade8d62366fd",
            "8695cfb1d7e14c4ca370805548ea23b7",
            "a30dc886fb8a439c8eb36c0e49ab48fe",
            "60c533fd078449b6a33064420cce627b",
            "afc691399d0448ed9dea102c5f9b8550",
            "3b66bd0ed0a042c991f08aba3ac1408f",
            "50bd19c428b8482fb68209397d6bd135",
            "73c936b3860c4e55a7eeac24788c0890",
            "f77a63fea48a44dc9ab4738e252d875b",
            "2ac35836e3214fb8a61c595413375e3b",
            "4dc671d782494e81b84ab6470bb49b2d",
            "d3711321d1f84e03a7ddf36c53cb26ad",
            "2cd71fa3a8db4176bf6738ea79aeb919",
            "af742a527e114c7383d36b6ff5b1d16f",
            "884931a58bd94505b1b0fdc060a86433",
            "c15c0cfbf3bb4817806362ae9caa162e",
            "c59c14632b6d433db95e5668bb332f63",
            "fae3b45a792b4356b1c7e3499eb79bcc",
            "0f06b0e83cb84808aa794fb461a638f7"
          ]
        },
        "id": "-Iefn5W-sVju",
        "outputId": "9480b451-bc96-4a2c-b2b2-6e39c16dbe45"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: \n",
            "The secret `HF_TOKEN` does not exist in your Colab secrets.\n",
            "To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.\n",
            "You will be able to reuse this secret in all of your notebooks.\n",
            "Please note that authentication is recommended but still optional to access public models or datasets.\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "4887e407b61742a0bbbd9afb596a24ea",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "vocab.json:   0%|          | 0.00/927k [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "5a6281d834b6428b8a3495662b15cfae",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "merges.txt:   0%|          | 0.00/696k [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "7c1959c72c4947ef87e123d49c43c203",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "config.json:   0%|          | 0.00/595 [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "f77a63fea48a44dc9ab4738e252d875b",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "pytorch_model.bin:   0%|          | 0.00/1.56G [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "Define device\n",
        "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
        "\n",
        "# Load BioGPT tokenizer and model for text generation\n",
        "biogpt_model_name = \"microsoft/biogpt\"\n",
        "biogpt_tokenizer = BioGptTokenizer.from_pretrained(biogpt_model_name)\n",
        "biogpt_model = BioGptForCausalLM.from_pretrained(biogpt_model_name).to(device)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "Wpae0JzftAQn"
      },
      "outputs": [],
      "source": [
        "\n",
        "# Update generate_text with additional parameters\n",
        "def generate_text(prompt, max_new_tokens=150, temperature=0.7, top_p=0.95):\n",
        "    inputs = biogpt_tokenizer(prompt, return_tensors=\"pt\", truncation=True, max_length=1024).input_ids.to(device)\n",
        "\n",
        "    try:\n",
        "        outputs = biogpt_model.generate(\n",
        "            inputs,\n",
        "            max_new_tokens=max_new_tokens,\n",
        "            temperature=temperature,\n",
        "            top_p=top_p,\n",
        "            num_return_sequences=1,\n",
        "            do_sample=True\n",
        "        )\n",
        "        return biogpt_tokenizer.decode(outputs[0], skip_special_tokens=True)\n",
        "    except Exception as e:\n",
        "        print(f\"Generation error: {str(e)}\")\n",
        "        return prompt\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "id": "CKjHYyP-xDTc"
      },
      "outputs": [],
      "source": [
        "def augment_dataset(df, start_id=3221):\n",
        "    augmented_data = []\n",
        "    class_counts = df['Class'].value_counts().sort_values(ascending=True)\n",
        "    print(\"Original class distribution:\")\n",
        "    print(class_counts)\n",
        "\n",
        "    target_multipliers = {\n",
        "        8: 20, 9: 10, 3: 5, 5: 3, 6: 2,\n",
        "        2: 1.5, 1: 1.2, 4: 1.1, 7: 1\n",
        "    }\n",
        "\n",
        "    current_generated_id = start_id\n",
        "\n",
        "    for class_id, multiplier in target_multipliers.items():\n",
        "        # Get original samples for this class\n",
        "        class_df = df[df['Class'] == class_id]\n",
        "        original_count = len(class_df)\n",
        "\n",
        "        # Always keep original samples\n",
        "        augmented_data.append(class_df)\n",
        "\n",
        "        # Calculate target including original samples\n",
        "        target_count = min(int(original_count * multiplier), 1000)\n",
        "        needed = max(0, target_count - original_count)\n",
        "\n",
        "        if needed == 0:\n",
        "            print(f\"Class {class_id} already sufficient: {original_count} samples\")\n",
        "            continue\n",
        "\n",
        "        print(f\"Augmenting class {class_id} from {original_count} to {target_count} (+{needed})\")\n",
        "        generated_count = 0\n",
        "        attempts = 0\n",
        "        max_attempts = needed * 2\n",
        "\n",
        "        while generated_count < needed and attempts < max_attempts:\n",
        "            sample = class_df.sample(1)\n",
        "            prompt = (\n",
        "                f\"Generate clinical text about {sample['Gene'].values[0]} \"\n",
        "                f\"{sample['Variation'].values[0]} for class {class_id}. \"\n",
        "                f\"Context: {sample['Text'].values[0][:500]}\"\n",
        "            )\n",
        "\n",
        "            generated_text = generate_text(prompt)\n",
        "\n",
        "            if generated_text and generated_text != prompt:\n",
        "                new_entry = sample.copy()\n",
        "                new_entry['Text'] = generated_text\n",
        "                new_entry['Generated_ID'] = current_generated_id\n",
        "                current_generated_id += 1\n",
        "                augmented_data.append(new_entry)\n",
        "                generated_count += 1\n",
        "\n",
        "            attempts += 1\n",
        "\n",
        "        print(f\"Added {generated_count} new samples for class {class_id} (Total: {original_count + generated_count})\")\n",
        "\n",
        "    final_df = pd.concat(augmented_data)\n",
        "    print(\"\\nFinal class distribution:\")\n",
        "    print(final_df['Class'].value_counts().sort_values(ascending=True))\n",
        "    return final_df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 778,
          "referenced_widgets": [
            "1fd58801f3ca4cf79b5b64bbc3d3317b",
            "d8597a95fcb541f6882faf2ddfa95ebd",
            "037e6fba13254a0a95924f0ad0d5c0ba",
            "237d1793183e4533acaeafd3fb834d11",
            "096fcc68242d4a79bd83fde2d7c3a992",
            "40653463cbf04cee9a41be0d117f8f2d",
            "2e2debb2d1364986b9d2f89b87b7b648",
            "a2fa994e07a048d882a7d5c95eb6c521",
            "8a7ddb954b6f49ab968f381cac135e8c",
            "8e31dafdc75f4605afe0a2c4b0da12dc",
            "fdd5396749e54b9c9175fbc7540c2dd2"
          ]
        },
        "id": "NVpuuhdznDpW",
        "outputId": "2c5fa7d8-a7d4-47b8-8f28-bf4d614e4bcc"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Original class distribution:\n",
            "Class\n",
            "8     19\n",
            "9     37\n",
            "3     89\n",
            "5    242\n",
            "6    275\n",
            "2    452\n",
            "1    568\n",
            "4    686\n",
            "7    953\n",
            "Name: count, dtype: int64\n",
            "Augmenting class 8 from 19 to 380 (+361)\n"
          ]
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "1fd58801f3ca4cf79b5b64bbc3d3317b",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "model.safetensors:   0%|          | 0.00/1.56G [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Added 361 new samples for class 8 (Total: 380)\n",
            "Augmenting class 9 from 37 to 370 (+333)\n",
            "Added 333 new samples for class 9 (Total: 370)\n",
            "Augmenting class 3 from 89 to 445 (+356)\n",
            "Added 356 new samples for class 3 (Total: 445)\n",
            "Augmenting class 5 from 242 to 726 (+484)\n",
            "Added 484 new samples for class 5 (Total: 726)\n",
            "Augmenting class 6 from 275 to 550 (+275)\n",
            "Added 275 new samples for class 6 (Total: 550)\n",
            "Augmenting class 2 from 452 to 678 (+226)\n",
            "Added 226 new samples for class 2 (Total: 678)\n",
            "Augmenting class 1 from 568 to 681 (+113)\n",
            "Added 113 new samples for class 1 (Total: 681)\n",
            "Augmenting class 4 from 686 to 754 (+68)\n",
            "Added 68 new samples for class 4 (Total: 754)\n",
            "Class 7 already sufficient: 953 samples\n",
            "\n",
            "Final class distribution:\n",
            "Class\n",
            "9    370\n",
            "8    380\n",
            "3    445\n",
            "6    550\n",
            "2    678\n",
            "1    681\n",
            "5    726\n",
            "4    754\n",
            "7    953\n",
            "Name: count, dtype: int64\n"
          ]
        }
      ],
      "source": [
        "# Augment the dataset while preserving gene-variation relationships\n",
        "augmented_train = augment_dataset(train)\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "id": "bBhLm8Z5GiRt"
      },
      "outputs": [],
      "source": [
        "\n",
        "# Split the augmented data\n",
        "\n",
        "# Split dataset into training and validation sets (80-20 split)\n",
        "augment_texts, valaug_texts, augment_labels, valaug_labels = train_test_split(\n",
        "    augmented_train['Text'],\n",
        "    augmented_train['Class'],\n",
        "    test_size=0.2,\n",
        "    stratify=augmented_train['Class'],\n",
        "    random_state=42\n",
        ")\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 266,
          "referenced_widgets": [
            "39bf7f982e934c0cbace16fe01390308",
            "2006193fc5de44f981eed17a5ab509e7",
            "fc01e398f18946f1ab8e3602abfffae1",
            "1b261d960e1544828604f17bfec16dfd",
            "43265c2f5a09465ca1265804a8730a83",
            "a0c6f3f0540848dc823e55a214b3f393",
            "a57f853a464140babb57f59e76f7abbc",
            "73c6ebaaa0a74f5083d0ec32565efcc7",
            "f48ef23ec1ac4de89e064941998a52de",
            "f73ad57413ab46198a7189a4997fc069",
            "8d107236529c457f998e1f097beb1381",
            "d5f8db856320461d9692488592484d9c",
            "bd7a1e928b814ab6ba49437e02a2a25b",
            "83bc57c957db40ffa1d5af96a1706615",
            "90660f5fd5ed4b68a3cb58b87763e08f",
            "e08d37c0ceb741a4b0ac10dc08c0b2b5",
            "d13288b54ab240e1ae73afce651cb468",
            "9fbc30b7ef6a4973b141725de13559b5",
            "deb188e60d8f4ce38d2e2fb6b9450bc9",
            "5a7b5c8f2a424a9088a264a90ab09df2",
            "8101283d1df6450cbccf2f94a5c49503",
            "7371365739fb4ea8804364b762d5d982",
            "ebbed84720d64bff882b5156c010c66f",
            "0773892c78764e7fb0af2342cb4214e3",
            "ab123feb80e54400af0b5f22ceb983bd",
            "b3feaa73f0dd4d3aa0bef52e6e8b32f7",
            "61297c1d55ba4a92a7a38636fc70fea3",
            "9dba2695452d4b928cea9eac24f2513a",
            "fb0c404be8e14827808cef036eb9f703",
            "ab8e13db83fd4b41acdfac6f8f164baa",
            "1d0146ca0668478fa8d70017e7666286",
            "ec3a0b641bd14f87a4b22f5775e19ca7",
            "0f230cea07d94593875df35e2cce8756",
            "3528677b8beb496da5e99d650cb9954f",
            "331719c0890842adb135a007ed64c5d2",
            "7f61ec32fff54e8a86b81033d447c224",
            "c88de913b57445f28bae4e02fa480d84",
            "08b5c84493fd4557a44c4be3a86e903b",
            "15010981b124459fb59b9b643361945b",
            "0f0f144d459d443c93bc16a04ebc3041",
            "ffd8ea3524d544afa0d1aa0c612fbe33",
            "e0f5598f572d4488b060dd334268b362",
            "098596aaddfa4691a4692dc42be29f2c",
            "471c2cc1344a4773a8eb917c7cb5f78f",
            "cb26c8edff53437ea89bf3d71e428b5a",
            "21c738579bc64079b228c6a4f663ad33",
            "c37a8173761f4e578940007e2e4f8b93",
            "ff9a7ed635274fd18693ce926273c328",
            "3bc2a1499dae448e92f1fbf7079cc24f",
            "8c2807586c344032be126ffdb7cccd33",
            "d618b9446bad42afa687c92d2c5b07b8",
            "95ccc8c232e84cfdbb17fcb0234953b1",
            "7d12d690f8fb40fe8d89ac74914ff8b8",
            "d0df9b897fc344fba3cfb3d595516b60",
            "8b848b313af448d497940b373983255f"
          ]
        },
        "id": "3pCoOdEAD0oE",
        "outputId": "4e8ec3f6-78f0-4913-acb0-4f345a7a2e48"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "39bf7f982e934c0cbace16fe01390308",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "tokenizer_config.json:   0%|          | 0.00/51.0 [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "d5f8db856320461d9692488592484d9c",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "vocab.txt:   0%|          | 0.00/213k [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "ebbed84720d64bff882b5156c010c66f",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "3528677b8beb496da5e99d650cb9954f",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "config.json:   0%|          | 0.00/433 [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "cb26c8edff53437ea89bf3d71e428b5a",
              "version_major": 2,
              "version_minor": 0
            },
            "text/plain": [
              "model.safetensors:   0%|          | 0.00/436M [00:00<?, ?B/s]"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Some weights of BertForSequenceClassification were not initialized from the model checkpoint at monologg/biobert_v1.1_pubmed and are newly initialized: ['classifier.bias', 'classifier.weight']\n",
            "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n",
            "The new embeddings will be initialized from a multivariate normal distribution that has old embeddings' mean and covariance. As described in this article: https://nlp.stanford.edu/~johnhew/vocab-expansion.html. To disable this, use `mean_resizing=False`\n"
          ]
        },
        {
          "data": {
            "text/plain": [
              "Embedding(29001, 768, padding_idx=0)"
            ]
          },
          "execution_count": 12,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# Load BioBERT tokenizer and model\n",
        "model_name = 'monologg/biobert_v1.1_pubmed'\n",
        "tokenizer = BertTokenizer.from_pretrained(model_name)\n",
        "\n",
        "# Add missing terms to tokenizer\n",
        "missing_terms = [\"dna\", \"fig\", \"missense\", \"mutants\", \"supplementary\"]\n",
        "tokenizer.add_tokens(missing_terms)\n",
        "\n",
        "pretrained_model = BertForSequenceClassification.from_pretrained(model_name, num_labels=9).to(device)\n",
        "pretrained_model.resize_token_embeddings(len(tokenizer))\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ZoQQekjacADH"
      },
      "source": [
        "# gradual unfreezing"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "jRLlwC1Nb13l"
      },
      "outputs": [],
      "source": [
        "\n",
        "# Unfreeze last few layers for fine-tuning\n",
        "def unfreeze_layers1(model, num_epochs, freeze_at_epoch=7):\n",
        "    for name, param in model.named_parameters():\n",
        "        # Initially freeze all layers\n",
        "        param.requires_grad = False\n",
        "\n",
        "    # Unfreeze layers gradually after certain epochs\n",
        "    if num_epochs > freeze_at_epoch:\n",
        "        for name, param in model.named_parameters():\n",
        "            if 'encoder.layer.10' in name or 'encoder.layer.11' in name:  # Example for unfreezing last layers\n",
        "                param.requires_grad = True\n",
        "\n",
        "    return model\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "id": "zxz8_xoR1hj1"
      },
      "outputs": [],
      "source": [
        "import torch.nn as nn # Import the nn module from PyTorch\n",
        "import torch.nn.functional as F\n",
        "# Weighted Focal Loss\n",
        "class WeightedFocalLoss(nn.Module):\n",
        "    def __init__(self, alpha=0.5, gamma=2, device='cuda'):\n",
        "        super().__init__()\n",
        "        self.alpha = alpha\n",
        "        self.gamma = gamma\n",
        "        self.device = device\n",
        "\n",
        "    def forward(self, inputs, targets):\n",
        "        targets = targets.to(self.device)\n",
        "        ce_loss = F.cross_entropy(inputs, targets.long(), reduction=\"none\")  # Ensure targets are long\n",
        "        pt = torch.exp(-ce_loss)\n",
        "        focal_loss = self.alpha * (1 - pt) ** self.gamma * ce_loss\n",
        "        weighted_loss = focal_loss * class_weights[targets]\n",
        "        return weighted_loss.mean()\n",
        "\n",
        "# Define loss function\n",
        "loss_fn = WeightedFocalLoss(alpha=0.5, gamma=3, device=device)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "id": "2LEV83HMD8cb"
      },
      "outputs": [],
      "source": [
        "# Custom Trainer class\n",
        "class CustomLossTrainer(Trainer):\n",
        "    def __init__(self, *args, loss_fn=None, **kwargs):\n",
        "        super().__init__(*args, **kwargs)\n",
        "        if loss_fn is None:\n",
        "            raise ValueError(\"A custom loss function must be provided!\")\n",
        "        self.loss_fn = loss_fn\n",
        "        self.label_names = [\"labels\"]\n",
        "    def compute_loss(self, model, inputs, return_outputs=False, **kwargs):\n",
        "        labels = inputs.get(\"labels\")\n",
        "        outputs = model(**inputs)\n",
        "        logits = outputs.get(\"logits\")\n",
        "        loss = self.loss_fn(logits, labels)\n",
        "        return (loss, outputs) if return_outputs else loss\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {
        "id": "EqKlWiFi2ULU"
      },
      "outputs": [],
      "source": [
        "# Dataset class\n",
        "class MutationDataset(Dataset):\n",
        "    def __init__(self, texts, labels):\n",
        "        # Convert texts to a list of strings if it's not already\n",
        "        if not isinstance(texts, list):\n",
        "            texts = texts.tolist()  # Convert Series or other objects to a list\n",
        "\n",
        "        self.encodings = tokenizer(\n",
        "            texts, truncation=True, padding=\"max_length\", max_length=512, return_tensors='pt', return_token_type_ids=False,\n",
        "        )\n",
        "        # Convert 1-9 to 0-8 for each label in the list using list comprehension\n",
        "        self.labels = [label - 1 for label in labels]\n",
        "\n",
        "\n",
        "    def __getitem__(self, idx):\n",
        "        return {\n",
        "            'input_ids': self.encodings['input_ids'][idx].clone().detach().to(device),\n",
        "            'attention_mask': self.encodings['attention_mask'][idx].clone().detach().to(device),\n",
        "            'labels': torch.tensor(self.labels[idx], dtype=torch.long).to(device)  # Ensure dtype is long\n",
        "        }\n",
        "\n",
        "    def __len__(self):\n",
        "        return len(self.labels)\n",
        "\n",
        "# Create train and validation datasets\n",
        "train_dataset = MutationDataset(augment_texts, augment_labels)\n",
        "val_dataset = MutationDataset(valaug_texts, valaug_labels)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 651
        },
        "id": "Mjjks26C1s1M",
        "outputId": "c534946c-b5c4-4e7d-9b5f-af97a9e7c6ec"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.11/dist-packages/transformers/training_args.py:1575: FutureWarning: `evaluation_strategy` is deprecated and will be removed in version 4.46 of 🤗 Transformers. Use `eval_strategy` instead\n",
            "  warnings.warn(\n",
            "Some weights of BertForSequenceClassification were not initialized from the model checkpoint at monologg/biobert_v1.1_pubmed and are newly initialized: ['classifier.bias', 'classifier.weight']\n",
            "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n",
            "<ipython-input-15-78ce3af2793f>:4: FutureWarning: `tokenizer` is deprecated and will be removed in version 5.0.0 for `CustomLossTrainer.__init__`. Use `processing_class` instead.\n",
            "  super().__init__(*args, **kwargs)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "\n",
              "    <div>\n",
              "      \n",
              "      <progress value='6094' max='6094' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
              "      [6094/6094 12:00, Epoch 11/11]\n",
              "    </div>\n",
              "    <table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              " <tr style=\"text-align: left;\">\n",
              "      <th>Epoch</th>\n",
              "      <th>Training Loss</th>\n",
              "      <th>Validation Loss</th>\n",
              "      <th>Accuracy</th>\n",
              "      <th>F1</th>\n",
              "      <th>Precision</th>\n",
              "      <th>Recall</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <td>1</td>\n",
              "      <td>0.687200</td>\n",
              "      <td>0.377788</td>\n",
              "      <td>0.571300</td>\n",
              "      <td>0.566471</td>\n",
              "      <td>0.692259</td>\n",
              "      <td>0.571300</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>2</td>\n",
              "      <td>0.296100</td>\n",
              "      <td>0.201473</td>\n",
              "      <td>0.703971</td>\n",
              "      <td>0.709154</td>\n",
              "      <td>0.761104</td>\n",
              "      <td>0.703971</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>3</td>\n",
              "      <td>0.208300</td>\n",
              "      <td>0.181224</td>\n",
              "      <td>0.725632</td>\n",
              "      <td>0.729788</td>\n",
              "      <td>0.788221</td>\n",
              "      <td>0.725632</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>4</td>\n",
              "      <td>0.170800</td>\n",
              "      <td>0.162237</td>\n",
              "      <td>0.738267</td>\n",
              "      <td>0.738758</td>\n",
              "      <td>0.747610</td>\n",
              "      <td>0.738267</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>5</td>\n",
              "      <td>0.166100</td>\n",
              "      <td>0.153181</td>\n",
              "      <td>0.727437</td>\n",
              "      <td>0.730488</td>\n",
              "      <td>0.745248</td>\n",
              "      <td>0.727437</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>6</td>\n",
              "      <td>0.140400</td>\n",
              "      <td>0.149888</td>\n",
              "      <td>0.746390</td>\n",
              "      <td>0.750417</td>\n",
              "      <td>0.771452</td>\n",
              "      <td>0.746390</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>7</td>\n",
              "      <td>0.137500</td>\n",
              "      <td>0.149639</td>\n",
              "      <td>0.741877</td>\n",
              "      <td>0.742293</td>\n",
              "      <td>0.745005</td>\n",
              "      <td>0.741877</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>8</td>\n",
              "      <td>0.122600</td>\n",
              "      <td>0.143346</td>\n",
              "      <td>0.759025</td>\n",
              "      <td>0.760927</td>\n",
              "      <td>0.772775</td>\n",
              "      <td>0.759025</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>9</td>\n",
              "      <td>0.118200</td>\n",
              "      <td>0.141719</td>\n",
              "      <td>0.752708</td>\n",
              "      <td>0.753052</td>\n",
              "      <td>0.753827</td>\n",
              "      <td>0.752708</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>10</td>\n",
              "      <td>0.101900</td>\n",
              "      <td>0.143948</td>\n",
              "      <td>0.742780</td>\n",
              "      <td>0.742726</td>\n",
              "      <td>0.745778</td>\n",
              "      <td>0.742780</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <td>11</td>\n",
              "      <td>0.104000</td>\n",
              "      <td>0.142705</td>\n",
              "      <td>0.752708</td>\n",
              "      <td>0.753816</td>\n",
              "      <td>0.757865</td>\n",
              "      <td>0.752708</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table><p>"
            ],
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "text/html": [
              "\n",
              "    <div>\n",
              "      \n",
              "      <progress value='139' max='139' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
              "      [139/139 00:08]\n",
              "    </div>\n",
              "    "
            ],
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Validation Results: {'eval_loss': 0.14270511269569397, 'eval_accuracy': 0.7527075812274369, 'eval_f1': 0.7538164420705623, 'eval_precision': 0.7578652657678782, 'eval_recall': 0.7527075812274369, 'eval_runtime': 8.4098, 'eval_samples_per_second': 131.75, 'eval_steps_per_second': 16.528, 'epoch': 11.0}\n"
          ]
        },
        {
          "data": {
            "text/plain": [
              "('./results/tokenizer_config.json',\n",
              " './results/special_tokens_map.json',\n",
              " './results/vocab.txt',\n",
              " './results/added_tokens.json')"
            ]
          },
          "execution_count": 17,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "\n",
        "# Compute class weights\n",
        "labels_array = np.array(augment_labels) - 1  # Convert train_labels to a NumPy array before subtraction\n",
        "classes = np.unique(labels_array)\n",
        "cw = compute_class_weight('balanced', classes=classes, y=labels_array)\n",
        "class_weights = torch.tensor(cw, dtype=torch.float).to(device)\n",
        "# Training arguments\n",
        "training_args = TrainingArguments(\n",
        "    output_dir='./results',\n",
        "    num_train_epochs=11,\n",
        "    per_device_train_batch_size=8,\n",
        "    per_device_eval_batch_size=8,\n",
        "    learning_rate=2e-5,\n",
        "    warmup_ratio=0.1,\n",
        "    weight_decay=0.01,\n",
        "    evaluation_strategy='epoch',\n",
        "    fp16=True,\n",
        "    logging_dir='./logs',\n",
        "    report_to='none',\n",
        "    dataloader_pin_memory=False\n",
        ")\n",
        "\n",
        "# Reload model for each fold and unfreeze layers gradually\n",
        "pretrained_model = BertForSequenceClassification.from_pretrained(model_name, num_labels=9).to(device)\n",
        "pretrained_model.resize_token_embeddings(len(tokenizer))\n",
        "\n",
        "#Unfreeze layers after certain epochs\n",
        "pretrained_model = unfreeze_layers1(pretrained_model, num_epochs=30)\n",
        "\n",
        "\n",
        "# Trainer\n",
        "trainer = CustomLossTrainer(\n",
        "    model=pretrained_model,\n",
        "    args=training_args,\n",
        "    train_dataset=train_dataset,\n",
        "    eval_dataset=val_dataset,\n",
        "    tokenizer=tokenizer,\n",
        "    compute_metrics=lambda p: {\n",
        "        'accuracy': accuracy_score(p.label_ids, p.predictions.argmax(-1)),\n",
        "        'f1': f1_score(p.label_ids, p.predictions.argmax(-1), average='weighted'),\n",
        "        'precision': precision_score(p.label_ids, p.predictions.argmax(-1), average='weighted'),\n",
        "        'recall': recall_score(p.label_ids, p.predictions.argmax(-1), average='weighted')\n",
        "    },\n",
        "    loss_fn=loss_fn\n",
        ")\n",
        "\n",
        "# Train & Evaluate\n",
        "trainer.train()\n",
        "results = trainer.evaluate()\n",
        "print(\"Validation Results:\", results)\n",
        "\n",
        "\n",
        "# Save the model weights and configuration\n",
        "trainer.save_model('./results')\n",
        "\n",
        "# In your training script after tokenizer modifications\n",
        "tokenizer.save_pretrained('./results')  # Save modified tokenizer with custom tokens\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {
        "id": "NHwT5PgUzqvb"
      },
      "outputs": [],
      "source": [
        "import torch\n",
        "from transformers import BertTokenizer, BertForSequenceClassification\n",
        "import numpy as np\n",
        "from tqdm import tqdm\n",
        "import gc\n",
        "\n",
        "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
        "\n",
        "# Load FINE-TUNED model and tokenizer with clear naming\n",
        "model_path = './results'\n",
        "bert_tokenizer = BertTokenizer.from_pretrained(model_path)\n",
        "bert_model = BertForSequenceClassification.from_pretrained(model_path).to(device)\n",
        "\n",
        "def clear_memory():\n",
        "    torch.cuda.empty_cache()\n",
        "    gc.collect()\n",
        "\n",
        "def get_embeddings_batched(texts, batch_size=16):\n",
        "    \"\"\"Get [CLS] embeddings using fine-tuned BERT\"\"\"\n",
        "    bert_model.eval()  # Set to evaluation mode\n",
        "    embeddings = []\n",
        "\n",
        "    with torch.no_grad():\n",
        "        for i in tqdm(range(0, len(texts), batch_size)):\n",
        "            batch = texts[i:i+batch_size]\n",
        "\n",
        "            # Tokenize with BERT tokenizer\n",
        "            inputs = bert_tokenizer(\n",
        "                batch,\n",
        "                padding=True,\n",
        "                truncation=True,\n",
        "                max_length=512,\n",
        "                return_tensors=\"pt\"\n",
        "            ).to(device)\n",
        "\n",
        "            # Get hidden states from BERT base\n",
        "            outputs = bert_model.bert(**inputs)\n",
        "\n",
        "            # Extract CLS embeddings\n",
        "            cls_emb = outputs.last_hidden_state[:, 0, :].cpu().numpy()\n",
        "            embeddings.append(cls_emb)\n",
        "\n",
        "            # Clean memory\n",
        "            del inputs, outputs\n",
        "            clear_memory()\n",
        "\n",
        "    return np.vstack(embeddings)\n",
        "\n",
        "# Usage\n",
        "clear_memory()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "E_I0rnJ_rsLy",
        "outputId": "a9409bcd-c8cc-43df-d027-833f7332115c"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "['variation_label_encoder.pkl']"
            ]
          },
          "execution_count": 20,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "from sklearn.preprocessing import LabelEncoder\n",
        "import joblib\n",
        "# Label encode categorical features\n",
        "le_gene = LabelEncoder()\n",
        "le_variation = LabelEncoder()\n",
        "\n",
        "\n",
        "augmented_train['Gene_encoded'] = le_gene.fit_transform(augmented_train['Gene'])\n",
        "joblib.dump(le_gene, 'gene_label_encoder.pkl')  # Save for later\n",
        "\n",
        "augmented_train['Variation_encoded'] = le_variation.fit_transform(augmented_train['Variation'])\n",
        "joblib.dump(le_variation, 'variation_label_encoder.pkl')  # Save for later"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pZgOTbFFkVhY",
        "outputId": "c1553ee4-1b24-4eec-fc60-549a35f406ba"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "100%|██████████| 347/347 [13:52<00:00,  2.40s/it]\n"
          ]
        }
      ],
      "source": [
        "import numpy as np\n",
        "import pandas as pd\n",
        "from sklearn.metrics import log_loss, f1_score, accuracy_score, precision_score, recall_score\n",
        "from sklearn.model_selection import StratifiedKFold\n",
        "from collections import Counter\n",
        "import xgboost as xgb\n",
        "from imblearn.over_sampling import SMOTE\n",
        "\n",
        "# Ensure classes are 0-based\n",
        "augmented_train['Class'] = augmented_train['Class'] - 1  # If classes start at 1\n",
        "\n",
        "# Generate embeddings for the entire augmented dataset\n",
        "all_embeddings = get_embeddings_batched(augmented_train['Text'].tolist(), batch_size=16)\n",
        "\n",
        "# Now create X_all using the entire augmented_train DataFrame\n",
        "X_all = np.hstack([all_embeddings, augmented_train[['Gene_encoded', 'Variation_encoded']].values])\n",
        "y_all = augmented_train['Class'].values # Use values to get a NumPy array\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xmgZNsxyEAfR",
        "outputId": "4fab04a0-26b1-488e-894e-0e6d72a1e732"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Original classes: [0 1 2 3 4 5 6 7 8]\n",
            "Remapped classes: [0 1 2 3 4 5 6 7 8]\n",
            "\n",
            "Fold 1\n",
            "[0]\ttrain-mlogloss:1.89245\tval-mlogloss:1.89912\n",
            "[50]\ttrain-mlogloss:0.30775\tval-mlogloss:0.50783\n",
            "[75]\ttrain-mlogloss:0.25517\tval-mlogloss:0.51194\n",
            "\n",
            "Fold 1 Evaluation:\n",
            "log_loss: 0.5119\n",
            "accuracy: 0.8204\n",
            "f1_macro: 0.8389\n",
            "f1_weighted: 0.8212\n",
            "\n",
            "Fold 2\n",
            "[0]\ttrain-mlogloss:1.89079\tval-mlogloss:1.90653\n",
            "[50]\ttrain-mlogloss:0.29753\tval-mlogloss:0.57069\n",
            "[71]\ttrain-mlogloss:0.25155\tval-mlogloss:0.57788\n",
            "\n",
            "Fold 2 Evaluation:\n",
            "log_loss: 0.5786\n",
            "accuracy: 0.7879\n",
            "f1_macro: 0.8129\n",
            "f1_weighted: 0.7886\n",
            "\n",
            "Fold 3\n",
            "[0]\ttrain-mlogloss:1.88996\tval-mlogloss:1.90108\n",
            "[50]\ttrain-mlogloss:0.30626\tval-mlogloss:0.52410\n",
            "[76]\ttrain-mlogloss:0.25433\tval-mlogloss:0.53109\n",
            "\n",
            "Fold 3 Evaluation:\n",
            "log_loss: 0.5311\n",
            "accuracy: 0.8220\n",
            "f1_macro: 0.8454\n",
            "f1_weighted: 0.8239\n",
            "\n",
            "Fold 4\n",
            "[0]\ttrain-mlogloss:1.88999\tval-mlogloss:1.91007\n",
            "[50]\ttrain-mlogloss:0.30345\tval-mlogloss:0.53461\n",
            "[81]\ttrain-mlogloss:0.24139\tval-mlogloss:0.53361\n",
            "\n",
            "Fold 4 Evaluation:\n",
            "log_loss: 0.5338\n",
            "accuracy: 0.8031\n",
            "f1_macro: 0.8279\n",
            "f1_weighted: 0.8054\n",
            "\n",
            "Fold 5\n",
            "[0]\ttrain-mlogloss:1.89023\tval-mlogloss:1.91330\n",
            "[50]\ttrain-mlogloss:0.29552\tval-mlogloss:0.57058\n",
            "[73]\ttrain-mlogloss:0.24484\tval-mlogloss:0.57706\n",
            "\n",
            "Fold 5 Evaluation:\n",
            "log_loss: 0.5771\n",
            "accuracy: 0.7977\n",
            "f1_macro: 0.8255\n",
            "f1_weighted: 0.7976\n",
            "\n",
            "========================================\n",
            "Final Cross-Validation Results:\n",
            "========================================\n",
            "log_loss       : 0.5465 ± 0.0267\n",
            "accuracy       : 0.8062 ± 0.0132\n",
            "f1_macro       : 0.8301 ± 0.0112\n",
            "f1_weighted    : 0.8073 ± 0.0135\n"
          ]
        }
      ],
      "source": [
        "import numpy as np\n",
        "import xgboost as xgb\n",
        "from sklearn.preprocessing import StandardScaler\n",
        "from sklearn.model_selection import KFold\n",
        "from sklearn.metrics import log_loss, accuracy_score, f1_score\n",
        "\n",
        "# Assuming X_all and y_all are your features and labels\n",
        "X = X_all  # Shape (5000, n_features)\n",
        "y = y_all  # Shape (5000,)\n",
        "\n",
        "# ========== CRUCIAL FIX ==========\n",
        "# Validate and remap labels to 0-based indexing\n",
        "unique_classes = np.unique(y)\n",
        "print(f\"Original classes: {unique_classes}\")\n",
        "\n",
        "# Create label mapping (e.g., [1,2,3] -> [0,1,2])\n",
        "label_mapping = {orig: new for new, orig in enumerate(unique_classes)}\n",
        "y_remapped = np.vectorize(label_mapping.get)(y)\n",
        "\n",
        "# Verify new labels\n",
        "print(f\"Remapped classes: {np.unique(y_remapped)}\")\n",
        "num_classes = len(unique_classes)\n",
        "# =================================\n",
        "\n",
        "# Initialize components\n",
        "scaler = StandardScaler()\n",
        "kf = KFold(n_splits=5, shuffle=True, random_state=42)\n",
        "fold_results = []\n",
        "\n",
        "for fold, (train_idx, val_idx) in enumerate(kf.split(X)):\n",
        "    print(f\"\\nFold {fold + 1}\")\n",
        "\n",
        "    # Split data\n",
        "    X_train, X_val = X[train_idx], X[val_idx]\n",
        "    y_train, y_val = y_remapped[train_idx], y_remapped[val_idx]  # Use remapped labels\n",
        "\n",
        "    # Scale features\n",
        "    X_train_scaled = scaler.fit_transform(X_train)\n",
        "    X_val_scaled = scaler.transform(X_val)\n",
        "\n",
        "    # Create DMatrix objects\n",
        "    dtrain = xgb.DMatrix(X_train_scaled, label=y_train)\n",
        "    dval = xgb.DMatrix(X_val_scaled, label=y_val)\n",
        "\n",
        "    # XGBoost parameters (MUST match actual class count)\n",
        "    params = {\n",
        "        'objective': 'multi:softprob',\n",
        "        'num_class': num_classes,  # Use actual count from data\n",
        "        'tree_method': 'hist',\n",
        "        'max_depth': 4,\n",
        "        'learning_rate': 0.1,\n",
        "        'subsample': 0.8,\n",
        "        'colsample_bytree': 0.8,\n",
        "        'random_state': 42\n",
        "    }\n",
        "\n",
        "    # Train model\n",
        "    model = xgb.train(\n",
        "        params,\n",
        "        dtrain,\n",
        "        num_boost_round=200,\n",
        "        evals=[(dtrain, 'train'), (dval, 'val')],\n",
        "        early_stopping_rounds=20,\n",
        "        verbose_eval=50\n",
        "    )\n",
        "\n",
        "    # Predictions\n",
        "    y_pred_proba = model.predict(dval)\n",
        "    y_pred = np.argmax(y_pred_proba, axis=1)\n",
        "\n",
        "    # Calculate metrics\n",
        "    fold_metrics = {\n",
        "        'log_loss': log_loss(y_val, y_pred_proba),\n",
        "        'accuracy': accuracy_score(y_val, y_pred),\n",
        "        'f1_macro': f1_score(y_val, y_pred, average='macro'),\n",
        "        'f1_weighted': f1_score(y_val, y_pred, average='weighted')\n",
        "    }\n",
        "\n",
        "    print(f\"\\nFold {fold + 1} Evaluation:\")\n",
        "    for metric, value in fold_metrics.items():\n",
        "        print(f\"{metric}: {value:.4f}\")\n",
        "\n",
        "    fold_results.append(fold_metrics)\n",
        "\n",
        "# Cross-validation summary\n",
        "print(\"\\n\" + \"=\" * 40)\n",
        "print(\"Final Cross-Validation Results:\")\n",
        "print(\"=\" * 40)\n",
        "for metric in ['log_loss', 'accuracy', 'f1_macro', 'f1_weighted']:\n",
        "    scores = [res[metric] for res in fold_results]\n",
        "    print(f\"{metric:15}: {np.mean(scores):.4f} ± {np.std(scores):.4f}\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QD10YOvnOHe2",
        "outputId": "32d9af65-0b62-4fef-81e5-b97cca3e0752"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Original classes: [0 1 2 3 4 5 6 7 8]\n",
            "Remapped classes: [0 1 2 3 4 5 6 7 8]\n",
            "Training completed and components saved in JSON/Joblib format!\n"
          ]
        }
      ],
      "source": [
        "import numpy as np\n",
        "import xgboost as xgb\n",
        "from sklearn.preprocessing import StandardScaler\n",
        "import pickle\n",
        "\n",
        "# Assuming X_all and y_all are your features and labels\n",
        "X = X_all  # Shape (5000, n_features)\n",
        "y = y_all  # Shape (5000,)\n",
        "\n",
        "# ========== Label Remapping ==========\n",
        "unique_classes = np.unique(y)\n",
        "print(f\"Original classes: {unique_classes}\")\n",
        "\n",
        "# Create and apply label mapping\n",
        "label_mapping = {orig: new for new, orig in enumerate(unique_classes)}\n",
        "y_remapped = np.vectorize(label_mapping.get)(y)\n",
        "print(f\"Remapped classes: {np.unique(y_remapped)}\")\n",
        "num_classes = len(unique_classes)\n",
        "\n",
        "# ========== Feature Scaling ==========\n",
        "scaler = StandardScaler()\n",
        "X_scaled = scaler.fit_transform(X)\n",
        "\n",
        "# ========== Full Dataset Training ==========\n",
        "# Create DMatrix for full dataset\n",
        "dtrain = xgb.DMatrix(X_scaled, label=y_remapped)\n",
        "\n",
        "# Parameters (keep same as before)\n",
        "params = {\n",
        "    'objective': 'multi:softprob',\n",
        "    'num_class': num_classes,\n",
        "    'tree_method': 'hist',\n",
        "    'max_depth': 4,\n",
        "    'learning_rate': 0.1,\n",
        "    'subsample': 0.8,\n",
        "    'colsample_bytree': 0.8,\n",
        "    'random_state': 42\n",
        "}\n",
        "\n",
        "# Train final model on all data\n",
        "final_model = xgb.train(\n",
        "    params,\n",
        "    dtrain,\n",
        "    num_boost_round=200  # Use optimal rounds from CV if available\n",
        ")\n",
        "\n",
        "# ========== Save Components (JSON version) ==========\n",
        "# Save the trained model in JSON format\n",
        "final_model.save_model('xgb_model.json')  # This saves as JSON by default when using .json extension\n",
        "\n",
        "# Save the scaler and mapping using joblib (better than pickle for sklearn objects)\n",
        "from joblib import dump\n",
        "\n",
        "dump(scaler, 'standard_scaler.joblib')\n",
        "dump(label_mapping, 'label_mapping.joblib')\n",
        "\n",
        "print(\"Training completed and components saved in JSON/Joblib format!\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wcVeDra3vwu5",
        "outputId": "b9d27db0-dcf5-4e80-d6e5-be8d59217dda"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "<ipython-input-26-12527716c292>:13: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
            "  test_text = pd.read_csv(\"test_text.csv\", sep='\\|\\|', names=['ID', 'Text'], skiprows=1)\n",
            "100%|██████████| 355/355 [19:45<00:00,  3.34s/it]\n"
          ]
        }
      ],
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "import torch\n",
        "from tqdm import tqdm\n",
        "import xgboost as xgb\n",
        "import joblib\n",
        "from transformers import BertTokenizer  # Added missing import\n",
        "\n",
        "\n",
        "# === Main Prediction Pipeline ===\n",
        "# Load data\n",
        "test_variants = pd.read_csv(\"test_variants.csv\")\n",
        "test_text = pd.read_csv(\"test_text.csv\", sep='\\|\\|', names=['ID', 'Text'], skiprows=1)\n",
        "test = pd.merge(test_variants, test_text, on='ID')\n",
        "\n",
        "# === FIX: Remove rows with invalid 'Text' values ===\n",
        "# Keep only rows where 'Text' is a non-empty string\n",
        "test = test[test['Text'].apply(lambda x: isinstance(x, str) and x.strip() != \"\")]\n",
        "\n",
        "# Generate BERT embeddings\n",
        "# Generate BERT embeddings\n",
        "text_embeddings = get_embeddings_batched(test['Text'].tolist())\n",
        "\n",
        "\n",
        "\n",
        "# Load encoders and encode features\n",
        "le_gene = joblib.load('gene_label_encoder.pkl')\n",
        "le_variation = joblib.load('variation_label_encoder.pkl')\n",
        "\n",
        "test['Gene_encoded'] = test['Gene'].apply(lambda x: le_gene.transform([x])[0] if x in le_gene.classes_ else -1)\n",
        "test['Variation_encoded'] = test['Variation'].apply(lambda x: le_variation.transform([x])[0] if x in le_variation.classes_ else -1)\n",
        "\n",
        "# Combine features\n",
        "X_test = np.hstack([\n",
        "    text_embeddings,\n",
        "    test[['Gene_encoded', 'Variation_encoded']].values\n",
        "])\n",
        "\n",
        "# Apply scaling\n",
        "scaler = joblib.load('standard_scaler.joblib')\n",
        "X_test_scaled = scaler.transform(X_test)\n",
        "\n",
        "# Make predictions\n",
        "xgb_model = xgb.Booster()\n",
        "xgb_model.load_model('xgb_model.json')\n",
        "dtest = xgb.DMatrix(X_test_scaled)\n",
        "test_probs = xgb_model.predict(dtest)\n",
        "\n",
        "# Re-map probabilities to original class order\n",
        "label_mapping = joblib.load('label_mapping.joblib')\n",
        "original_classes = sorted(label_mapping.keys())\n",
        "submission_probs = test_probs[:, np.argsort(original_classes)]\n",
        "\n",
        "# Create submission\n",
        "submission = pd.DataFrame({\n",
        "    'ID': test['ID'],\n",
        "    **{f'Class{cls}': submission_probs[:, i]\n",
        "       for i, cls in enumerate(sorted(original_classes))}\n",
        "})\n",
        "submission.to_csv('submission.csv', index=False)\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CySNeCzQqxH3",
        "outputId": "fa2332ae-2609-48a2-fa4f-4e90d0dea529"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Sun Feb 23 21:19:02 2025       \n",
            "+-----------------------------------------------------------------------------------------+\n",
            "| NVIDIA-SMI 550.54.15              Driver Version: 550.54.15      CUDA Version: 12.4     |\n",
            "|-----------------------------------------+------------------------+----------------------+\n",
            "| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |\n",
            "| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |\n",
            "|                                         |                        |               MIG M. |\n",
            "|=========================================+========================+======================|\n",
            "|   0  Tesla T4                       Off |   00000000:00:04.0 Off |                    0 |\n",
            "| N/A   55C    P8             10W /   70W |       0MiB /  15360MiB |      0%      Default |\n",
            "|                                         |                        |                  N/A |\n",
            "+-----------------------------------------+------------------------+----------------------+\n",
            "                                                                                         \n",
            "+-----------------------------------------------------------------------------------------+\n",
            "| Processes:                                                                              |\n",
            "|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |\n",
            "|        ID   ID                                                               Usage      |\n",
            "|=========================================================================================|\n",
            "|  No running processes found                                                             |\n",
            "+-----------------------------------------------------------------------------------------+\n"
          ]
        }
      ],
      "source": [
        "!nvidia-smi"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "t9M3C0cWqxga"
      },
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "gpuType": "T4",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "037e6fba13254a0a95924f0ad0d5c0ba": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_a2fa994e07a048d882a7d5c95eb6c521",
            "max": 1560700544,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_8a7ddb954b6f49ab968f381cac135e8c",
            "value": 1560700544
          }
        },
        "048d2ab5a72744489745fd98e004b97f": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_afc691399d0448ed9dea102c5f9b8550",
            "max": 595,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_3b66bd0ed0a042c991f08aba3ac1408f",
            "value": 595
          }
        },
        "0773892c78764e7fb0af2342cb4214e3": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_9dba2695452d4b928cea9eac24f2513a",
            "placeholder": "​",
            "style": "IPY_MODEL_fb0c404be8e14827808cef036eb9f703",
            "value": "special_tokens_map.json: 100%"
          }
        },
        "08b5c84493fd4557a44c4be3a86e903b": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "096fcc68242d4a79bd83fde2d7c3a992": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "098596aaddfa4691a4692dc42be29f2c": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "0f06b0e83cb84808aa794fb461a638f7": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "0f0f144d459d443c93bc16a04ebc3041": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "0f230cea07d94593875df35e2cce8756": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "15010981b124459fb59b9b643361945b": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1b261d960e1544828604f17bfec16dfd": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_f73ad57413ab46198a7189a4997fc069",
            "placeholder": "​",
            "style": "IPY_MODEL_8d107236529c457f998e1f097beb1381",
            "value": " 51.0/51.0 [00:00&lt;00:00, 5.24kB/s]"
          }
        },
        "1d0146ca0668478fa8d70017e7666286": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "1fd58801f3ca4cf79b5b64bbc3d3317b": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_d8597a95fcb541f6882faf2ddfa95ebd",
              "IPY_MODEL_037e6fba13254a0a95924f0ad0d5c0ba",
              "IPY_MODEL_237d1793183e4533acaeafd3fb834d11"
            ],
            "layout": "IPY_MODEL_096fcc68242d4a79bd83fde2d7c3a992"
          }
        },
        "2006193fc5de44f981eed17a5ab509e7": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_a0c6f3f0540848dc823e55a214b3f393",
            "placeholder": "​",
            "style": "IPY_MODEL_a57f853a464140babb57f59e76f7abbc",
            "value": "tokenizer_config.json: 100%"
          }
        },
        "21c738579bc64079b228c6a4f663ad33": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8c2807586c344032be126ffdb7cccd33",
            "placeholder": "​",
            "style": "IPY_MODEL_d618b9446bad42afa687c92d2c5b07b8",
            "value": "model.safetensors: 100%"
          }
        },
        "237d1793183e4533acaeafd3fb834d11": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8e31dafdc75f4605afe0a2c4b0da12dc",
            "placeholder": "​",
            "style": "IPY_MODEL_fdd5396749e54b9c9175fbc7540c2dd2",
            "value": " 1.56G/1.56G [01:05&lt;00:00, 23.7MB/s]"
          }
        },
        "2ac35836e3214fb8a61c595413375e3b": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_af742a527e114c7383d36b6ff5b1d16f",
            "placeholder": "​",
            "style": "IPY_MODEL_884931a58bd94505b1b0fdc060a86433",
            "value": "pytorch_model.bin: 100%"
          }
        },
        "2cd71fa3a8db4176bf6738ea79aeb919": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "2e2debb2d1364986b9d2f89b87b7b648": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "331719c0890842adb135a007ed64c5d2": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_15010981b124459fb59b9b643361945b",
            "placeholder": "​",
            "style": "IPY_MODEL_0f0f144d459d443c93bc16a04ebc3041",
            "value": "config.json: 100%"
          }
        },
        "336ee518677b4aa38215224668044067": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "3528677b8beb496da5e99d650cb9954f": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_331719c0890842adb135a007ed64c5d2",
              "IPY_MODEL_7f61ec32fff54e8a86b81033d447c224",
              "IPY_MODEL_c88de913b57445f28bae4e02fa480d84"
            ],
            "layout": "IPY_MODEL_08b5c84493fd4557a44c4be3a86e903b"
          }
        },
        "374f7999b6044aab820c205222cd4180": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_593d005df53346dabc08434184e94dfa",
            "placeholder": "​",
            "style": "IPY_MODEL_d93d4c49a28d40fcb4fc29095e0f08df",
            "value": "vocab.json: 100%"
          }
        },
        "39bf7f982e934c0cbace16fe01390308": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_2006193fc5de44f981eed17a5ab509e7",
              "IPY_MODEL_fc01e398f18946f1ab8e3602abfffae1",
              "IPY_MODEL_1b261d960e1544828604f17bfec16dfd"
            ],
            "layout": "IPY_MODEL_43265c2f5a09465ca1265804a8730a83"
          }
        },
        "3b66bd0ed0a042c991f08aba3ac1408f": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "3bc0982573374f5fa808a9c538911014": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_688230e990f4463e9d2331d13f420698",
            "placeholder": "​",
            "style": "IPY_MODEL_f6f1b644e1f64939ac73373045b05744",
            "value": "merges.txt: 100%"
          }
        },
        "3bc2a1499dae448e92f1fbf7079cc24f": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "40653463cbf04cee9a41be0d117f8f2d": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "43265c2f5a09465ca1265804a8730a83": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "44328b117863496b92e77e17b08f9769": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8b2ec685bae84da58720089676789bc0",
            "max": 696298,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_93b5980a620949e98ae7c89adcadf08c",
            "value": 696298
          }
        },
        "471c2cc1344a4773a8eb917c7cb5f78f": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "4887e407b61742a0bbbd9afb596a24ea": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_374f7999b6044aab820c205222cd4180",
              "IPY_MODEL_749b2a61b6ba424f9f4c28eaaedcde8a",
              "IPY_MODEL_93e36dc05d794cf0b2fdc08d3b1a21c2"
            ],
            "layout": "IPY_MODEL_855f2978e943473fa47c24042f3c8ab7"
          }
        },
        "4dc671d782494e81b84ab6470bb49b2d": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_c15c0cfbf3bb4817806362ae9caa162e",
            "max": 1560781537,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_c59c14632b6d433db95e5668bb332f63",
            "value": 1560781537
          }
        },
        "50bd19c428b8482fb68209397d6bd135": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "593d005df53346dabc08434184e94dfa": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "5a6281d834b6428b8a3495662b15cfae": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_3bc0982573374f5fa808a9c538911014",
              "IPY_MODEL_44328b117863496b92e77e17b08f9769",
              "IPY_MODEL_969a0195f1ad4316b201c49844b2a9e8"
            ],
            "layout": "IPY_MODEL_78e410d1459447e0a30a2260e6e42664"
          }
        },
        "5a7b5c8f2a424a9088a264a90ab09df2": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "60c533fd078449b6a33064420cce627b": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "61297c1d55ba4a92a7a38636fc70fea3": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "688230e990f4463e9d2331d13f420698": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "6d41fa432c924ba59c878feb184d0599": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "7371365739fb4ea8804364b762d5d982": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "73c6ebaaa0a74f5083d0ec32565efcc7": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "73c936b3860c4e55a7eeac24788c0890": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "749b2a61b6ba424f9f4c28eaaedcde8a": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_6d41fa432c924ba59c878feb184d0599",
            "max": 927059,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_94fec15f4468460bb36164106f3974b6",
            "value": 927059
          }
        },
        "78e410d1459447e0a30a2260e6e42664": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "7c1959c72c4947ef87e123d49c43c203": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_b1fd1d076d544d9a86198dfbb4893c3b",
              "IPY_MODEL_048d2ab5a72744489745fd98e004b97f",
              "IPY_MODEL_a12ee47525d74b70ab71ade8d62366fd"
            ],
            "layout": "IPY_MODEL_8695cfb1d7e14c4ca370805548ea23b7"
          }
        },
        "7d12d690f8fb40fe8d89ac74914ff8b8": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "7f61ec32fff54e8a86b81033d447c224": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ffd8ea3524d544afa0d1aa0c612fbe33",
            "max": 433,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_e0f5598f572d4488b060dd334268b362",
            "value": 433
          }
        },
        "8101283d1df6450cbccf2f94a5c49503": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "83bc57c957db40ffa1d5af96a1706615": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_deb188e60d8f4ce38d2e2fb6b9450bc9",
            "max": 213450,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_5a7b5c8f2a424a9088a264a90ab09df2",
            "value": 213450
          }
        },
        "855f2978e943473fa47c24042f3c8ab7": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8695cfb1d7e14c4ca370805548ea23b7": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "884931a58bd94505b1b0fdc060a86433": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "8a7ddb954b6f49ab968f381cac135e8c": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "8b2ec685bae84da58720089676789bc0": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8b848b313af448d497940b373983255f": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "8c2807586c344032be126ffdb7cccd33": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8d107236529c457f998e1f097beb1381": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "8e31dafdc75f4605afe0a2c4b0da12dc": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "90660f5fd5ed4b68a3cb58b87763e08f": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8101283d1df6450cbccf2f94a5c49503",
            "placeholder": "​",
            "style": "IPY_MODEL_7371365739fb4ea8804364b762d5d982",
            "value": " 213k/213k [00:00&lt;00:00, 12.4MB/s]"
          }
        },
        "93b5980a620949e98ae7c89adcadf08c": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "93e36dc05d794cf0b2fdc08d3b1a21c2": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_cca22a41f2f14b9faf8c14112f21d028",
            "placeholder": "​",
            "style": "IPY_MODEL_336ee518677b4aa38215224668044067",
            "value": " 927k/927k [00:00&lt;00:00, 1.45MB/s]"
          }
        },
        "94fec15f4468460bb36164106f3974b6": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "95ccc8c232e84cfdbb17fcb0234953b1": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "969a0195f1ad4316b201c49844b2a9e8": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_c63bedd10d1e4441a980fec1c1b5283d",
            "placeholder": "​",
            "style": "IPY_MODEL_a7b2f68e27594cf2972db7e0495d2e90",
            "value": " 696k/696k [00:00&lt;00:00, 3.19MB/s]"
          }
        },
        "9dba2695452d4b928cea9eac24f2513a": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "9fbc30b7ef6a4973b141725de13559b5": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "a0c6f3f0540848dc823e55a214b3f393": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a12ee47525d74b70ab71ade8d62366fd": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_50bd19c428b8482fb68209397d6bd135",
            "placeholder": "​",
            "style": "IPY_MODEL_73c936b3860c4e55a7eeac24788c0890",
            "value": " 595/595 [00:00&lt;00:00, 68.6kB/s]"
          }
        },
        "a2fa994e07a048d882a7d5c95eb6c521": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a30dc886fb8a439c8eb36c0e49ab48fe": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a57f853a464140babb57f59e76f7abbc": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "a7b2f68e27594cf2972db7e0495d2e90": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "ab123feb80e54400af0b5f22ceb983bd": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ab8e13db83fd4b41acdfac6f8f164baa",
            "max": 112,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_1d0146ca0668478fa8d70017e7666286",
            "value": 112
          }
        },
        "ab8e13db83fd4b41acdfac6f8f164baa": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "af742a527e114c7383d36b6ff5b1d16f": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "afc691399d0448ed9dea102c5f9b8550": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "b1fd1d076d544d9a86198dfbb4893c3b": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_a30dc886fb8a439c8eb36c0e49ab48fe",
            "placeholder": "​",
            "style": "IPY_MODEL_60c533fd078449b6a33064420cce627b",
            "value": "config.json: 100%"
          }
        },
        "b3feaa73f0dd4d3aa0bef52e6e8b32f7": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ec3a0b641bd14f87a4b22f5775e19ca7",
            "placeholder": "​",
            "style": "IPY_MODEL_0f230cea07d94593875df35e2cce8756",
            "value": " 112/112 [00:00&lt;00:00, 10.6kB/s]"
          }
        },
        "bd7a1e928b814ab6ba49437e02a2a25b": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_d13288b54ab240e1ae73afce651cb468",
            "placeholder": "​",
            "style": "IPY_MODEL_9fbc30b7ef6a4973b141725de13559b5",
            "value": "vocab.txt: 100%"
          }
        },
        "c15c0cfbf3bb4817806362ae9caa162e": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "c37a8173761f4e578940007e2e4f8b93": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_95ccc8c232e84cfdbb17fcb0234953b1",
            "max": 435755812,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_7d12d690f8fb40fe8d89ac74914ff8b8",
            "value": 435755812
          }
        },
        "c59c14632b6d433db95e5668bb332f63": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "c63bedd10d1e4441a980fec1c1b5283d": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "c88de913b57445f28bae4e02fa480d84": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_098596aaddfa4691a4692dc42be29f2c",
            "placeholder": "​",
            "style": "IPY_MODEL_471c2cc1344a4773a8eb917c7cb5f78f",
            "value": " 433/433 [00:00&lt;00:00, 43.4kB/s]"
          }
        },
        "cb26c8edff53437ea89bf3d71e428b5a": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_21c738579bc64079b228c6a4f663ad33",
              "IPY_MODEL_c37a8173761f4e578940007e2e4f8b93",
              "IPY_MODEL_ff9a7ed635274fd18693ce926273c328"
            ],
            "layout": "IPY_MODEL_3bc2a1499dae448e92f1fbf7079cc24f"
          }
        },
        "cca22a41f2f14b9faf8c14112f21d028": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d0df9b897fc344fba3cfb3d595516b60": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d13288b54ab240e1ae73afce651cb468": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d3711321d1f84e03a7ddf36c53cb26ad": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_fae3b45a792b4356b1c7e3499eb79bcc",
            "placeholder": "​",
            "style": "IPY_MODEL_0f06b0e83cb84808aa794fb461a638f7",
            "value": " 1.56G/1.56G [01:05&lt;00:00, 24.1MB/s]"
          }
        },
        "d5f8db856320461d9692488592484d9c": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_bd7a1e928b814ab6ba49437e02a2a25b",
              "IPY_MODEL_83bc57c957db40ffa1d5af96a1706615",
              "IPY_MODEL_90660f5fd5ed4b68a3cb58b87763e08f"
            ],
            "layout": "IPY_MODEL_e08d37c0ceb741a4b0ac10dc08c0b2b5"
          }
        },
        "d618b9446bad42afa687c92d2c5b07b8": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "d8597a95fcb541f6882faf2ddfa95ebd": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_40653463cbf04cee9a41be0d117f8f2d",
            "placeholder": "​",
            "style": "IPY_MODEL_2e2debb2d1364986b9d2f89b87b7b648",
            "value": "model.safetensors: 100%"
          }
        },
        "d93d4c49a28d40fcb4fc29095e0f08df": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "deb188e60d8f4ce38d2e2fb6b9450bc9": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "e08d37c0ceb741a4b0ac10dc08c0b2b5": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "e0f5598f572d4488b060dd334268b362": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "ebbed84720d64bff882b5156c010c66f": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_0773892c78764e7fb0af2342cb4214e3",
              "IPY_MODEL_ab123feb80e54400af0b5f22ceb983bd",
              "IPY_MODEL_b3feaa73f0dd4d3aa0bef52e6e8b32f7"
            ],
            "layout": "IPY_MODEL_61297c1d55ba4a92a7a38636fc70fea3"
          }
        },
        "ec3a0b641bd14f87a4b22f5775e19ca7": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "f48ef23ec1ac4de89e064941998a52de": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "ProgressStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "f6f1b644e1f64939ac73373045b05744": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "f73ad57413ab46198a7189a4997fc069": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "f77a63fea48a44dc9ab4738e252d875b": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HBoxModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_2ac35836e3214fb8a61c595413375e3b",
              "IPY_MODEL_4dc671d782494e81b84ab6470bb49b2d",
              "IPY_MODEL_d3711321d1f84e03a7ddf36c53cb26ad"
            ],
            "layout": "IPY_MODEL_2cd71fa3a8db4176bf6738ea79aeb919"
          }
        },
        "fae3b45a792b4356b1c7e3499eb79bcc": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "fb0c404be8e14827808cef036eb9f703": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "fc01e398f18946f1ab8e3602abfffae1": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "FloatProgressModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_73c6ebaaa0a74f5083d0ec32565efcc7",
            "max": 51,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_f48ef23ec1ac4de89e064941998a52de",
            "value": 51
          }
        },
        "fdd5396749e54b9c9175fbc7540c2dd2": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "ff9a7ed635274fd18693ce926273c328": {
          "model_module": "@jupyter-widgets/controls",
          "model_module_version": "1.5.0",
          "model_name": "HTMLModel",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_d0df9b897fc344fba3cfb3d595516b60",
            "placeholder": "​",
            "style": "IPY_MODEL_8b848b313af448d497940b373983255f",
            "value": " 436M/436M [00:02&lt;00:00, 211MB/s]"
          }
        },
        "ffd8ea3524d544afa0d1aa0c612fbe33": {
          "model_module": "@jupyter-widgets/base",
          "model_module_version": "1.2.0",
          "model_name": "LayoutModel",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        }
      }
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
